<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>basics</title>
    <style>
        /* Add your custom CSS styles here */
    </style>
</head>
<body>
    
        
            <div class="markdown">
                <h1 id="Bigraph-Schema-Basics">Bigraph Schema Basics<a class="anchor-link" href="#Bigraph-Schema-Basics">&#182;</a></h1><p>This tutorial demonstrates use of bigraph-viz, a visualization tool for bigraph composition schema. 
The tutorial stars with a single <code>store</code> and increases in detail as you scroll to the bottom.</p>
<p>See the <a href="https://github.com/vivarium-collective/bigraph-viz">Bigraph-viz GitHub</a> for the source code.</p>

            </div>
        
    
        
            <div class="markdown">
                <h2 id="Imports-and-settings">Imports and settings<a class="anchor-link" href="#Imports-and-settings">&#182;</a></h2>
            </div>
        
    
        
            <div class="code">
                <pre>from bigraph_viz import plot_bigraph, plot_flow, plot_multitimestep

plot_settings = {
    'remove_process_place_edges': True,
}
save_images = False
if save_images:
    plot_settings.update({
        'out_dir': 'out',
        'dpi': '250'
    })</pre>
                
            </div>
        
    
        
            <div class="markdown">
                <h2 id="Stores">Stores<a class="anchor-link" href="#Stores">&#182;</a></h2>
            </div>
        
    
        
            <div class="markdown">
                <h3 id="Simple-stores">Simple stores<a class="anchor-link" href="#Simple-stores">&#182;</a></h3><p>An store can be declared in a JSON dictionary with the store's name matched to a dictionary. Leaf node values are declared by the <code>_value</code> key and types declared by the <code>_type</code> key.</p>

            </div>
        
    
        
            <div class="code">
                <pre>simple_store_spec = {
    'store1': {
        '_value': 1.0, 
        '_type': 'float',
    },
}
plot_bigraph(simple_store_spec, **plot_settings, plot_schema=True, filename='store')</pre>
                
                    
                        
                            
                                <pre><graphviz.graphs.Digraph at 0x105dafb20></pre>
                            
                        
                    
                
            </div>
        
    
        
            <div class="markdown">
                <h3 id="Hierarchy">Hierarchy<a class="anchor-link" href="#Hierarchy">&#182;</a></h3><p>A hierarchy is a place graph of nested stores. Stores can be placed within stores using JSON dictionary nesting.</p>

            </div>
        
    
        
            <div class="code">
                <pre>hierarchy_spec = {
    'store1': {
        'store1.1': {
            'store1.1.1': {
                '_value': 1.1, 
                '_type': 'float',
            },
            'store1.1.2': {
                '_value': 2, 
                '_type': 'int',
            },
            'store1.1.3': {
                '_value': 0.01, 
                '_type': 'float',
                'store1.1.3.1': {
                    '_value': 0.1, 
                    '_type': 'float',
                },
            },
        },
        'store1.2': {
            '_value': 1, 
            '_type': 'int',
        },
    },
}
plot_bigraph(hierarchy_spec, **plot_settings, filename='hierarchy')</pre>
                
                    
                        
                            
                                <pre><graphviz.graphs.Digraph at 0x1070beb20></pre>
                            
                        
                    
                
            </div>
        
    
        
            <div class="markdown">
                <h2 id="Processes">Processes<a class="anchor-link" href="#Processes">&#182;</a></h2>
            </div>
        
    
        
            <div class="markdown">
                <h3 id="Single-process">Single process<a class="anchor-link" href="#Single-process">&#182;</a></h3><p>Processes require ports, which are declared with the <code>_ports</code> key mapped to a dict that has port names matched to types.</p>

            </div>
        
    
        
            <div class="code">
                <pre>process_spec = {
    'process1': {
        '_ports': {
            'port1': {'_type': 'type'},
            'port2': {'_type': 'type'},
        },
    },
}
plot_bigraph(process_spec, **plot_settings, rankdir='RL', filename='process')</pre>
                
                    
                        
                            
                                <pre><graphviz.graphs.Digraph at 0x105daf760></pre>
                            
                        
                    
                
            </div>
        
    
        
            <div class="markdown">
                <h3 id="Multiple-processes">Multiple processes<a class="anchor-link" href="#Multiple-processes">&#182;</a></h3>
            </div>
        
    
        
            <div class="code">
                <pre>process_schema = {
    '_ports': {
        'port1': {'_type': 'type'},
        'port2': {'_type': 'type'}
    }
}

processes_spec = {
    'process1': process_schema,
    'process2': process_schema,
    'process3': process_schema,
}
plot_bigraph(processes_spec, **plot_settings, rankdir='BT', filename='multiple_processes')</pre>
                
                    
                        
                            
                                <pre><graphviz.graphs.Digraph at 0x1070be1c0></pre>
                            
                        
                    
                
            </div>
        
    
        
            <div class="markdown">
                <h2 id="Wires">Wires<a class="anchor-link" href="#Wires">&#182;</a></h2><p>To connect a process's ports to stores, add wiring with a <code>_wires</code> key that maps port names to relative paths in the store hierarchy. The ports used by the wires need to match the ports in the schema.</p>

            </div>
        
    
        
            <div class="code">
                <pre>connected_process_spec = {
    'process1': {
        '_ports': {
            'port1': {'_type': 'type'},
            'port2': {'_type': 'type'},
        },
        '_wires': {
            'port1': 'node1',
            'port2': 'node2',
        }
    },
    'node1': {
        '_value': 1.1, '_type': 'float'
    },
    'node2': {
        '_value': 2, '_type': 'int'
    },
}
plot_bigraph(connected_process_spec, **plot_settings, filename='wires')</pre>
                
                    
                        
                            
                                <pre><graphviz.graphs.Digraph at 0x1070beaf0></pre>
                            
                        
                    
                
            </div>
        
    
        
            <div class="markdown">
                <h3 id="Advanced-wiring">Advanced wiring<a class="anchor-link" href="#Advanced-wiring">&#182;</a></h3><p><strong>TODO</strong></p>

            </div>
        
    
        
            <div class="markdown">
                <h2 id="Composites">Composites<a class="anchor-link" href="#Composites">&#182;</a></h2>
            </div>
        
    
        
            <div class="markdown">
                <h3 id="Flat-composite">Flat composite<a class="anchor-link" href="#Flat-composite">&#182;</a></h3><p>A flat composite connects stores with multiple connected processes, without nesting.</p>

            </div>
        
    
        
            <div class="code">
                <pre>flat_composite_spec = {
    'store1.1': {
        '_value': 1.1, 
        '_type': 'float',
    },
    'store1.2': {
        '_value': 2, 
        '_type': 'int',
    },
    'process1': {
        '_ports': {
            'port1': {'_type': 'type'}, 
            'port2': {'_type': 'type'},
        },
        '_wires': {
            'port1': 'store1.1',
            'port2': 'store1.2',
        }
    },
    'process2': {
        '_ports': {
            'port1': {'_type': 'type'}, 
            'port2': {'_type': 'type'},
        },
        '_wires': {
            'port1': 'store1.1',
            'port2': 'store1.2',
        }
    },
}
plot_bigraph(flat_composite_spec, **plot_settings, rankdir='RL', filename='flat_composite')</pre>
                
                    
                        
                            
                                <pre><graphviz.graphs.Digraph at 0x1070dee20></pre>
                            
                        
                    
                
            </div>
        
    
        
            <div class="markdown">
                <h3 id="Nested-composite">Nested composite<a class="anchor-link" href="#Nested-composite">&#182;</a></h3><p>A nested composite has store hierarchies with multiple connected processes.</p>

            </div>
        
    
        
            <div class="code">
                <pre>nested_composite_spec = {
    'store1': {
        'store1.1': {
            '_value': 1.1, 
            '_type': 'float',
        },
        'store1.2': {
            '_value': 2, 
            '_type': 'int',
        },
        'process1': {
            '_ports': {
                'port1': {'_type': 'type'}, 
                'port2': {'_type': 'type'},
            },
            '_wires': {
                'port1': 'store1.1',
                'port2': 'store1.2',
            }
        },
        'process2': {
            '_ports': {
                'port1': {'_type': 'type'}, 
                'port2': {'_type': 'type'},
            },
            '_wires': {
                'port1': 'store1.1',
                'port2': 'store1.2',
            }
        },
    },
    'process3': {
        '_wires': {
            'port1': 'store1',
        }
    }
}
plot_bigraph(nested_composite_spec, **plot_settings, filename='nested_composite')</pre>
                
                    
                        
                            
                                <pre><graphviz.graphs.Digraph at 0x105daf700></pre>
                            
                        
                    
                
            </div>
        
    
        
            <div class="markdown">
                <h3 id="Composite-process">Composite process<a class="anchor-link" href="#Composite-process">&#182;</a></h3><p>Composite processes are processes with internal stores and internal processes, which can run as their own simulation. The composite process has schema, allowing it to connect to a super-simulation. This allows for improved distributed computation, with composite processes able to run on their own computer and syncronize to the super-simulation via message passing.</p>

            </div>
        
    
        
            <div class="code">
                <pre>composite_process_spec = {
    'composite_process': {
        'store1.1': {
            '_value': 1.1, 
            '_type': 'float',
        },
        'store1.2': {
            '_value': 2, 
            '_type': 'int',
        },
        'process1': {
            '_ports': {
                'port1': 'type', 
                'port2': 'type',
            },
            '_wires': {
                'port1': 'store1.1',
                'port2': 'store1.2',
            }
        },
        'process2': {
            '_ports': {
                'port1': {'_type': 'type'}, 
                'port2': {'_type': 'type'},
            },
            '_wires': {
                'port1': 'store1.1',
                'port2': 'store1.2',
            }
        },
        '_ports': {
            'port1': {'_type': 'type'}, 
            'port2': {'_type': 'type'}, 
        },
        '_tunnels': {
            'port1': 'store1.1',
            'port2': 'store1.2',
        }
    }
}
plot_bigraph(composite_process_spec, **plot_settings, filename='composite_process')</pre>
                
                    
                        
                            
                                <pre><graphviz.graphs.Digraph at 0x1070de520></pre>
                            
                        
                    
                
            </div>
        
    
        
            <div class="markdown">
                <h2 id="Orchestration">Orchestration<a class="anchor-link" href="#Orchestration">&#182;</a></h2>
            </div>
        
    
        
            <div class="code">
                <pre>plot_settings2={}
if save_images:
    plot_settings2.update({
        'out_dir': 'out',
        'dpi': '250'
    })</pre>
                
            </div>
        
    
        
            <div class="markdown">
                <h3 id="Multi-timestepping">Multi-timestepping<a class="anchor-link" href="#Multi-timestepping">&#182;</a></h3>
            </div>
        
    
        
            <div class="code">
                <pre>multitimestep_spec = {
    'temporal<br/>process2': {
        '_ports': {'port1': {'_type': 'type'}},
        '_wires': {'port1': 'state'},
        '_sync_step': 1.0,
    },
    'temporal<br/>process1': {
        '_ports': {'port1': {'_type': 'type'}},
        '_wires': {'port1': 'state'},
        '_sync_step': 0.5,
    },
}
plot_multitimestep(multitimestep_spec, total_time=2.0, **plot_settings2, filename='multitimestep')</pre>
                
                    
                        
                            
                                <pre><graphviz.graphs.Digraph at 0x1070ded30></pre>
                            
                        
                    
                
            </div>
        
    
        
            <div class="markdown">
                <h3 id="Flows">Flows<a class="anchor-link" href="#Flows">&#182;</a></h3>
            </div>
        
    
        
            <div class="code">
                <pre>process_schema = {
    '_type': 'step_process',
    '_ports': {
        'port1': {'_type': 'type'},
        'port2': {'_type': 'type'}
    }
}

flow_spec = {
    'step1': {
        '_depends_on': [], 
        **process_schema
    },
    'step2': {
        '_depends_on': 'step1', 
        **process_schema
    },
    'step3': {
        '_depends_on': [], 
        **process_schema
    },
    'step4': {
        '_depends_on': ['step2', 'step3'], 
        **process_schema
    },
}
plot_flow(flow_spec, **plot_settings2, filename='flow')</pre>
                
                    
                        
                            
                                <pre><graphviz.graphs.Digraph at 0x1070bebe0></pre>
                            
                        
                    
                
            </div>
        
    
        
            <div class="markdown">
                <h2 id="Examples">Examples<a class="anchor-link" href="#Examples">&#182;</a></h2>
            </div>
        
    
        
            <div class="markdown">
                <h3 id="Cell-structure-and-function">Cell structure and function<a class="anchor-link" href="#Cell-structure-and-function">&#182;</a></h3><p>Bigraphs can represent the hierarchical nesting of cellular structures, and processes represent their functions.</p>

            </div>
        
    
        
            <div class="code">
                <pre>cell_struct_func = {
        'cell': {
            'membrane': {
                'transporters': {'_type': 'concentrations'},
                'lipids': {'_type': 'concentrations'},
                'transmembrane<br/>transport': {
                    '_value': {
                        '_process': 'transport URI',
                        '_config': {'parameter': 1}
                    },
                    '_wires': {
                        'transporters': 'transporters',
                        'internal': ['..', 'cytoplasm', 'metabolites']},
                    '_ports': {
                        'transporters': {'_type': 'concentrations'},
                        'internal': {'_type': 'concentrations'},
                        'external': {'_type': 'concentrations'}
                    }
                }
            },
            'cytoplasm': {
                'metabolites': {
                    '_value': 1.1,
                    '_type': 'concentrations'
                },
                'ribosomal<br/>complexes': {
                    '_value': 2.2,
                    '_type': 'concentrations'
                },
                'transcript<br/>regulation<br/>complex': {
                    '_value': 0.01,
                    '_type': 'concentrations',
                    'transcripts': {
                        '_value': 0.1,
                        '_type': 'concentrations'
                    }
                },
                'translation': {
                    '_wires': {
                        'p1': 'ribosomal<br/>complexes',
                        'p2': ['transcript<br/>regulation<br/>complex', 'transcripts']}}},
            'nucleoid': {
                'chromosome': {
                    'genes': 'sequences'
                }
            }
        }
    }
plot_bigraph(cell_struct_func, plot_schema=False, remove_process_place_edges=True, out_dir='out', filename='cell')</pre>
                
                    
                        
                    
                        
                            
                                <pre><graphviz.graphs.Digraph at 0x1070d9340></pre>
                            
                        
                    
                
            </div>
        
    
        
            <div class="markdown">
                <h3 id="simulation-experiment">simulation experiment<a class="anchor-link" href="#simulation-experiment">&#182;</a></h3><p>The composition schema was developed for integrative simulations, but can also be used for workflows such as simulation experiments./</p>

            </div>
        
    
        
            <div class="code">
                <pre>simulation_experiment_spec = {
    'parameter<br/>estimator': {
        '_ports': {
            'input data': {'_type': 'type'}, 
            'model': {'_type': 'SBML'}, 
            'parameters': {'_type': 'type'},},
        '_wires': {
            'input data': 'observed data',
            'model': 'model',
            'parameters': 'parameters',
        }
    },
    'simulator': {
        '_ports': {
            'model': {'_type': 'SBML'}, 
            'parameters': {'_type': 'type'},
            'total time': {'_type': 'time'},
            'results': {'_type': 'type'}
        },
        '_wires': {
            'parameters': 'parameters',
            'model': 'model',
            'total time': 'total time',
            'results': 'results',
        }
    },
    'analysis': {
        '_ports': {
            'results': {'_type': 'type'}
        },
        '_wires': {
            'results': 'results',
        }
    },
    'parameters': {},
    'model': {},
    'results': {},
    'observed data': {},
    'total time': {},
}

plot_bigraph(simulation_experiment_spec, **plot_settings, rankdir='RL', filename='simulation_experiment')</pre>
                
                    
                        
                            
                                <pre><graphviz.graphs.Digraph at 0x1070d90a0></pre>
                            
                        
                    
                
            </div>
        
    
        
            <div class="code">
                <pre></pre>
                
            </div>
        
    
</body>
</html>