{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5338cfab-cf15-4c63-a091-c1cbb51bf8dd",
   "metadata": {},
   "source": [
    "# Process Bigraph Schema\n",
    "This tutorial covers Process Bigraph Schema, starting with single `store` and increasing in detail as you scroll towards the bottom. It uses the libraries [bigraph-schema](https://github.com/vivarium-collective/bigraph-schema) for the schema validation and type registry and [bigraph-viz](https://github.com/vivarium-collective/bigraph-viz) for visualization.\n",
    "\n",
    "## Bigraph-schema and Bigraph-viz\n",
    "\n",
    "[Bigraph-schema](https://github.com/vivarium-collective/bigraph-schema) is a serializable type schema for Process Bigraph simulations. From this library, we will use the `validate_schema` method to check our schemas, and `fill` to fill in missing details to prepare the schema for simulation. We will also use the `type_registry` to access different schemas, and register new ones.\n",
    "\n",
    "[Bigraph-viz](https://github.com/vivarium-collective/bigraph-viz) is an easy-to-use plotting tool for compositional bigraph schema, built on top of Graphviz. It reads raw Compositional Bigraph Schemas and generates a figure that can be displayed in a notebook or saved to file. Bigraph-viz's main plotting function is `plot_bigraph` for the bigraph structure. `plot_flow` and `plot_multitimestep` make plots of process orchestration."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "df69e36f-3da8-4bfb-8bc0-0aeb04a6f7d7",
   "metadata": {},
   "source": [
    "### Installation"
   ]
  },
  {
   "cell_type": "code",
   "id": "63fb7d1a-cd49-4aac-8d4f-31bb54496a51",
   "metadata": {
    "tags": [],
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:16.581865Z",
     "start_time": "2025-04-06T16:43:16.133457Z"
    }
   },
   "source": [
    "# !pip install bigraph-schema\n",
    "# !pip install bigraph-viz\n",
    "!pip freeze | grep bigraph"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-e git+https://github.com/vivarium-collective/bigraph-schema.git@0112b12c2f063851c3b88463af4bb59ab4f20d70#egg=bigraph_schema\r\n",
      "-e git+https://github.com/vivarium-collective/bigraph-viz.git@5c1057de02b201f67e6ea3f80c1337b08522a1de#egg=bigraph_viz\r\n",
      "-e git+https://github.com/vivarium-collective/process-bigraph.git@f6e29cba3c9342bc302540e7c5e60f9bba38a683#egg=process_bigraph\r\n"
     ]
    }
   ],
   "execution_count": 109
  },
  {
   "cell_type": "markdown",
   "id": "f700afa2-bfee-49bf-a4b1-e23f0540461d",
   "metadata": {},
   "source": [
    "### Imports"
   ]
  },
  {
   "cell_type": "code",
   "id": "2ab3cb32-e007-45c4-807e-b1b442ac1608",
   "metadata": {
    "tags": [],
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:16.594026Z",
     "start_time": "2025-04-06T16:43:16.591871Z"
    }
   },
   "source": [
    "# from bigraph_viz import plot_bigraph, plot_flow, plot_multitimestep, pf\n",
    "from bigraph_viz import plot_bigraph, replace_regex_recursive, VisualizeTypes\n",
    "\n",
    "plot_settings = {\n",
    "    'remove_process_place_edges': True\n",
    "}\n",
    "save_images = True\n",
    "if save_images:\n",
    "    plot_settings.update({\n",
    "        'out_dir': 'out',\n",
    "        'dpi': '70'\n",
    "    })"
   ],
   "outputs": [],
   "execution_count": 110
  },
  {
   "cell_type": "markdown",
   "id": "0e716c23-5fbe-4c08-a868-09611ca40ec4",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Stores"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a6e9cc2-e0df-4f73-b4c3-eddb1055b088",
   "metadata": {},
   "source": [
    "### Simple stores\n",
    "A store can be declared in a JSON dictionary with the store's name mapped to its `value`. "
   ]
  },
  {
   "cell_type": "code",
   "id": "d711b952-089a-4938-817d-d1ccad9de56e",
   "metadata": {
    "tags": [],
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:16.645819Z",
     "start_time": "2025-04-06T16:43:16.598658Z"
    }
   },
   "source": [
    "simple_store_state = {\n",
    "    'store1': 1.0,\n",
    "}\n",
    "plot_bigraph(simple_store_state,\n",
    "             **plot_settings,\n",
    "             show_values=True,\n",
    "             filename='simple_store')"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/simple_store\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n<!-- Generated by graphviz version 9.0.0 (0)\n -->\n<!-- Title: bigraph Pages: 1 -->\n<svg width=\"60pt\" height=\"60pt\"\n viewBox=\"0.00 0.00 62.21 62.21\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 58.21)\">\n<title>bigraph</title>\n<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-58.21 58.21,-58.21 58.21,4 -4,4\"/>\n<!-- (&#39;store1&#39;,) -->\n<g id=\"node1\" class=\"node\">\n<title>(&#39;store1&#39;,)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"27.11\" cy=\"-27.11\" rx=\"27.11\" ry=\"27.11\"/>\n<text text-anchor=\"start\" x=\"7.78\" y=\"-23.51\" font-family=\"Times,serif\" font-size=\"12.00\">store1:1</text>\n</g>\n</g>\n</svg>\n",
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x11abb3430>"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 111
  },
  {
   "cell_type": "markdown",
   "id": "db1cead7-e691-4291-920e-827d69c00727",
   "metadata": {},
   "source": [
    "To include values *and* types in a node label, you can instead have the store name map to a dict with `_value` and `_type` keys."
   ]
  },
  {
   "cell_type": "code",
   "id": "f3e5fa71-7ffc-4a8f-a1a0-dc547c886f84",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:16.697310Z",
     "start_time": "2025-04-06T16:43:16.652662Z"
    }
   },
   "source": [
    "typed_store_spec = {\n",
    "    'store1': {\n",
    "        '_value': 1.0, \n",
    "        '_type': 'float',\n",
    "    },\n",
    "}\n",
    "plot_bigraph(typed_store_spec,\n",
    "             **plot_settings,\n",
    "             show_values=True,\n",
    "             show_types=True,\n",
    "             filename='store')"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/store\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n<!-- Generated by graphviz version 9.0.0 (0)\n -->\n<!-- Title: bigraph Pages: 1 -->\n<svg width=\"58pt\" height=\"58pt\"\n viewBox=\"0.00 0.00 59.66 59.66\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 55.66)\">\n<title>bigraph</title>\n<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-55.66 55.66,-55.66 55.66,4 -4,4\"/>\n<!-- (&#39;store1&#39;,) -->\n<g id=\"node1\" class=\"node\">\n<title>(&#39;store1&#39;,)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"25.83\" cy=\"-25.83\" rx=\"25.83\" ry=\"25.83\"/>\n<text text-anchor=\"start\" x=\"11.16\" y=\"-28.23\" font-family=\"Times,serif\" font-size=\"12.00\">store1</text>\n<text text-anchor=\"start\" x=\"11.16\" y=\"-16.23\" font-family=\"Times,serif\" font-size=\"12.00\">[float]</text>\n</g>\n</g>\n</svg>\n",
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x12a13a940>"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 112
  },
  {
   "cell_type": "markdown",
   "id": "de1011cb-903b-4c15-baed-a62f4afdd6b9",
   "metadata": {},
   "source": [
    "### Hierarchy\n",
    "A hierarchy is a place graph of nested stores. Stores can be placed within stores using JSON dictionary nesting."
   ]
  },
  {
   "cell_type": "code",
   "id": "0f3f0824-f89d-4c6e-830e-2c5a321f7b33",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:16.749578Z",
     "start_time": "2025-04-06T16:43:16.703870Z"
    }
   },
   "source": [
    "hierarchy_spec = {\n",
    "    'store1': {\n",
    "        'store1.1': {\n",
    "            'store1.1.1': 'any',\n",
    "            'store1.1.2': 'any',\n",
    "            'store1.1.3': {\n",
    "                'store1.1.3.1': 'any',\n",
    "            },\n",
    "        },\n",
    "        'store1.2': 'any',\n",
    "    },\n",
    "}\n",
    "plot_bigraph(hierarchy_spec,\n",
    "             **plot_settings,\n",
    "             filename='hierarchy')"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/hierarchy\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n<!-- Generated by graphviz version 9.0.0 (0)\n -->\n<!-- Title: bigraph Pages: 1 -->\n<svg width=\"235pt\" height=\"343pt\"\n viewBox=\"0.00 0.00 241.78 352.56\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 348.56)\">\n<title>bigraph</title>\n<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-348.56 237.78,-348.56 237.78,4 -4,4\"/>\n<!-- (&#39;store1&#39;,) -->\n<g id=\"node1\" class=\"node\">\n<title>(&#39;store1&#39;,)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"155.55\" cy=\"-322.97\" rx=\"21.59\" ry=\"21.59\"/>\n<text text-anchor=\"start\" x=\"140.89\" y=\"-319.37\" font-family=\"Times,serif\" font-size=\"12.00\">store1</text>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.1&#39;) -->\n<g id=\"node2\" class=\"node\">\n<title>(&#39;store1&#39;, &#39;store1.1&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"119.55\" cy=\"-238.47\" rx=\"26.91\" ry=\"26.91\"/>\n<text text-anchor=\"start\" x=\"100.39\" y=\"-234.87\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1</text>\n</g>\n<!-- (&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;) -->\n<g id=\"edge1\" class=\"edge\">\n<title>(&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M147.02,-302.42C141.99,-290.88 135.57,-276.19 130.21,-263.9\"/>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.2&#39;) -->\n<g id=\"node7\" class=\"node\">\n<title>(&#39;store1&#39;, &#39;store1.2&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"191.55\" cy=\"-238.47\" rx=\"26.91\" ry=\"26.91\"/>\n<text text-anchor=\"start\" x=\"172.39\" y=\"-234.87\" font-family=\"Times,serif\" font-size=\"12.00\">store1.2</text>\n</g>\n<!-- (&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.2&#39;) -->\n<g id=\"edge6\" class=\"edge\">\n<title>(&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.2&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M164.08,-302.42C169.11,-290.88 175.52,-276.19 180.89,-263.9\"/>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;) -->\n<g id=\"node3\" class=\"node\">\n<title>(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"37.55\" cy=\"-143.33\" rx=\"32.23\" ry=\"32.23\"/>\n<text text-anchor=\"start\" x=\"13.89\" y=\"-139.73\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1.3</text>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;) -->\n<g id=\"edge2\" class=\"edge\">\n<title>(&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M101.76,-217.26C89.13,-202.92 72.13,-183.61 58.8,-168.46\"/>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.1&#39;) -->\n<g id=\"node5\" class=\"node\">\n<title>(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.1&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"119.55\" cy=\"-143.33\" rx=\"32.23\" ry=\"32.23\"/>\n<text text-anchor=\"start\" x=\"95.89\" y=\"-139.73\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1.1</text>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.1&#39;) -->\n<g id=\"edge4\" class=\"edge\">\n<title>(&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M119.55,-211.03C119.55,-200.27 119.55,-187.74 119.55,-176.46\"/>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.2&#39;) -->\n<g id=\"node6\" class=\"node\">\n<title>(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.2&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"201.55\" cy=\"-143.33\" rx=\"32.23\" ry=\"32.23\"/>\n<text text-anchor=\"start\" x=\"177.89\" y=\"-139.73\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1.2</text>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.2&#39;) -->\n<g id=\"edge5\" class=\"edge\">\n<title>(&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.2&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M137.34,-217.26C149.97,-202.92 166.97,-183.61 180.3,-168.46\"/>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;, &#39;store1.1.3.1&#39;) -->\n<g id=\"node4\" class=\"node\">\n<title>(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;, &#39;store1.1.3.1&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"37.55\" cy=\"-37.55\" rx=\"37.55\" ry=\"37.55\"/>\n<text text-anchor=\"start\" x=\"9.39\" y=\"-33.95\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1.3.1</text>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;, &#39;store1.1.3.1&#39;) -->\n<g id=\"edge3\" class=\"edge\">\n<title>(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;, &#39;store1.1.3.1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M37.55,-110.34C37.55,-99.5 37.55,-87.3 37.55,-76.07\"/>\n</g>\n</g>\n</svg>\n",
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x12a207c10>"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 113
  },
  {
   "cell_type": "markdown",
   "id": "c825e07c-8cd5-4df1-8958-aa0f36d61045",
   "metadata": {},
   "source": [
    "## Processes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47ea1a83-aeac-4048-9593-7b30588b0ad8",
   "metadata": {},
   "source": [
    "### Single process\n",
    "Processes require ports, which are declared with the `_ports` key mapped to a dict that has port names matched to types."
   ]
  },
  {
   "cell_type": "code",
   "id": "7c74237b-0d66-4e8f-a8ca-79174fadad8b",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:16.803795Z",
     "start_time": "2025-04-06T16:43:16.757339Z"
    }
   },
   "source": [
    "process_spec = {\n",
    "    'process1': {\n",
    "        '_type': 'edge',\n",
    "        '_inputs': {\n",
    "            'port1': 'any',\n",
    "            'port2': 'any',\n",
    "        },\n",
    "    },\n",
    "}\n",
    "plot_bigraph(process_spec,\n",
    "             **plot_settings,\n",
    "             rankdir='RL',\n",
    "             filename='process')"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/process\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n<!-- Generated by graphviz version 9.0.0 (0)\n -->\n<!-- Title: bigraph Pages: 1 -->\n<svg width=\"142pt\" height=\"78pt\"\n viewBox=\"0.00 0.00 146.26 80.15\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 76.15)\">\n<title>bigraph</title>\n<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-76.15 142.26,-76.15 142.26,4 -4,4\"/>\n<!-- (&#39;process1&#39;,) -->\n<g id=\"node1\" class=\"node\">\n<title>(&#39;process1&#39;,)</title>\n<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"54,-53.58 0,-53.58 0,-17.58 54,-17.58 54,-53.58\"/>\n<text text-anchor=\"start\" x=\"6.01\" y=\"-31.98\" font-family=\"Times,serif\" font-size=\"12.00\">process1</text>\n</g>\n<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)_input -->\n<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)_input&#45;&gt;(&#39;process1&#39;,) -->\n<g id=\"edge1\" class=\"edge\">\n<title>(&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)_input&#45;&gt;(&#39;process1&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M110.78,-55.47C99.18,-52.69 81.54,-48.45 65.52,-44.6\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"66.8,-41.3 56.26,-42.37 65.17,-48.11 66.8,-41.3\"/>\n<text text-anchor=\"start\" x=\"72\" y=\"-52.58\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n</g>\n<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)_input -->\n<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)_input&#45;&gt;(&#39;process1&#39;,) -->\n<g id=\"edge2\" class=\"edge\">\n<title>(&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)_input&#45;&gt;(&#39;process1&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M110.78,-16.54C99.29,-19.19 81.88,-23.19 65.99,-26.84\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"65.23,-23.43 56.26,-29.08 66.79,-30.25 65.23,-23.43\"/>\n<text text-anchor=\"start\" x=\"72\" y=\"-26.58\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n</g>\n</g>\n</svg>\n",
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x12a2da220>"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 114
  },
  {
   "cell_type": "markdown",
   "id": "68a2c480-32fe-43d3-b618-8fc6c647b8ca",
   "metadata": {},
   "source": [
    "### Multiple processes"
   ]
  },
  {
   "cell_type": "code",
   "id": "8c2457a3-fc47-4850-8b9e-9ee98e81567d",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:16.857500Z",
     "start_time": "2025-04-06T16:43:16.811210Z"
    }
   },
   "source": [
    "process_schema = {\n",
    "    '_type': 'process',\n",
    "    '_inputs': {\n",
    "        'port1': 'Any',\n",
    "    },\n",
    "    '_outputs': {\n",
    "        'port2': 'Any'\n",
    "    },\n",
    "}\n",
    "\n",
    "processes_spec = {\n",
    "    'process1': process_schema,\n",
    "    'process2': process_schema,\n",
    "    'process3': process_schema,\n",
    "}\n",
    "plot_bigraph(processes_spec,\n",
    "             **plot_settings,\n",
    "             rankdir='BT',\n",
    "             filename='multiple_processes')"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/multiple_processes\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n<!-- Generated by graphviz version 9.0.0 (0)\n -->\n<!-- Title: bigraph Pages: 1 -->\n<svg width=\"266pt\" height=\"114pt\"\n viewBox=\"0.00 0.00 273.15 117.15\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 113.15)\">\n<title>bigraph</title>\n<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-113.15 269.15,-113.15 269.15,4 -4,4\"/>\n<!-- (&#39;process1&#39;,) -->\n<g id=\"node1\" class=\"node\">\n<title>(&#39;process1&#39;,)</title>\n<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"80.58,-109.15 26.58,-109.15 26.58,-73.15 80.58,-73.15 80.58,-109.15\"/>\n<text text-anchor=\"start\" x=\"32.58\" y=\"-87.55\" font-family=\"Times,serif\" font-size=\"12.00\">process1</text>\n</g>\n<!-- (&#39;process2&#39;,) -->\n<g id=\"node2\" class=\"node\">\n<title>(&#39;process2&#39;,)</title>\n<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"170.58,-109.15 116.58,-109.15 116.58,-73.15 170.58,-73.15 170.58,-109.15\"/>\n<text text-anchor=\"start\" x=\"122.58\" y=\"-87.55\" font-family=\"Times,serif\" font-size=\"12.00\">process2</text>\n</g>\n<!-- (&#39;process3&#39;,) -->\n<g id=\"node3\" class=\"node\">\n<title>(&#39;process3&#39;,)</title>\n<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"251.58,-109.15 197.58,-109.15 197.58,-73.15 251.58,-73.15 251.58,-109.15\"/>\n<text text-anchor=\"start\" x=\"203.58\" y=\"-87.55\" font-family=\"Times,serif\" font-size=\"12.00\">process3</text>\n</g>\n<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)_input -->\n<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)_input&#45;&gt;(&#39;process1&#39;,) -->\n<g id=\"edge1\" class=\"edge\">\n<title>(&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)_input&#45;&gt;(&#39;process1&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M18.16,-27.29C21.26,-35.43 25.66,-46.13 30.47,-55.15 31.81,-57.68 33.29,-60.26 34.83,-62.82\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"31.78,-64.55 40.1,-71.11 37.69,-60.79 31.78,-64.55\"/>\n<text text-anchor=\"start\" x=\"30.58\" y=\"-47.15\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n</g>\n<!-- (&#39;process2&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)_input -->\n<!-- (&#39;process2&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)_input&#45;&gt;(&#39;process2&#39;,) -->\n<g id=\"edge2\" class=\"edge\">\n<title>(&#39;process2&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)_input&#45;&gt;(&#39;process2&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M115.64,-27.61C117.72,-35.69 120.76,-46.2 124.47,-55.15 125.44,-57.49 126.52,-59.88 127.66,-62.25\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"124.49,-63.74 132.2,-71.01 130.71,-60.52 124.49,-63.74\"/>\n<text text-anchor=\"start\" x=\"123.58\" y=\"-47.15\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n</g>\n<!-- (&#39;process3&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)_input -->\n<!-- (&#39;process3&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)_input&#45;&gt;(&#39;process3&#39;,) -->\n<g id=\"edge3\" class=\"edge\">\n<title>(&#39;process3&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)_input&#45;&gt;(&#39;process3&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M205.92,-27.92C205.84,-35.92 206.3,-46.25 208.47,-55.15 209.02,-57.44 209.73,-59.76 210.54,-62.05\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"207.21,-63.16 214.26,-71.07 213.68,-60.49 207.21,-63.16\"/>\n<text text-anchor=\"start\" x=\"207.58\" y=\"-47.15\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n</g>\n<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)_output -->\n<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)_output&#45;&gt;(&#39;process1&#39;,) -->\n<g id=\"edge4\" class=\"edge\">\n<title>(&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)_output&#45;&gt;(&#39;process1&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M56.94,-39.31C56.23,-50.04 55.41,-62.38 54.76,-72.21\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"60.41,-39.81 57.58,-29.6 53.43,-39.35 60.41,-39.81\"/>\n<text text-anchor=\"start\" x=\"55.58\" y=\"-47.15\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n</g>\n<!-- (&#39;process2&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)_output -->\n<!-- (&#39;process2&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)_output&#45;&gt;(&#39;process2&#39;,) -->\n<g id=\"edge5\" class=\"edge\">\n<title>(&#39;process2&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)_output&#45;&gt;(&#39;process2&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M153.05,-39.04C151.04,-49.88 148.72,-62.39 146.88,-72.33\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"156.48,-39.7 154.86,-29.23 149.6,-38.42 156.48,-39.7\"/>\n<text text-anchor=\"start\" x=\"151.58\" y=\"-47.15\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n</g>\n<!-- (&#39;process3&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)_output -->\n<!-- (&#39;process3&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)_output&#45;&gt;(&#39;process3&#39;,) -->\n<g id=\"edge6\" class=\"edge\">\n<title>(&#39;process3&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)_output&#45;&gt;(&#39;process3&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M243.21,-37.99C239.28,-49 234.66,-61.93 231,-72.16\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"246.42,-39.42 246.49,-28.82 239.83,-37.06 246.42,-39.42\"/>\n<text text-anchor=\"start\" x=\"239.58\" y=\"-47.15\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n</g>\n</g>\n</svg>\n",
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x12a2dac40>"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 115
  },
  {
   "cell_type": "markdown",
   "id": "15e5f5a6-8dd4-41c3-ba37-33ed9e3e0830",
   "metadata": {},
   "source": [
    "## Wires\n",
    "To connect a process's ports to stores, add wiring with a `_wires` key that maps port names to relative paths in the store hierarchy. The ports used by the wires need to match the ports in the schema."
   ]
  },
  {
   "cell_type": "code",
   "id": "a6ecad5d-8948-4636-910c-79284a3e8679",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:16.912675Z",
     "start_time": "2025-04-06T16:43:16.864539Z"
    }
   },
   "source": [
    "connected_process_spec = {\n",
    "    'process1': {\n",
    "        '_type': 'edge',\n",
    "        '_inputs': {\n",
    "            'port1': 'float',\n",
    "            'port2': 'integer',\n",
    "            'port3': 'integer',\n",
    "        },\n",
    "        'inputs': {\n",
    "            'port1': ['node1'],\n",
    "            'port2': ['node2'],\n",
    "            'port3': ['node3'],\n",
    "        }\n",
    "    },\n",
    "    'node1': 'float',  # TODO -- shouldn't these fill?\n",
    "    'node2': 'integer',\n",
    "    'node3': 'integer',\n",
    "}\n",
    "plot_bigraph(connected_process_spec,\n",
    "             **plot_settings,\n",
    "             filename='wires')"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/wires\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n<!-- Generated by graphviz version 9.0.0 (0)\n -->\n<!-- Title: bigraph Pages: 1 -->\n<svg width=\"168pt\" height=\"129pt\"\n viewBox=\"0.00 0.00 173.18 133.18\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 129.18)\">\n<title>bigraph</title>\n<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-129.18 169.18,-129.18 169.18,4 -4,4\"/>\n<!-- (&#39;node1&#39;,) -->\n<g id=\"node1\" class=\"node\">\n<title>(&#39;node1&#39;,)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"21.59\" cy=\"-103.59\" rx=\"21.59\" ry=\"21.59\"/>\n<text text-anchor=\"start\" x=\"6.93\" y=\"-99.99\" font-family=\"Times,serif\" font-size=\"12.00\">node1</text>\n</g>\n<!-- (&#39;process1&#39;,) -->\n<g id=\"node4\" class=\"node\">\n<title>(&#39;process1&#39;,)</title>\n<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"109.59,-36 55.59,-36 55.59,0 109.59,0 109.59,-36\"/>\n<text text-anchor=\"start\" x=\"61.6\" y=\"-14.4\" font-family=\"Times,serif\" font-size=\"12.00\">process1</text>\n</g>\n<!-- (&#39;node1&#39;,)&#45;&gt;(&#39;process1&#39;,) -->\n<g id=\"edge3\" class=\"edge\">\n<title>(&#39;node1&#39;,)&#45;&gt;(&#39;process1&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M34.23,-85.27C42.54,-73.88 53.59,-58.75 63.02,-45.81\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"65.59,-48.24 68.65,-38.1 59.93,-44.11 65.59,-48.24\"/>\n<text text-anchor=\"start\" x=\"55.59\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n</g>\n<!-- (&#39;node2&#39;,) -->\n<g id=\"node2\" class=\"node\">\n<title>(&#39;node2&#39;,)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"82.59\" cy=\"-103.59\" rx=\"21.59\" ry=\"21.59\"/>\n<text text-anchor=\"start\" x=\"67.93\" y=\"-99.99\" font-family=\"Times,serif\" font-size=\"12.00\">node2</text>\n</g>\n<!-- (&#39;node2&#39;,)&#45;&gt;(&#39;process1&#39;,) -->\n<g id=\"edge1\" class=\"edge\">\n<title>(&#39;node2&#39;,)&#45;&gt;(&#39;process1&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M82.59,-81.06C82.59,-71.08 82.59,-59.07 82.59,-48.28\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"86.09,-48.3 82.59,-38.3 79.09,-48.3 86.09,-48.3\"/>\n<text text-anchor=\"start\" x=\"82.59\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n</g>\n<!-- (&#39;node3&#39;,) -->\n<g id=\"node3\" class=\"node\">\n<title>(&#39;node3&#39;,)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"143.59\" cy=\"-103.59\" rx=\"21.59\" ry=\"21.59\"/>\n<text text-anchor=\"start\" x=\"128.93\" y=\"-99.99\" font-family=\"Times,serif\" font-size=\"12.00\">node3</text>\n</g>\n<!-- (&#39;node3&#39;,)&#45;&gt;(&#39;process1&#39;,) -->\n<g id=\"edge2\" class=\"edge\">\n<title>(&#39;node3&#39;,)&#45;&gt;(&#39;process1&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M130.95,-85.27C122.64,-73.88 111.59,-58.75 102.16,-45.81\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"105.25,-44.11 96.53,-38.1 99.59,-48.24 105.25,-44.11\"/>\n<text text-anchor=\"start\" x=\"114.59\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port3</text>\n</g>\n</g>\n</svg>\n",
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x12a2b04c0>"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 116
  },
  {
   "cell_type": "markdown",
   "id": "e037285d-116a-404d-96ec-ba47318a2e8b",
   "metadata": {},
   "source": [
    "### Advanced wiring\n",
    "**TODO**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01b17201-2489-45c8-a06b-9fdc29a714a7",
   "metadata": {},
   "source": [
    "## Composites"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "daf5d428-db23-4985-80ff-6b93d22973aa",
   "metadata": {},
   "source": [
    "### Flat composite\n",
    "A flat composite connects stores with multiple connected processes, without nesting."
   ]
  },
  {
   "cell_type": "code",
   "id": "781a84e6-bbe2-4708-b441-47efc611b46f",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:16.970172Z",
     "start_time": "2025-04-06T16:43:16.919201Z"
    }
   },
   "source": [
    "flat_composite_spec = {\n",
    "    'store1.1': 'float',\n",
    "    'store1.2': 'integer',\n",
    "    'process1': {\n",
    "        '_type': 'process',\n",
    "        '_outputs': {\n",
    "            'port1': 'float',\n",
    "            'port2': 'integer',\n",
    "        },\n",
    "        'outputs': {\n",
    "            'port1': ['store1.1'],\n",
    "            'port2': ['store1.2'],\n",
    "        }\n",
    "    },\n",
    "    'process2': {\n",
    "        '_type': 'process',\n",
    "        '_inputs': {\n",
    "            'port1': 'float',\n",
    "            'port2': 'integer',\n",
    "        },\n",
    "        'inputs': {\n",
    "            'port1': ['store1.1'],\n",
    "            'port2': ['store1.2'],\n",
    "        }\n",
    "    },\n",
    "}\n",
    "plot_bigraph(flat_composite_spec,\n",
    "             **plot_settings,\n",
    "             rankdir='RL',\n",
    "             filename='flat_composite')"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/flat_composite\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n<!-- Generated by graphviz version 9.0.0 (0)\n -->\n<!-- Title: bigraph Pages: 1 -->\n<svg width=\"168pt\" height=\"137pt\"\n viewBox=\"0.00 0.00 172.93 140.82\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 136.82)\">\n<title>bigraph</title>\n<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-136.82 168.93,-136.82 168.93,4 -4,4\"/>\n<!-- (&#39;store1.1&#39;,) -->\n<g id=\"node1\" class=\"node\">\n<title>(&#39;store1.1&#39;,)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"138.02\" cy=\"-105.91\" rx=\"26.91\" ry=\"26.91\"/>\n<text text-anchor=\"start\" x=\"118.86\" y=\"-102.31\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1</text>\n</g>\n<!-- (&#39;process1&#39;,) -->\n<g id=\"node3\" class=\"node\">\n<title>(&#39;process1&#39;,)</title>\n<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"54,-118.91 0,-118.91 0,-82.91 54,-82.91 54,-118.91\"/>\n<text text-anchor=\"start\" x=\"6.01\" y=\"-97.31\" font-family=\"Times,serif\" font-size=\"12.00\">process1</text>\n</g>\n<!-- (&#39;store1.1&#39;,)&#45;&gt;(&#39;process1&#39;,) -->\n<g id=\"edge4\" class=\"edge\">\n<title>(&#39;store1.1&#39;,)&#45;&gt;(&#39;process1&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M99.1,-104.17C84.49,-103.5 68.12,-102.75 54.77,-102.14\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"98.69,-107.66 108.84,-104.62 99.01,-100.66 98.69,-107.66\"/>\n<text text-anchor=\"start\" x=\"72\" y=\"-104.91\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n</g>\n<!-- (&#39;process2&#39;,) -->\n<g id=\"node4\" class=\"node\">\n<title>(&#39;process2&#39;,)</title>\n<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"54,-53.91 0,-53.91 0,-17.91 54,-17.91 54,-53.91\"/>\n<text text-anchor=\"start\" x=\"6.01\" y=\"-32.31\" font-family=\"Times,serif\" font-size=\"12.00\">process2</text>\n</g>\n<!-- (&#39;store1.1&#39;,)&#45;&gt;(&#39;process2&#39;,) -->\n<g id=\"edge2\" class=\"edge\">\n<title>(&#39;store1.1&#39;,)&#45;&gt;(&#39;process2&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M114.43,-91.41C99.94,-82.1 80.79,-69.81 64.21,-59.16\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"66.37,-56.39 56.07,-53.93 62.59,-62.28 66.37,-56.39\"/>\n<text text-anchor=\"start\" x=\"72\" y=\"-78.91\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n</g>\n<!-- (&#39;store1.2&#39;,) -->\n<g id=\"node2\" class=\"node\">\n<title>(&#39;store1.2&#39;,)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"138.02\" cy=\"-26.91\" rx=\"26.91\" ry=\"26.91\"/>\n<text text-anchor=\"start\" x=\"118.86\" y=\"-23.31\" font-family=\"Times,serif\" font-size=\"12.00\">store1.2</text>\n</g>\n<!-- (&#39;store1.2&#39;,)&#45;&gt;(&#39;process1&#39;,) -->\n<g id=\"edge3\" class=\"edge\">\n<title>(&#39;store1.2&#39;,)&#45;&gt;(&#39;process1&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M100.28,-36.89C90.61,-40.45 80.51,-45.07 72,-50.91 59.8,-59.28 48.8,-71.78 40.85,-82.19\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"101.12,-40.3 109.48,-33.79 98.88,-33.67 101.12,-40.3\"/>\n<text text-anchor=\"start\" x=\"72\" y=\"-52.91\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n</g>\n<!-- (&#39;store1.2&#39;,)&#45;&gt;(&#39;process2&#39;,) -->\n<g id=\"edge1\" class=\"edge\">\n<title>(&#39;store1.2&#39;,)&#45;&gt;(&#39;process2&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M111.91,-17.52C99.8,-14.2 85.1,-11.94 72,-14.91 69.83,-15.4 67.65,-16.01 65.47,-16.71\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"64.26,-13.43 56.19,-20.29 66.78,-19.96 64.26,-13.43\"/>\n<text text-anchor=\"start\" x=\"72\" y=\"-16.91\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n</g>\n</g>\n</svg>\n",
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x12a2da040>"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 117
  },
  {
   "cell_type": "markdown",
   "id": "398509c7-cf5c-4cce-a179-3a39752b0dde",
   "metadata": {},
   "source": [
    "### Nested composite\n",
    "A nested composite has store hierarchies with multiple connected processes."
   ]
  },
  {
   "cell_type": "code",
   "id": "701e759d-7f9b-41bb-911c-a8208c892462",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:17.025660Z",
     "start_time": "2025-04-06T16:43:16.977251Z"
    }
   },
   "source": [
    "nested_composite_spec = {\n",
    "    'store1': {\n",
    "        'store1.1': 'float',\n",
    "        'store1.2': 'integer',\n",
    "        'process1': {\n",
    "            '_type': 'process',\n",
    "            '_inputs': {\n",
    "                'port1': 'float',\n",
    "                'port2': 'integer',\n",
    "            },\n",
    "            'inputs': {\n",
    "                'port1': ['store1.1'],\n",
    "                'port2': ['store1.2'],\n",
    "            }\n",
    "        },\n",
    "        'process2': {\n",
    "            '_type': 'process',\n",
    "            '_outputs': {\n",
    "                'port1': 'float',\n",
    "                'port2': 'integer',\n",
    "            },\n",
    "            'outputs': {\n",
    "                'port1': ['store1.1'],\n",
    "                'port2': ['store1.2'],\n",
    "            }\n",
    "        },\n",
    "    },\n",
    "    'process3': {\n",
    "        '_type': 'process',\n",
    "        '_inputs': {\n",
    "            'port1': 'tree',\n",
    "        },\n",
    "        'inputs': {\n",
    "            'port1': ['store1'],\n",
    "        }\n",
    "    }\n",
    "}\n",
    "plot_bigraph(nested_composite_spec, \n",
    "             **plot_settings,\n",
    "             filename='nested_composite')"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/nested_composite\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n<!-- Generated by graphviz version 9.0.0 (0)\n -->\n<!-- Title: bigraph Pages: 1 -->\n<svg width=\"277pt\" height=\"227pt\"\n viewBox=\"0.00 0.00 285.00 233.00\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 229)\">\n<title>bigraph</title>\n<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-229 281,-229 281,4 -4,4\"/>\n<!-- (&#39;store1&#39;,) -->\n<g id=\"node1\" class=\"node\">\n<title>(&#39;store1&#39;,)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"140\" cy=\"-203.41\" rx=\"21.59\" ry=\"21.59\"/>\n<text text-anchor=\"start\" x=\"125.34\" y=\"-199.81\" font-family=\"Times,serif\" font-size=\"12.00\">store1</text>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.1&#39;) -->\n<g id=\"node2\" class=\"node\">\n<title>(&#39;store1&#39;, &#39;store1.1&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"140\" cy=\"-108.91\" rx=\"26.91\" ry=\"26.91\"/>\n<text text-anchor=\"start\" x=\"120.84\" y=\"-105.31\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1</text>\n</g>\n<!-- (&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;) -->\n<g id=\"edge3\" class=\"edge\">\n<title>(&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M140,-180.96C140,-167.72 140,-150.64 140,-136.54\"/>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.2&#39;) -->\n<g id=\"node3\" class=\"node\">\n<title>(&#39;store1&#39;, &#39;store1.2&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"68\" cy=\"-108.91\" rx=\"26.91\" ry=\"26.91\"/>\n<text text-anchor=\"start\" x=\"48.84\" y=\"-105.31\" font-family=\"Times,serif\" font-size=\"12.00\">store1.2</text>\n</g>\n<!-- (&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.2&#39;) -->\n<g id=\"edge4\" class=\"edge\">\n<title>(&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.2&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M126.79,-185.44C114.86,-170.11 97.24,-147.48 84.41,-130.99\"/>\n</g>\n<!-- (&#39;store1&#39;, &#39;process1&#39;) -->\n<g id=\"node4\" class=\"node\">\n<title>(&#39;store1&#39;, &#39;process1&#39;)</title>\n<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"54,-36 0,-36 0,0 54,0 54,-36\"/>\n<text text-anchor=\"start\" x=\"6.01\" y=\"-14.4\" font-family=\"Times,serif\" font-size=\"12.00\">process1</text>\n</g>\n<!-- (&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;process1&#39;) -->\n<!-- (&#39;store1&#39;, &#39;process2&#39;) -->\n<g id=\"node5\" class=\"node\">\n<title>(&#39;store1&#39;, &#39;process2&#39;)</title>\n<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"186,-36 132,-36 132,0 186,0 186,-36\"/>\n<text text-anchor=\"start\" x=\"138.01\" y=\"-14.4\" font-family=\"Times,serif\" font-size=\"12.00\">process2</text>\n</g>\n<!-- (&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;process2&#39;) -->\n<!-- (&#39;process3&#39;,) -->\n<g id=\"node6\" class=\"node\">\n<title>(&#39;process3&#39;,)</title>\n<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"277,-126.91 223,-126.91 223,-90.91 277,-90.91 277,-126.91\"/>\n<text text-anchor=\"start\" x=\"229.01\" y=\"-105.31\" font-family=\"Times,serif\" font-size=\"12.00\">process3</text>\n</g>\n<!-- (&#39;store1&#39;,)&#45;&gt;(&#39;process3&#39;,) -->\n<g id=\"edge7\" class=\"edge\">\n<title>(&#39;store1&#39;,)&#45;&gt;(&#39;process3&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M160.11,-193.87C174.67,-187.08 194.29,-176.6 209,-163.82 217.89,-156.09 226.11,-146.16 232.81,-136.92\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"235.53,-139.14 238.35,-128.92 229.78,-135.15 235.53,-139.14\"/>\n<text text-anchor=\"start\" x=\"218\" y=\"-155.82\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process1&#39;) -->\n<g id=\"edge6\" class=\"edge\">\n<title>(&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M118.75,-91.19C101.82,-77.87 77.88,-59.03 58.69,-43.94\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"60.92,-41.24 50.9,-37.8 56.59,-46.74 60.92,-41.24\"/>\n<text text-anchor=\"start\" x=\"84\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process2&#39;) -->\n<g id=\"edge9\" class=\"edge\">\n<title>(&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process2&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M147.93,-70.78C150.44,-59.04 153.1,-46.6 155.19,-36.85\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"144.58,-69.72 145.91,-80.23 151.43,-71.19 144.58,-69.72\"/>\n<text text-anchor=\"start\" x=\"152\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.2&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process1&#39;) -->\n<g id=\"edge5\" class=\"edge\">\n<title>(&#39;store1&#39;, &#39;store1.2&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M47.01,-90.82C39.53,-83.44 31.95,-74.16 27.89,-64 25.91,-59.05 24.93,-53.54 24.54,-48.13\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"28.04,-48.17 24.52,-38.18 21.04,-48.19 28.04,-48.17\"/>\n<text text-anchor=\"start\" x=\"28\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n</g>\n<!-- (&#39;store1&#39;, &#39;store1.2&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process2&#39;) -->\n<g id=\"edge8\" class=\"edge\">\n<title>(&#39;store1&#39;, &#39;store1.2&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process2&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M95.51,-81.03C110.13,-66.74 127.71,-49.57 140.66,-36.92\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"93.09,-78.5 88.38,-88 97.98,-83.51 93.09,-78.5\"/>\n<text text-anchor=\"start\" x=\"122\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n</g>\n</g>\n</svg>\n",
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x11abaf730>"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 118
  },
  {
   "cell_type": "markdown",
   "id": "0c8c1704-2331-44eb-bd05-72e36ccb6858",
   "metadata": {},
   "source": [
    "### Composite process\n",
    "Composite processes are processes with internal stores and internal processes, which can run as their own simulation. The composite process has schema, allowing it to connect to a super-simulation. This allows for improved distributed computation, with composite processes able to run on their own computer and syncronize to the super-simulation via message passing."
   ]
  },
  {
   "cell_type": "code",
   "id": "638c5dc7-4cfd-46e4-b66a-8c42ca98ff1d",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:17.081490Z",
     "start_time": "2025-04-06T16:43:17.034507Z"
    }
   },
   "source": [
    "composite_process_spec = {\n",
    "    'composite_process': {\n",
    "        '_type': 'composite',\n",
    "        'store1.1': 'any',\n",
    "        'store1.2': 'any',\n",
    "        'process1': {\n",
    "            '_type': 'process',\n",
    "            '_outputs': {\n",
    "                'port1': 'float',\n",
    "                'port2': 'integer',\n",
    "            },\n",
    "            'outputs': {\n",
    "                'port1': ['store1.1'],\n",
    "                'port2': ['store1.2'],\n",
    "            }\n",
    "        },\n",
    "        'process2': {\n",
    "            '_type': 'process',\n",
    "            '_outputs': {\n",
    "                'port1': 'float',\n",
    "                'port2': 'integer',\n",
    "            },\n",
    "            'outputs': {\n",
    "                'port1': ['store1.1'],\n",
    "                'port2': ['store1.2'],\n",
    "            }\n",
    "        },\n",
    "        '_inputs': {\n",
    "            'port1': 'any',\n",
    "            'port2': 'any',\n",
    "        },\n",
    "        'bridge': {\n",
    "            'inputs': {\n",
    "                'port1': ['store1.1'],\n",
    "                'port2': ['store1.2'],\n",
    "            }\n",
    "        }\n",
    "    }\n",
    "}\n",
    "\n",
    "plot_bigraph(\n",
    "    composite_process_spec, \n",
    "    **plot_settings, filename='composite_process')"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/composite_process\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n<!-- Generated by graphviz version 9.0.0 (0)\n -->\n<!-- Title: bigraph Pages: 1 -->\n<svg width=\"202pt\" height=\"291pt\"\n viewBox=\"0.00 0.00 207.28 298.97\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 294.97)\">\n<title>bigraph</title>\n<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-294.97 203.28,-294.97 203.28,4 -4,4\"/>\n<!-- (&#39;composite_process&#39;, &#39;store1.1&#39;) -->\n<g id=\"node1\" class=\"node\">\n<title>(&#39;composite_process&#39;, &#39;store1.1&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"78\" cy=\"-108.91\" rx=\"26.91\" ry=\"26.91\"/>\n<text text-anchor=\"start\" x=\"58.84\" y=\"-105.31\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1</text>\n</g>\n<!-- (&#39;composite_process&#39;,) -->\n<g id=\"node3\" class=\"node\">\n<title>(&#39;composite_process&#39;,)</title>\n<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"144.26,-217.82 45.74,-217.82 45.74,-181.82 144.26,-181.82 144.26,-217.82\"/>\n<text text-anchor=\"start\" x=\"49.34\" y=\"-196.22\" font-family=\"Times,serif\" font-size=\"12.00\">composite_process</text>\n</g>\n<!-- (&#39;composite_process&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;composite_process&#39;,) -->\n<g id=\"edge6\" class=\"edge\">\n<title>(&#39;composite_process&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;composite_process&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M44.65,-129.11C32.25,-138.82 22.83,-151.21 30.34,-163.82 34.46,-170.73 40.47,-176.36 47.18,-180.91\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"46.59,-132.02 52.76,-123.41 42.56,-126.29 46.59,-132.02\"/>\n<text text-anchor=\"start\" x=\"31\" y=\"-155.82\" font-family=\"Times,serif\" font-size=\"10.00\">bridge_port1</text>\n</g>\n<!-- (&#39;composite_process&#39;, &#39;process1&#39;) -->\n<g id=\"node4\" class=\"node\">\n<title>(&#39;composite_process&#39;, &#39;process1&#39;)</title>\n<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"186,-36 132,-36 132,0 186,0 186,-36\"/>\n<text text-anchor=\"start\" x=\"138.01\" y=\"-14.4\" font-family=\"Times,serif\" font-size=\"12.00\">process1</text>\n</g>\n<!-- (&#39;composite_process&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;) -->\n<g id=\"edge8\" class=\"edge\">\n<title>(&#39;composite_process&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M103.6,-79.81C116.4,-65.76 131.52,-49.16 142.72,-36.87\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"101.14,-77.31 97,-87.06 106.32,-82.02 101.14,-77.31\"/>\n<text text-anchor=\"start\" x=\"125\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n</g>\n<!-- (&#39;composite_process&#39;, &#39;process2&#39;) -->\n<g id=\"node5\" class=\"node\">\n<title>(&#39;composite_process&#39;, &#39;process2&#39;)</title>\n<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"54,-36 0,-36 0,0 54,0 54,-36\"/>\n<text text-anchor=\"start\" x=\"6.01\" y=\"-14.4\" font-family=\"Times,serif\" font-size=\"12.00\">process2</text>\n</g>\n<!-- (&#39;composite_process&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;) -->\n<g id=\"edge10\" class=\"edge\">\n<title>(&#39;composite_process&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M46.11,-87.03C38.81,-80.58 32,-72.82 27.89,-64 24,-55.64 23.51,-45.43 24.11,-36.84\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"43.6,-89.51 53.59,-93.04 47.99,-84.05 43.6,-89.51\"/>\n<text text-anchor=\"start\" x=\"28\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n</g>\n<!-- (&#39;composite_process&#39;, &#39;store1.2&#39;) -->\n<g id=\"node2\" class=\"node\">\n<title>(&#39;composite_process&#39;, &#39;store1.2&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"150\" cy=\"-108.91\" rx=\"26.91\" ry=\"26.91\"/>\n<text text-anchor=\"start\" x=\"130.84\" y=\"-105.31\" font-family=\"Times,serif\" font-size=\"12.00\">store1.2</text>\n</g>\n<!-- (&#39;composite_process&#39;, &#39;store1.2&#39;)&#45;&gt;(&#39;composite_process&#39;,) -->\n<g id=\"edge5\" class=\"edge\">\n<title>(&#39;composite_process&#39;, &#39;store1.2&#39;)&#45;&gt;(&#39;composite_process&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M122.23,-136.24C117.47,-141.71 112.92,-147.69 109.34,-153.82 104.47,-162.18 101.04,-172.39 98.76,-180.98\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"124.52,-138.92 128.75,-129.21 119.38,-134.16 124.52,-138.92\"/>\n<text text-anchor=\"start\" x=\"109\" y=\"-155.82\" font-family=\"Times,serif\" font-size=\"10.00\">bridge_port2</text>\n</g>\n<!-- (&#39;composite_process&#39;, &#39;store1.2&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;) -->\n<g id=\"edge7\" class=\"edge\">\n<title>(&#39;composite_process&#39;, &#39;store1.2&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M153.83,-70.06C155,-58.56 156.22,-46.46 157.18,-36.93\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"150.38,-69.43 152.85,-79.73 157.34,-70.13 150.38,-69.43\"/>\n<text text-anchor=\"start\" x=\"155\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n</g>\n<!-- (&#39;composite_process&#39;, &#39;store1.2&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;) -->\n<g id=\"edge9\" class=\"edge\">\n<title>(&#39;composite_process&#39;, &#39;store1.2&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M118.94,-85.46C98.13,-70.42 71.11,-50.89 51.76,-36.9\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"116.63,-88.11 126.78,-91.13 120.73,-82.43 116.63,-88.11\"/>\n<text text-anchor=\"start\" x=\"88\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n</g>\n<!-- (&#39;composite_process&#39;,)&#45;&gt;(&#39;composite_process&#39;, &#39;store1.1&#39;) -->\n<g id=\"edge3\" class=\"edge\">\n<title>(&#39;composite_process&#39;,)&#45;&gt;(&#39;composite_process&#39;, &#39;store1.1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M91.56,-180.83C89.11,-168.04 85.79,-150.66 83.05,-136.31\"/>\n</g>\n<!-- (&#39;composite_process&#39;,)&#45;&gt;(&#39;composite_process&#39;, &#39;store1.2&#39;) -->\n<g id=\"edge4\" class=\"edge\">\n<title>(&#39;composite_process&#39;,)&#45;&gt;(&#39;composite_process&#39;, &#39;store1.2&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M144.95,-182.63C152.93,-177.83 160.14,-171.68 165,-163.82 170.77,-154.48 168.53,-142.88 164.21,-132.84\"/>\n</g>\n<!-- (&#39;composite_process&#39;,)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;) -->\n<!-- (&#39;composite_process&#39;,)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;) -->\n<!-- (&#39;composite_process&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)_input -->\n<!-- (&#39;composite_process&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)_input&#45;&gt;(&#39;composite_process&#39;,) -->\n<g id=\"edge11\" class=\"edge\">\n<title>(&#39;composite_process&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)_input&#45;&gt;(&#39;composite_process&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M72.06,-262.98C71.87,-254.95 72.32,-244.62 74.89,-235.82 75.59,-233.41 76.49,-231 77.52,-228.62\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"80.51,-230.46 81.93,-219.96 74.27,-227.28 80.51,-230.46\"/>\n<text text-anchor=\"start\" x=\"74\" y=\"-237.82\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n</g>\n<!-- (&#39;composite_process&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)_input -->\n<!-- (&#39;composite_process&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)_input&#45;&gt;(&#39;composite_process&#39;,) -->\n<g id=\"edge12\" class=\"edge\">\n<title>(&#39;composite_process&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)_input&#45;&gt;(&#39;composite_process&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M114.1,-263.58C111.26,-254.26 107.31,-241.27 103.74,-229.53\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"107.18,-228.81 100.92,-220.26 100.48,-230.84 107.18,-228.81\"/>\n<text text-anchor=\"start\" x=\"108\" y=\"-237.82\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n</g>\n</g>\n</svg>\n",
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x12a393550>"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 119
  },
  {
   "cell_type": "markdown",
   "id": "847cef26-65a1-4da6-9dd3-eb5839277970",
   "metadata": {},
   "source": [
    "## Orchestration"
   ]
  },
  {
   "cell_type": "code",
   "id": "e76c97a8-3037-464a-929e-ba4414c73c34",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:17.090435Z",
     "start_time": "2025-04-06T16:43:17.088457Z"
    }
   },
   "source": [
    "plot_settings2={}\n",
    "if save_images:\n",
    "    plot_settings2.update({\n",
    "        'out_dir': 'out',\n",
    "        'dpi': '250'\n",
    "    })"
   ],
   "outputs": [],
   "execution_count": 120
  },
  {
   "cell_type": "markdown",
   "id": "48975890-06a4-462a-9b1d-6bddd51779fe",
   "metadata": {},
   "source": [
    "### Multi-timestepping\n",
    "\n",
    "Temporal processes can be run by multi-timestepping, with each process updating the state according to its `sync_state` state, which it can update adaptively."
   ]
  },
  {
   "cell_type": "code",
   "id": "f0eff7b0-f0e2-4d56-8b16-ef0fc1ce5aba",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:17.097072Z",
     "start_time": "2025-04-06T16:43:17.095590Z"
    }
   },
   "source": [
    "# multitimestep_spec = {\n",
    "#     'temporal process2': {\n",
    "#         '_ports': {'port1': 'Any'},\n",
    "#         'wires': {'port1': 'state'},\n",
    "#         '_sync_step': 1.0,\n",
    "#     },\n",
    "#     'temporal process1': {\n",
    "#         '_ports': {'port1': 'Any'},\n",
    "#         'wires': {'port1': 'state'},\n",
    "#         '_sync_step': 0.5,\n",
    "#     },\n",
    "# }\n",
    "# multitimestep_spec = replace_regex_recursive(multitimestep_spec)\n",
    "# plot_multitimestep(multitimestep_spec, total_time=2.0, **plot_settings2, filename='multitimestep')"
   ],
   "outputs": [],
   "execution_count": 121
  },
  {
   "cell_type": "markdown",
   "id": "a00cf677-ebac-4309-8ac9-a5b2855724af",
   "metadata": {},
   "source": [
    "### Flows\n",
    "\n",
    "A directed acyclic graph (DAG) – a type of workflow that declares the dependencies between processes. A flow run is not temporal, and the order is entirely based on dependencies. It can triggered between time steps, running the steps in the order as determined by their dependencies."
   ]
  },
  {
   "cell_type": "code",
   "id": "db45c99b-b97b-47bc-a73e-c669882f3969",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:17.103619Z",
     "start_time": "2025-04-06T16:43:17.102124Z"
    }
   },
   "source": [
    "# flow = {\n",
    "#     'step1': {\n",
    "#         '_type': 'step_process',\n",
    "#         '_ports': {},\n",
    "#         '_depends_on': [], \n",
    "#     },\n",
    "#     'step2': {\n",
    "#         '_type': 'step_process',\n",
    "#         '_ports': {},\n",
    "#         '_depends_on': 'step1', \n",
    "#     },\n",
    "#     'step3': {\n",
    "#         '_type': 'step_process',\n",
    "#         '_ports': {},\n",
    "#         '_depends_on': [], \n",
    "#     },\n",
    "#     'step4': {\n",
    "#         '_type': 'step_process',\n",
    "#         '_ports': {},\n",
    "#         '_depends_on': ['step2', 'step3'], \n",
    "#     },\n",
    "# }\n",
    "\n",
    "# plot_flow(flow, **plot_settings2, filename='flow')"
   ],
   "outputs": [],
   "execution_count": 122
  },
  {
   "cell_type": "markdown",
   "id": "5882581a-c5ef-40b5-acbd-efddad269807",
   "metadata": {},
   "source": [
    "## Examples"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff81178f-36e4-4200-982a-ed8155b87934",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Multiscale map of a cell\n",
    "\n",
    "Bigraphs can represent the hierarchical nesting of cellular structures. Below is a multiscale map of a cell from [Qin et al. A multi-scale map of cell structure fusing protein images and interactions. Nature. 2021.](https://pubmed.ncbi.nlm.nih.gov/34819669/)"
   ]
  },
  {
   "cell_type": "code",
   "id": "47bdf8e5-0bdd-4cac-b57e-2449c75fa178",
   "metadata": {
    "tags": [],
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:17.206147Z",
     "start_time": "2025-04-06T16:43:17.108638Z"
    }
   },
   "source": [
    "music_map = {\n",
    "    'cell': {\n",
    "        'cytoplasm': {\n",
    "            'secretory organelles': {\n",
    "                'transmembrane transport systems': {\n",
    "                    'ion transmembrane transport systems': {}}},\n",
    "            'cytoplasmic organelles': {\n",
    "                'subgroup of cytoplasmic organelles': {\n",
    "                    'metabolic organelles': {\n",
    "                        'subgroup of metabolic organelles': {},\n",
    "                        'mitochondrion': {}},\n",
    "                    'ribonucleoprotein complex family': {\n",
    "                        'RNA processing complex 1': {'RNA splicing complex 1': {}},\n",
    "                        'ribosome biogenesis community': {\n",
    "                            'ribosome': {\n",
    "                                'ribosomal subunit': {\n",
    "                                    'mitochondirial large ribosomal subunit': {\n",
    "                                        # 'large ribosomal subunit subcomplex 1': {}\n",
    "                                    },\n",
    "                                    'mito-cyto ribosomal cluster': {},\n",
    "                                    'cotranslational protein targeting to membrane system': {},\n",
    "                                    'ribosomal complex 5': {},\n",
    "                                    'ribosomal complex 2': {}}},\n",
    "                            'nucleic acid binding complex': {}}}}},\n",
    "            'transcriptional regulation complex family': {\n",
    "                'transcriptional regulation complexes': {}}},\n",
    "        'nucleus': {\n",
    "            'DNA metabolic assembly': {},\n",
    "            'nuclear lumen': {\n",
    "                'nucleolus': {\n",
    "                    'ribonucleprotein complex family': {},\n",
    "                    'RNA processing complex family': {\n",
    "                        'RNA processing complex 1': {},\n",
    "                        'RNA splicing complex family': {}},\n",
    "                    'splicosomal complex family': {}},\n",
    "                'nuclear splicing speckle': {}},\n",
    "            'nuclear body': {},\n",
    "            'nucleo-plasm 1': {\n",
    "                'chromosome organization complex family': {\n",
    "                    'chromatin organization complex family': {\n",
    "                        'HAT complex family': {\n",
    "                            'NuA4 HAT complex': {}}}}},\n",
    "            'nucleo-plasm 2': {\n",
    "                'chromatin regulation complex': {}},\n",
    "            'nuclear transcriptional speckle': {\n",
    "                'negative regulation of RNA biosynthesis process': {}}}}}\n",
    "music_map1 = replace_regex_recursive(music_map)\n",
    "plot_bigraph(music_map1, **plot_settings, filename='cell_hierarchy')"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/cell_hierarchy\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n<!-- Generated by graphviz version 9.0.0 (0)\n -->\n<!-- Title: bigraph Pages: 1 -->\n<svg width=\"705pt\" height=\"700pt\"\n viewBox=\"0.00 0.00 724.95 720.00\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<g id=\"graph0\" class=\"graph\" transform=\"scale(0.629041 0.629041) rotate(0) translate(4 1108.81)\">\n<title>bigraph</title>\n<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-1108.81 1116.45,-1108.81 1116.45,4 -4,4\"/>\n<!-- (&#39;cell&#39;,) -->\n<g id=\"node1\" class=\"node\">\n<title>(&#39;cell&#39;,)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"482.39\" cy=\"-1086.81\" rx=\"18\" ry=\"18\"/>\n<text text-anchor=\"start\" x=\"473.73\" y=\"-1083.21\" font-family=\"Times,serif\" font-size=\"12.00\">cell</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;) -->\n<g id=\"node2\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"325.39\" cy=\"-999.39\" rx=\"33.41\" ry=\"33.41\"/>\n<text text-anchor=\"start\" x=\"300.73\" y=\"-995.79\" font-family=\"Times,serif\" font-size=\"12.00\">cytoplasm</text>\n</g>\n<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;) -->\n<g id=\"edge1\" class=\"edge\">\n<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M466.18,-1076.99C440.02,-1062.75 388.08,-1034.49 355.12,-1016.57\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;) -->\n<g id=\"node25\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"737.39\" cy=\"-999.39\" rx=\"25.92\" ry=\"25.92\"/>\n<text text-anchor=\"start\" x=\"719.07\" y=\"-995.79\" font-family=\"Times,serif\" font-size=\"12.00\">nucleus</text>\n</g>\n<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;) -->\n<g id=\"edge24\" class=\"edge\">\n<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M499.92,-1079.93C544.26,-1065.09 660.98,-1025.99 712.43,-1008.76\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;) -->\n<g id=\"node3\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"154.39\" cy=\"-875.88\" rx=\"39.81\" ry=\"39.81\"/>\n<text text-anchor=\"start\" x=\"132.41\" y=\"-878.28\" font-family=\"Times,serif\" font-size=\"12.00\">secretory</text>\n<text text-anchor=\"start\" x=\"129.84\" y=\"-866.28\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;) -->\n<g id=\"edge2\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M295.74,-982.85C272.07,-969.92 238.5,-950.39 211.39,-929.98 201.45,-922.5 191.3,-913.54 182.36,-905.13\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n<g id=\"node6\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"266.39\" cy=\"-875.88\" rx=\"46.09\" ry=\"46.09\"/>\n<text text-anchor=\"start\" x=\"237.4\" y=\"-878.28\" font-family=\"Times,serif\" font-size=\"12.00\">cytoplasmic</text>\n<text text-anchor=\"start\" x=\"241.84\" y=\"-866.28\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n<g id=\"edge5\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M310.81,-968.36C303.44,-953.18 294.38,-934.53 286.37,-918.03\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node23\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"384.39\" cy=\"-875.88\" rx=\"54.1\" ry=\"54.1\"/>\n<text text-anchor=\"start\" x=\"349.74\" y=\"-890.28\" font-family=\"Times,serif\" font-size=\"12.00\">transcriptional</text>\n<text text-anchor=\"start\" x=\"360.07\" y=\"-878.28\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n<text text-anchor=\"start\" x=\"363.73\" y=\"-866.28\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"368.73\" y=\"-854.28\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge22\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M339.98,-968.36C346.26,-955.43 353.76,-939.98 360.8,-925.49\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n<g id=\"node4\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"87.39\" cy=\"-728.39\" rx=\"57.39\" ry=\"57.39\"/>\n<text text-anchor=\"start\" x=\"50.41\" y=\"-736.79\" font-family=\"Times,serif\" font-size=\"12.00\">transmembrane</text>\n<text text-anchor=\"start\" x=\"66.07\" y=\"-724.79\" font-family=\"Times,serif\" font-size=\"12.00\">transport</text>\n<text text-anchor=\"start\" x=\"68.39\" y=\"-712.79\" font-family=\"Times,serif\" font-size=\"12.00\">systems</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n<g id=\"edge3\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M137.83,-838.92C129.85,-821.58 120.13,-800.47 111.35,-781.42\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n<g id=\"node5\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"57.39\" cy=\"-569.58\" rx=\"57.39\" ry=\"57.39\"/>\n<text text-anchor=\"start\" x=\"49.73\" y=\"-583.98\" font-family=\"Times,serif\" font-size=\"12.00\">ion</text>\n<text text-anchor=\"start\" x=\"20.41\" y=\"-571.98\" font-family=\"Times,serif\" font-size=\"12.00\">transmembrane</text>\n<text text-anchor=\"start\" x=\"36.07\" y=\"-559.98\" font-family=\"Times,serif\" font-size=\"12.00\">transport</text>\n<text text-anchor=\"start\" x=\"38.39\" y=\"-547.98\" font-family=\"Times,serif\" font-size=\"12.00\">systems</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n<g id=\"edge4\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M76.65,-671.22C73.89,-656.81 70.93,-641.3 68.17,-626.88\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n<g id=\"node7\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"266.39\" cy=\"-728.39\" rx=\"46.09\" ry=\"46.09\"/>\n<text text-anchor=\"start\" x=\"244.06\" y=\"-742.79\" font-family=\"Times,serif\" font-size=\"12.00\">subgroup</text>\n<text text-anchor=\"start\" x=\"261.4\" y=\"-730.79\" font-family=\"Times,serif\" font-size=\"12.00\">of</text>\n<text text-anchor=\"start\" x=\"237.4\" y=\"-718.79\" font-family=\"Times,serif\" font-size=\"12.00\">cytoplasmic</text>\n<text text-anchor=\"start\" x=\"241.84\" y=\"-706.79\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n<g id=\"edge6\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M266.39,-828.85C266.39,-811.7 266.39,-792.25 266.39,-775.12\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;) -->\n<g id=\"node8\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"172.39\" cy=\"-569.58\" rx=\"39.81\" ry=\"39.81\"/>\n<text text-anchor=\"start\" x=\"148.74\" y=\"-571.98\" font-family=\"Times,serif\" font-size=\"12.00\">metabolic</text>\n<text text-anchor=\"start\" x=\"147.84\" y=\"-559.98\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;) -->\n<g id=\"edge7\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M242.67,-687.82C227.34,-662.23 207.57,-629.27 192.91,-604.81\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node11\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"295.39\" cy=\"-569.58\" rx=\"65.41\" ry=\"65.41\"/>\n<text text-anchor=\"start\" x=\"252.74\" y=\"-577.98\" font-family=\"Times,serif\" font-size=\"12.00\">ribonucleoprotein</text>\n<text text-anchor=\"start\" x=\"274.73\" y=\"-565.98\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"279.73\" y=\"-553.98\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge10\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M274.71,-682.45C277.45,-667.61 280.56,-650.8 283.52,-634.8\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;) -->\n<g id=\"node9\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"73.39\" cy=\"-422.15\" rx=\"39.81\" ry=\"39.81\"/>\n<text text-anchor=\"start\" x=\"51.06\" y=\"-436.55\" font-family=\"Times,serif\" font-size=\"12.00\">subgroup</text>\n<text text-anchor=\"start\" x=\"68.4\" y=\"-424.55\" font-family=\"Times,serif\" font-size=\"12.00\">of</text>\n<text text-anchor=\"start\" x=\"49.74\" y=\"-412.55\" font-family=\"Times,serif\" font-size=\"12.00\">metabolic</text>\n<text text-anchor=\"start\" x=\"48.84\" y=\"-400.55\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;) -->\n<g id=\"edge8\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M149.93,-535.58C133.73,-511.78 111.95,-479.79 95.77,-456.02\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;) -->\n<g id=\"node10\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"177.39\" cy=\"-422.15\" rx=\"46.02\" ry=\"46.02\"/>\n<text text-anchor=\"start\" x=\"142.07\" y=\"-418.55\" font-family=\"Times,serif\" font-size=\"12.00\">mitochondrion</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;) -->\n<g id=\"edge9\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M173.76,-528.84C174.4,-510.34 175.16,-488.22 175.82,-469.01\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n<g id=\"node12\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"291.39\" cy=\"-422.15\" rx=\"41.38\" ry=\"41.38\"/>\n<text text-anchor=\"start\" x=\"278.73\" y=\"-436.55\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n<text text-anchor=\"start\" x=\"265.73\" y=\"-424.55\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n<text text-anchor=\"start\" x=\"270.73\" y=\"-412.55\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"288.39\" y=\"-400.55\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n<g id=\"edge11\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M293.61,-503.52C293.24,-490.21 292.86,-476.55 292.53,-464.36\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;) -->\n<g id=\"node14\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"394.39\" cy=\"-422.15\" rx=\"43.74\" ry=\"43.74\"/>\n<text text-anchor=\"start\" x=\"372.06\" y=\"-430.55\" font-family=\"Times,serif\" font-size=\"12.00\">ribosome</text>\n<text text-anchor=\"start\" x=\"369.06\" y=\"-418.55\" font-family=\"Times,serif\" font-size=\"12.00\">biogenesis</text>\n<text text-anchor=\"start\" x=\"367.06\" y=\"-406.55\" font-family=\"Times,serif\" font-size=\"12.00\">community</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;) -->\n<g id=\"edge13\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M332.26,-514.43C344.79,-496.01 358.49,-475.9 369.82,-459.25\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n<g id=\"node13\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"291.39\" cy=\"-301.09\" rx=\"39.03\" ry=\"39.03\"/>\n<text text-anchor=\"start\" x=\"278.73\" y=\"-315.49\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n<text text-anchor=\"start\" x=\"272.4\" y=\"-303.49\" font-family=\"Times,serif\" font-size=\"12.00\">splicing</text>\n<text text-anchor=\"start\" x=\"270.73\" y=\"-291.49\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"288.39\" y=\"-279.49\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n<g id=\"edge12\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M291.39,-379.84C291.39,-367.22 291.39,-353.38 291.39,-340.92\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;) -->\n<g id=\"node15\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"387.39\" cy=\"-301.09\" rx=\"30.65\" ry=\"30.65\"/>\n<text text-anchor=\"start\" x=\"365.06\" y=\"-297.49\" font-family=\"Times,serif\" font-size=\"12.00\">ribosome</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;) -->\n<g id=\"edge14\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M391.83,-377.49C390.95,-362.49 389.98,-346.05 389.17,-332.35\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;) -->\n<g id=\"node22\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"475.39\" cy=\"-301.09\" rx=\"39.03\" ry=\"39.03\"/>\n<text text-anchor=\"start\" x=\"458.07\" y=\"-315.49\" font-family=\"Times,serif\" font-size=\"12.00\">nucleic</text>\n<text text-anchor=\"start\" x=\"465.4\" y=\"-303.49\" font-family=\"Times,serif\" font-size=\"12.00\">acid</text>\n<text text-anchor=\"start\" x=\"457.06\" y=\"-291.49\" font-family=\"Times,serif\" font-size=\"12.00\">binding</text>\n<text text-anchor=\"start\" x=\"454.73\" y=\"-279.49\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;) -->\n<g id=\"edge21\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M418.92,-385.1C429.91,-368.95 442.76,-350.05 453.43,-334.37\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;) -->\n<g id=\"node16\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"387.39\" cy=\"-187.03\" rx=\"39.03\" ry=\"39.03\"/>\n<text text-anchor=\"start\" x=\"363.4\" y=\"-189.43\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n<text text-anchor=\"start\" x=\"369.73\" y=\"-177.43\" font-family=\"Times,serif\" font-size=\"12.00\">subunit</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;) -->\n<g id=\"edge15\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M387.39,-269.47C387.39,-256.38 387.39,-240.95 387.39,-227.05\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;) -->\n<g id=\"node17\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"162.39\" cy=\"-56\" rx=\"55.05\" ry=\"55.05\"/>\n<text text-anchor=\"start\" x=\"127.07\" y=\"-70.4\" font-family=\"Times,serif\" font-size=\"12.00\">mitochondirial</text>\n<text text-anchor=\"start\" x=\"150.51\" y=\"-58.4\" font-family=\"Times,serif\" font-size=\"12.00\">large</text>\n<text text-anchor=\"start\" x=\"138.4\" y=\"-46.4\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n<text text-anchor=\"start\" x=\"144.73\" y=\"-34.4\" font-family=\"Times,serif\" font-size=\"12.00\">subunit</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;) -->\n<g id=\"edge16\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M350.4,-172.46C316.76,-159.32 266.4,-137.67 226.39,-112.01 218.38,-106.86 210.31,-100.79 202.73,-94.58\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;) -->\n<g id=\"node18\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"274.39\" cy=\"-56\" rx=\"39.03\" ry=\"39.03\"/>\n<text text-anchor=\"start\" x=\"251.07\" y=\"-64.4\" font-family=\"Times,serif\" font-size=\"12.00\">mito&#45;cyto</text>\n<text text-anchor=\"start\" x=\"250.4\" y=\"-52.4\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n<text text-anchor=\"start\" x=\"258.4\" y=\"-40.4\" font-family=\"Times,serif\" font-size=\"12.00\">cluster</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;) -->\n<g id=\"edge17\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M361.75,-156.87C349.78,-143.25 335.35,-126.81 322.39,-112.01 315.13,-103.71 307.26,-94.7 300.04,-86.42\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;) -->\n<g id=\"node19\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"387.39\" cy=\"-56\" rx=\"56\" ry=\"56\"/>\n<text text-anchor=\"start\" x=\"352.07\" y=\"-82.4\" font-family=\"Times,serif\" font-size=\"12.00\">cotranslational</text>\n<text text-anchor=\"start\" x=\"370.4\" y=\"-70.4\" font-family=\"Times,serif\" font-size=\"12.00\">protein</text>\n<text text-anchor=\"start\" x=\"366.18\" y=\"-58.4\" font-family=\"Times,serif\" font-size=\"12.00\">targeting</text>\n<text text-anchor=\"start\" x=\"382.73\" y=\"-46.4\" font-family=\"Times,serif\" font-size=\"12.00\">to</text>\n<text text-anchor=\"start\" x=\"362.07\" y=\"-34.4\" font-family=\"Times,serif\" font-size=\"12.00\">membrane</text>\n<text text-anchor=\"start\" x=\"370.73\" y=\"-22.4\" font-family=\"Times,serif\" font-size=\"12.00\">system</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;) -->\n<g id=\"edge18\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M387.39,-147.32C387.39,-136.48 387.39,-124.48 387.39,-112.86\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;) -->\n<g id=\"node20\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"500.39\" cy=\"-56\" rx=\"39.03\" ry=\"39.03\"/>\n<text text-anchor=\"start\" x=\"476.4\" y=\"-64.4\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n<text text-anchor=\"start\" x=\"479.73\" y=\"-52.4\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"497.39\" y=\"-40.4\" font-family=\"Times,serif\" font-size=\"12.00\">5</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;) -->\n<g id=\"edge19\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M413.04,-156.87C425.01,-143.25 439.44,-126.81 452.39,-112.01 459.66,-103.71 467.53,-94.7 474.75,-86.42\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;) -->\n<g id=\"node21\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"596.39\" cy=\"-56\" rx=\"39.03\" ry=\"39.03\"/>\n<text text-anchor=\"start\" x=\"572.4\" y=\"-64.4\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n<text text-anchor=\"start\" x=\"575.73\" y=\"-52.4\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"593.39\" y=\"-40.4\" font-family=\"Times,serif\" font-size=\"12.00\">2</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;) -->\n<g id=\"edge20\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M424.97,-173.89C459.24,-161.68 510.21,-140.56 548.39,-112.01 557.33,-105.33 565.79,-96.74 573.05,-88.35\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;) -->\n<g id=\"node24\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"384.39\" cy=\"-728.39\" rx=\"54.1\" ry=\"54.1\"/>\n<text text-anchor=\"start\" x=\"349.74\" y=\"-736.79\" font-family=\"Times,serif\" font-size=\"12.00\">transcriptional</text>\n<text text-anchor=\"start\" x=\"360.07\" y=\"-724.79\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n<text text-anchor=\"start\" x=\"358.74\" y=\"-712.79\" font-family=\"Times,serif\" font-size=\"12.00\">complexes</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;) -->\n<g id=\"edge23\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M384.39,-821.12C384.39,-808.81 384.39,-795.73 384.39,-783.42\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;) -->\n<g id=\"node26\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"551.39\" cy=\"-875.88\" rx=\"38.55\" ry=\"38.55\"/>\n<text text-anchor=\"start\" x=\"538.4\" y=\"-884.28\" font-family=\"Times,serif\" font-size=\"12.00\">DNA</text>\n<text text-anchor=\"start\" x=\"527.74\" y=\"-872.28\" font-family=\"Times,serif\" font-size=\"12.00\">metabolic</text>\n<text text-anchor=\"start\" x=\"529.06\" y=\"-860.28\" font-family=\"Times,serif\" font-size=\"12.00\">assembly</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;) -->\n<g id=\"edge25\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M712.64,-989.41C683.74,-978.21 635.32,-957.18 599.39,-929.98 590.78,-923.46 582.47,-915.24 575.26,-907.22\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;) -->\n<g id=\"node27\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"638.39\" cy=\"-875.88\" rx=\"30.06\" ry=\"30.06\"/>\n<text text-anchor=\"start\" x=\"620.74\" y=\"-878.28\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n<text text-anchor=\"start\" x=\"623.4\" y=\"-866.28\" font-family=\"Times,serif\" font-size=\"12.00\">lumen</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;) -->\n<g id=\"edge26\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M719.87,-979.32C707.65,-965.8 691.15,-947.1 677.39,-929.98 670.09,-920.89 662.41,-910.61 655.82,-901.54\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;) -->\n<g id=\"node35\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"737.39\" cy=\"-875.88\" rx=\"50.8\" ry=\"50.8\"/>\n<text text-anchor=\"start\" x=\"705.07\" y=\"-878.28\" font-family=\"Times,serif\" font-size=\"12.00\">nucleo&#45;plasm</text>\n<text text-anchor=\"start\" x=\"734.39\" y=\"-866.28\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;) -->\n<g id=\"edge34\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M737.39,-972.63C737.39,-959.53 737.39,-943.14 737.39,-927.64\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;) -->\n<g id=\"node40\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"857.39\" cy=\"-875.88\" rx=\"50.8\" ry=\"50.8\"/>\n<text text-anchor=\"start\" x=\"825.07\" y=\"-878.28\" font-family=\"Times,serif\" font-size=\"12.00\">nucleo&#45;plasm</text>\n<text text-anchor=\"start\" x=\"854.39\" y=\"-866.28\" font-family=\"Times,serif\" font-size=\"12.00\">2</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;) -->\n<g id=\"edge39\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M755.62,-979.94C773.06,-962.28 799.85,-935.15 821.57,-913.16\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;) -->\n<g id=\"node42\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"980.39\" cy=\"-875.88\" rx=\"54.1\" ry=\"54.1\"/>\n<text text-anchor=\"start\" x=\"962.74\" y=\"-884.28\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n<text text-anchor=\"start\" x=\"945.74\" y=\"-872.28\" font-family=\"Times,serif\" font-size=\"12.00\">transcriptional</text>\n<text text-anchor=\"start\" x=\"962.4\" y=\"-860.28\" font-family=\"Times,serif\" font-size=\"12.00\">speckle</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;) -->\n<g id=\"edge41\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M763.06,-991.62C799.12,-981.44 866.11,-960.16 917.39,-929.98 925.29,-925.34 933.17,-919.7 940.55,-913.85\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;) -->\n<g id=\"node44\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1082.39\" cy=\"-875.88\" rx=\"30.06\" ry=\"30.06\"/>\n<text text-anchor=\"start\" x=\"1064.74\" y=\"-878.28\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n<text text-anchor=\"start\" x=\"1070.39\" y=\"-866.28\" font-family=\"Times,serif\" font-size=\"12.00\">body</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;) -->\n<g id=\"edge43\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M763.93,-998.52C820.13,-997.57 953.38,-988.78 1043.39,-929.98 1053.55,-923.35 1061.98,-913.11 1068.37,-903.38\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;) -->\n<g id=\"node28\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"561.39\" cy=\"-728.39\" rx=\"31.44\" ry=\"31.44\"/>\n<text text-anchor=\"start\" x=\"538.4\" y=\"-724.79\" font-family=\"Times,serif\" font-size=\"12.00\">nucleolus</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;) -->\n<g id=\"edge27\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M624.45,-848.53C610.8,-822.74 590.08,-783.59 576.15,-757.27\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;) -->\n<g id=\"node34\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"642.39\" cy=\"-728.39\" rx=\"31.96\" ry=\"31.96\"/>\n<text text-anchor=\"start\" x=\"624.74\" y=\"-736.79\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n<text text-anchor=\"start\" x=\"623.4\" y=\"-724.79\" font-family=\"Times,serif\" font-size=\"12.00\">splicing</text>\n<text text-anchor=\"start\" x=\"624.4\" y=\"-712.79\" font-family=\"Times,serif\" font-size=\"12.00\">speckle</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;) -->\n<g id=\"edge33\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M639.21,-845.09C639.89,-820.65 640.84,-786.12 641.52,-761.16\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node29\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"440.39\" cy=\"-569.58\" rx=\"61.17\" ry=\"61.17\"/>\n<text text-anchor=\"start\" x=\"400.74\" y=\"-577.98\" font-family=\"Times,serif\" font-size=\"12.00\">ribonucleprotein</text>\n<text text-anchor=\"start\" x=\"419.73\" y=\"-565.98\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"424.73\" y=\"-553.98\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge28\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M542.22,-702.54C524.96,-680.17 499.11,-646.67 477.77,-619.02\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node30\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"561.39\" cy=\"-569.58\" rx=\"41.38\" ry=\"41.38\"/>\n<text text-anchor=\"start\" x=\"548.73\" y=\"-583.98\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n<text text-anchor=\"start\" x=\"535.73\" y=\"-571.98\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n<text text-anchor=\"start\" x=\"540.73\" y=\"-559.98\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"545.73\" y=\"-547.98\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge29\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M561.39,-696.03C561.39,-671.81 561.39,-638.08 561.39,-611.7\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node33\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"666.39\" cy=\"-569.58\" rx=\"45.63\" ry=\"45.63\"/>\n<text text-anchor=\"start\" x=\"637.73\" y=\"-577.98\" font-family=\"Times,serif\" font-size=\"12.00\">splicosomal</text>\n<text text-anchor=\"start\" x=\"645.73\" y=\"-565.98\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"650.73\" y=\"-553.98\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge32\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M578.97,-701.14C595.96,-675.77 621.88,-637.06 641.09,-608.38\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n<g id=\"node31\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"529.39\" cy=\"-422.15\" rx=\"41.38\" ry=\"41.38\"/>\n<text text-anchor=\"start\" x=\"516.73\" y=\"-436.55\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n<text text-anchor=\"start\" x=\"503.73\" y=\"-424.55\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n<text text-anchor=\"start\" x=\"508.73\" y=\"-412.55\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"526.39\" y=\"-400.55\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n<g id=\"edge30\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M552.57,-528.45C548.11,-508.19 542.71,-483.67 538.25,-463.39\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node32\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"627.39\" cy=\"-422.15\" rx=\"39.03\" ry=\"39.03\"/>\n<text text-anchor=\"start\" x=\"614.73\" y=\"-436.55\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n<text text-anchor=\"start\" x=\"608.4\" y=\"-424.55\" font-family=\"Times,serif\" font-size=\"12.00\">splicing</text>\n<text text-anchor=\"start\" x=\"606.73\" y=\"-412.55\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"611.73\" y=\"-400.55\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge31\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M578.39,-531.13C588.55,-508.74 601.38,-480.47 611.36,-458.47\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node36\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"748.39\" cy=\"-728.39\" rx=\"48.92\" ry=\"48.92\"/>\n<text text-anchor=\"start\" x=\"717.4\" y=\"-742.79\" font-family=\"Times,serif\" font-size=\"12.00\">chromosome</text>\n<text text-anchor=\"start\" x=\"718.51\" y=\"-730.79\" font-family=\"Times,serif\" font-size=\"12.00\">organization</text>\n<text text-anchor=\"start\" x=\"727.73\" y=\"-718.79\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"732.73\" y=\"-706.79\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge35\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M741.21,-824.4C742.35,-809.3 743.6,-792.78 744.73,-777.82\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node37\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"777.39\" cy=\"-569.58\" rx=\"47.35\" ry=\"47.35\"/>\n<text text-anchor=\"start\" x=\"753.07\" y=\"-583.98\" font-family=\"Times,serif\" font-size=\"12.00\">chromatin</text>\n<text text-anchor=\"start\" x=\"747.51\" y=\"-571.98\" font-family=\"Times,serif\" font-size=\"12.00\">organization</text>\n<text text-anchor=\"start\" x=\"756.73\" y=\"-559.98\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"761.73\" y=\"-547.98\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge36\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M757.25,-679.49C760.92,-659.67 765.15,-636.78 768.79,-617.11\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node38\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"777.39\" cy=\"-422.15\" rx=\"34.31\" ry=\"34.31\"/>\n<text text-anchor=\"start\" x=\"765.73\" y=\"-430.55\" font-family=\"Times,serif\" font-size=\"12.00\">HAT</text>\n<text text-anchor=\"start\" x=\"756.73\" y=\"-418.55\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"761.73\" y=\"-406.55\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge37\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M777.39,-521.36C777.39,-500.46 777.39,-476.34 777.39,-457.24\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;) -->\n<g id=\"node39\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"777.39\" cy=\"-301.09\" rx=\"34.31\" ry=\"34.31\"/>\n<text text-anchor=\"start\" x=\"762.73\" y=\"-309.49\" font-family=\"Times,serif\" font-size=\"12.00\">NuA4</text>\n<text text-anchor=\"start\" x=\"765.73\" y=\"-297.49\" font-family=\"Times,serif\" font-size=\"12.00\">HAT</text>\n<text text-anchor=\"start\" x=\"756.73\" y=\"-285.49\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;) -->\n<g id=\"edge38\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M777.39,-387.03C777.39,-371.09 777.39,-352.16 777.39,-336.22\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;) -->\n<g id=\"node41\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"857.39\" cy=\"-728.39\" rx=\"39.49\" ry=\"39.49\"/>\n<text text-anchor=\"start\" x=\"833.07\" y=\"-736.79\" font-family=\"Times,serif\" font-size=\"12.00\">chromatin</text>\n<text text-anchor=\"start\" x=\"833.07\" y=\"-724.79\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n<text text-anchor=\"start\" x=\"836.73\" y=\"-712.79\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;) -->\n<g id=\"edge40\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M857.39,-824.4C857.39,-806.18 857.39,-785.9 857.39,-768.77\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;) -->\n<g id=\"node43\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"980.39\" cy=\"-728.39\" rx=\"56\" ry=\"56\"/>\n<text text-anchor=\"start\" x=\"960.07\" y=\"-754.79\" font-family=\"Times,serif\" font-size=\"12.00\">negative</text>\n<text text-anchor=\"start\" x=\"956.07\" y=\"-742.79\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n<text text-anchor=\"start\" x=\"975.4\" y=\"-730.79\" font-family=\"Times,serif\" font-size=\"12.00\">of</text>\n<text text-anchor=\"start\" x=\"967.73\" y=\"-718.79\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n<text text-anchor=\"start\" x=\"950.73\" y=\"-706.79\" font-family=\"Times,serif\" font-size=\"12.00\">biosynthesis</text>\n<text text-anchor=\"start\" x=\"962.4\" y=\"-694.79\" font-family=\"Times,serif\" font-size=\"12.00\">process</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;) -->\n<g id=\"edge42\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M980.39,-821.12C980.39,-809.46 980.39,-797.11 980.39,-785.37\"/>\n</g>\n</g>\n</svg>\n",
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x11abb3e20>"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 123
  },
  {
   "cell_type": "code",
   "id": "ab431982-d156-4b7a-9918-384a5ea5132a",
   "metadata": {
    "tags": [],
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:17.215177Z",
     "start_time": "2025-04-06T16:43:17.212705Z"
    }
   },
   "source": [
    "music_map"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'cell': {'cytoplasm': {'secretory organelles': {'transmembrane transport systems': {'ion transmembrane transport systems': {}}},\n",
       "   'cytoplasmic organelles': {'subgroup of cytoplasmic organelles': {'metabolic organelles': {'subgroup of metabolic organelles': {},\n",
       "      'mitochondrion': {}},\n",
       "     'ribonucleoprotein complex family': {'RNA processing complex 1': {'RNA splicing complex 1': {}},\n",
       "      'ribosome biogenesis community': {'ribosome': {'ribosomal subunit': {'mitochondirial large ribosomal subunit': {},\n",
       "         'mito-cyto ribosomal cluster': {},\n",
       "         'cotranslational protein targeting to membrane system': {},\n",
       "         'ribosomal complex 5': {},\n",
       "         'ribosomal complex 2': {}}},\n",
       "       'nucleic acid binding complex': {}}}}},\n",
       "   'transcriptional regulation complex family': {'transcriptional regulation complexes': {}}},\n",
       "  'nucleus': {'DNA metabolic assembly': {},\n",
       "   'nuclear lumen': {'nucleolus': {'ribonucleprotein complex family': {},\n",
       "     'RNA processing complex family': {'RNA processing complex 1': {},\n",
       "      'RNA splicing complex family': {}},\n",
       "     'splicosomal complex family': {}},\n",
       "    'nuclear splicing speckle': {}},\n",
       "   'nuclear body': {},\n",
       "   'nucleo-plasm 1': {'chromosome organization complex family': {'chromatin organization complex family': {'HAT complex family': {'NuA4 HAT complex': {}}}}},\n",
       "   'nucleo-plasm 2': {'chromatin regulation complex': {}},\n",
       "   'nuclear transcriptional speckle': {'negative regulation of RNA biosynthesis process': {}}}}}"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 124
  },
  {
   "cell_type": "code",
   "id": "5a76608a-fff0-4005-9f32-c02afef39d9e",
   "metadata": {
    "tags": [],
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:17.352175Z",
     "start_time": "2025-04-06T16:43:17.220267Z"
    }
   },
   "source": [
    "import copy\n",
    "\n",
    "transcription_process = {\n",
    "    '_type': 'process',\n",
    "    '_outputs': {\n",
    "        'RNA processing': 'any',\n",
    "        'regulation': 'any',\n",
    "        'nucleic acids': 'any',\n",
    "        'negative regulation': 'any'\n",
    "    },\n",
    "    'outputs': {\n",
    "        'RNA processing': [\n",
    "            'nucleus', 'nuclear lumen', 'nucleolus', 'RNA processing complex family'],\n",
    "        'regulation': [\n",
    "            'cytoplasm', 'transcriptional regulation complex family', 'transcriptional regulation complexes'],\n",
    "        'nucleic acids': [\n",
    "            'cytoplasm', 'cytoplasmic organelles', 'subgroup of cytoplasmic organelles', \n",
    "            'ribonucleoprotein complex family', 'ribosome biogenesis community', 'nucleic acid binding complex'],\n",
    "        'negative regulation': [\n",
    "            'nucleus', 'nuclear transcriptional speckle', 'negative regulation of RNA biosynthesis process'],\n",
    "    }\n",
    "}\n",
    "\n",
    "translation_process = {\n",
    "    '_type': 'process',\n",
    "    '_outputs': {\n",
    "        'ribosomes': 'any',\n",
    "        'membrane': 'any'\n",
    "    },\n",
    "    'outputs': {\n",
    "        'ribosomes': [\n",
    "            'cytoplasm', 'cytoplasmic organelles', 'subgroup of cytoplasmic organelles', 'ribonucleoprotein complex family', 'ribosome biogenesis community', 'ribosome'],\n",
    "        # 'membrane': ['],\n",
    "    }\n",
    "}\n",
    "\n",
    "metabolism_process = {\n",
    "    '_type': 'process',\n",
    "    '_outputs': {\n",
    "        'organelles': 'any',\n",
    "        'mitochondrion': 'any',\n",
    "        'ion transport': 'any',\n",
    "    },\n",
    "    'outputs': {\n",
    "        'organelles': [\n",
    "            'cytoplasm', 'cytoplasmic organelles', 'subgroup of cytoplasmic organelles', 'metabolic organelles', ],\n",
    "        'mitochondrion': [\n",
    "            'cytoplasm', 'cytoplasmic organelles', 'subgroup of cytoplasmic organelles', 'metabolic organelles', 'mitochondrion'],\n",
    "        'ion transport': [\n",
    "            'cytoplasm', 'secretory organelles', 'transmembrane transport systems', 'ion transmembrane transport systems'],\n",
    "    }\n",
    "}\n",
    "\n",
    "cell_cycle_process = {\n",
    "    '_type': 'process',\n",
    "    '_outputs': {\n",
    "        'DNA': 'any',\n",
    "        'chromosome': 'any',\n",
    "        'chromatin': 'any',\n",
    "    },\n",
    "    'outputs': {        \n",
    "        'DNA': ['nucleus', 'DNA metabolic assembly'],\n",
    "        'chromosome': ['nucleus', 'nucleo-plasm 1', 'chromosome organization complex family',],\n",
    "        'chromatin': ['nucleus', 'nucleo-plasm 1', 'chromosome organization complex family', 'chromatin organization complex family'],\n",
    "    }\n",
    "}\n",
    "\n",
    "signalling_process = {\n",
    "    '_type': 'process',\n",
    "    '_outputs': {\n",
    "        'transport': 'any',\n",
    "        'trancript regulation': 'any',\n",
    "    },\n",
    "    'outputs': {        \n",
    "        'transport': [\n",
    "            'cytoplasm', 'secretory organelles', 'transmembrane transport systems'],\n",
    "        'trancript regulation': [\n",
    "            'cytoplasm', 'transcriptional regulation complex family', 'transcriptional regulation complexes'],\n",
    "    }\n",
    "}\n",
    "\n",
    "protein_transport_process = {\n",
    "    '_type': 'process',\n",
    "    '_outputs': {\n",
    "        'transmembrane transport': 'any',\n",
    "        'secretory': 'any',\n",
    "        'cotranslation': 'any',\n",
    "    },\n",
    "    'outputs': {        \n",
    "        'transmembrane transport': [\n",
    "            'cytoplasm', 'secretory organelles', 'transmembrane transport systems'],\n",
    "        'secretory': [\n",
    "            'cytoplasm', 'secretory organelles'],\n",
    "        'cotranslation': [\n",
    "            'cytoplasm', 'cytoplasmic organelles', 'subgroup of cytoplasmic organelles', 'ribonucleoprotein complex family', \n",
    "            'ribosome biogenesis community', 'ribosome', 'ribosomal subunit', 'cotranslational protein targeting to membrane system']\n",
    "    }\n",
    "}\n",
    "\n",
    "rna_processing_process = {\n",
    "    '_type': 'process',\n",
    "    '_outputs': {\n",
    "        'processing complex': 'any',\n",
    "    },\n",
    "    'outputs': {        \n",
    "        'processing complex': [\n",
    "            'nucleus', 'nuclear lumen', 'nucleolus', 'RNA processing complex family'],\n",
    "    }\n",
    "}\n",
    "\n",
    "# compose schema\n",
    "music_map['cell']['transcription'] = transcription_process\n",
    "music_map['cell']['translation'] = translation_process\n",
    "music_map['cell']['metabolism'] = metabolism_process\n",
    "music_map['cell']['cell cycle'] = cell_cycle_process\n",
    "music_map['cell']['signalling'] = signalling_process\n",
    "music_map['cell']['protein transport'] = protein_transport_process\n",
    "music_map['cell']['RNA processing'] = rna_processing_process\n",
    "\n",
    "music_map2 = replace_regex_recursive(music_map)\n",
    "\n",
    "# plot\n",
    "plot_settingsx = copy.deepcopy(plot_settings)\n",
    "plot_settingsx['label_margin'] = '0.02'\n",
    "plot_settingsx['node_border_colors'] = {\n",
    "    ('cell', 'translation'): 'blue',\n",
    "    ('cell', 'transcription'): 'blue',\n",
    "    ('cell', 'metabolism'): 'blue',\n",
    "    ('cell', 'cell<br/>cycle'): 'blue',\n",
    "    ('cell', 'signalling'): 'blue',\n",
    "    ('cell', 'protein<br/>transport'): 'blue',\n",
    "    ('cell', 'RNA<br/>processing'): 'blue',\n",
    "}\n",
    "# plot_settingsx.update({'out_dir': 'out','dpi': '250'})\n",
    "plot_bigraph(music_map2, **plot_settingsx, filename='cell_hierarchy_functions')"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/cell_hierarchy_functions\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n<!-- Generated by graphviz version 9.0.0 (0)\n -->\n<!-- Title: bigraph Pages: 1 -->\n<svg width=\"814pt\" height=\"700pt\"\n viewBox=\"0.00 0.00 837.01 720.00\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<g id=\"graph0\" class=\"graph\" transform=\"scale(0.493187 0.493187) rotate(0) translate(4 1415.34)\">\n<title>bigraph</title>\n<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-1415.34 1646,-1415.34 1646,4 -4,4\"/>\n<!-- (&#39;cell&#39;,) -->\n<g id=\"node1\" class=\"node\">\n<title>(&#39;cell&#39;,)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"771\" cy=\"-1393.34\" rx=\"18\" ry=\"18\"/>\n<text text-anchor=\"start\" x=\"762.34\" y=\"-1389.74\" font-family=\"Times,serif\" font-size=\"12.00\">cell</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;) -->\n<g id=\"node2\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"366\" cy=\"-1309.68\" rx=\"21.71\" ry=\"21.71\"/>\n<text text-anchor=\"start\" x=\"347.67\" y=\"-1306.08\" font-family=\"Times,serif\" font-size=\"12.00\">nucleus</text>\n</g>\n<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;) -->\n<g id=\"edge8\" class=\"edge\">\n<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M752.81,-1388.67C686.78,-1375.36 459.84,-1329.6 387.99,-1315.11\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;) -->\n<g id=\"node22\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"927\" cy=\"-1309.68\" rx=\"28.67\" ry=\"28.67\"/>\n<text text-anchor=\"start\" x=\"902.34\" y=\"-1306.08\" font-family=\"Times,serif\" font-size=\"12.00\">cytoplasm</text>\n</g>\n<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;) -->\n<g id=\"edge28\" class=\"edge\">\n<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M787.4,-1383.76C814.54,-1369.55 869.02,-1341.03 901.31,-1324.13\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;transcription&#39;) -->\n<g id=\"node45\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;transcription&#39;)</title>\n<polygon fill=\"none\" stroke=\"blue\" stroke-width=\"2\" points=\"726.77,-278.87 663.23,-278.87 663.23,-242.87 726.77,-242.87 726.77,-278.87\"/>\n<text text-anchor=\"start\" x=\"664.67\" y=\"-257.27\" font-family=\"Times,serif\" font-size=\"12.00\">transcription</text>\n</g>\n<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;) -->\n<!-- (&#39;cell&#39;, &#39;translation&#39;) -->\n<g id=\"node46\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;translation&#39;)</title>\n<polygon fill=\"none\" stroke=\"blue\" stroke-width=\"2\" points=\"1582,-278.87 1528,-278.87 1528,-242.87 1582,-242.87 1582,-278.87\"/>\n<text text-anchor=\"start\" x=\"1529.67\" y=\"-257.27\" font-family=\"Times,serif\" font-size=\"12.00\">translation</text>\n</g>\n<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;translation&#39;) -->\n<!-- (&#39;cell&#39;, &#39;metabolism&#39;) -->\n<g id=\"node47\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;metabolism&#39;)</title>\n<polygon fill=\"none\" stroke=\"blue\" stroke-width=\"2\" points=\"907.44,-406.82 848.56,-406.82 848.56,-370.82 907.44,-370.82 907.44,-406.82\"/>\n<text text-anchor=\"start\" x=\"850\" y=\"-385.22\" font-family=\"Times,serif\" font-size=\"12.00\">metabolism</text>\n</g>\n<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;) -->\n<!-- (&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;) -->\n<g id=\"node48\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;)</title>\n<polygon fill=\"none\" stroke=\"blue\" stroke-width=\"2\" points=\"88,-553.14 34,-553.14 34,-517.14 88,-517.14 88,-553.14\"/>\n<text text-anchor=\"start\" x=\"52.34\" y=\"-537.54\" font-family=\"Times,serif\" font-size=\"12.00\">cell</text>\n<text text-anchor=\"start\" x=\"48.34\" y=\"-525.54\" font-family=\"Times,serif\" font-size=\"12.00\">cycle</text>\n</g>\n<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;) -->\n<!-- (&#39;cell&#39;, &#39;signalling&#39;) -->\n<g id=\"node49\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;signalling&#39;)</title>\n<polygon fill=\"none\" stroke=\"blue\" stroke-width=\"2\" points=\"1284,-553.14 1230,-553.14 1230,-517.14 1284,-517.14 1284,-553.14\"/>\n<text text-anchor=\"start\" x=\"1233.33\" y=\"-531.54\" font-family=\"Times,serif\" font-size=\"12.00\">signalling</text>\n</g>\n<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;signalling&#39;) -->\n<!-- (&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;) -->\n<g id=\"node50\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;)</title>\n<polygon fill=\"none\" stroke=\"blue\" stroke-width=\"2\" points=\"1375,-36 1321,-36 1321,0 1375,0 1375,-36\"/>\n<text text-anchor=\"start\" x=\"1331\" y=\"-20.4\" font-family=\"Times,serif\" font-size=\"12.00\">protein</text>\n<text text-anchor=\"start\" x=\"1326.67\" y=\"-8.4\" font-family=\"Times,serif\" font-size=\"12.00\">transport</text>\n</g>\n<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;) -->\n<!-- (&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;) -->\n<g id=\"node51\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;)</title>\n<polygon fill=\"none\" stroke=\"blue\" stroke-width=\"2\" points=\"578.1,-553.14 523.9,-553.14 523.9,-517.14 578.1,-517.14 578.1,-553.14\"/>\n<text text-anchor=\"start\" x=\"538.33\" y=\"-537.54\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n<text text-anchor=\"start\" x=\"525.34\" y=\"-525.54\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n</g>\n<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;) -->\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;) -->\n<g id=\"node3\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"294\" cy=\"-1192.96\" rx=\"27\" ry=\"27\"/>\n<text text-anchor=\"start\" x=\"276.35\" y=\"-1195.36\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n<text text-anchor=\"start\" x=\"279\" y=\"-1183.36\" font-family=\"Times,serif\" font-size=\"12.00\">lumen</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;) -->\n<g id=\"edge9\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M354.43,-1290.25C341.87,-1270.23 321.79,-1238.25 308.27,-1216.69\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;) -->\n<g id=\"node11\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"576\" cy=\"-1192.96\" rx=\"51.05\" ry=\"51.05\"/>\n<text text-anchor=\"start\" x=\"558.35\" y=\"-1201.36\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n<text text-anchor=\"start\" x=\"541.34\" y=\"-1189.36\" font-family=\"Times,serif\" font-size=\"12.00\">transcriptional</text>\n<text text-anchor=\"start\" x=\"558.01\" y=\"-1177.36\" font-family=\"Times,serif\" font-size=\"12.00\">speckle</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;) -->\n<g id=\"edge17\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M387.06,-1301.77C416.96,-1291.5 472.87,-1270.52 516,-1244.01 523.51,-1239.39 531.08,-1233.89 538.18,-1228.26\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;) -->\n<g id=\"node13\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"198\" cy=\"-1192.96\" rx=\"47.75\" ry=\"47.75\"/>\n<text text-anchor=\"start\" x=\"165.68\" y=\"-1195.36\" font-family=\"Times,serif\" font-size=\"12.00\">nucleo&#45;plasm</text>\n<text text-anchor=\"start\" x=\"195\" y=\"-1183.36\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;) -->\n<g id=\"edge19\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M346.71,-1298.32C324.65,-1286.22 287.76,-1265.14 258,-1244.01 250.04,-1238.36 241.81,-1231.92 234.09,-1225.6\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;) -->\n<g id=\"node18\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"459\" cy=\"-1192.96\" rx=\"47.75\" ry=\"47.75\"/>\n<text text-anchor=\"start\" x=\"426.68\" y=\"-1195.36\" font-family=\"Times,serif\" font-size=\"12.00\">nucleo&#45;plasm</text>\n<text text-anchor=\"start\" x=\"456\" y=\"-1183.36\" font-family=\"Times,serif\" font-size=\"12.00\">2</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;) -->\n<g id=\"edge24\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M379.72,-1291.75C392.66,-1275.8 412.48,-1251.35 429.06,-1230.89\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;) -->\n<g id=\"node20\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"68\" cy=\"-1072.87\" rx=\"35.49\" ry=\"35.49\"/>\n<text text-anchor=\"start\" x=\"55\" y=\"-1081.27\" font-family=\"Times,serif\" font-size=\"12.00\">DNA</text>\n<text text-anchor=\"start\" x=\"44.34\" y=\"-1069.27\" font-family=\"Times,serif\" font-size=\"12.00\">metabolic</text>\n<text text-anchor=\"start\" x=\"45.67\" y=\"-1057.27\" font-family=\"Times,serif\" font-size=\"12.00\">assembly</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;) -->\n<g id=\"edge26\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M343.55,-1308.33C299.37,-1306.46 200.29,-1296.3 141,-1244.01 100.86,-1208.61 82.05,-1147.29 73.81,-1108.74\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;) -->\n<g id=\"node21\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"366\" cy=\"-1192.96\" rx=\"27\" ry=\"27\"/>\n<text text-anchor=\"start\" x=\"348.35\" y=\"-1195.36\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n<text text-anchor=\"start\" x=\"354\" y=\"-1183.36\" font-family=\"Times,serif\" font-size=\"12.00\">body</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;) -->\n<g id=\"edge27\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M366,-1287.1C366,-1268.28 366,-1240.9 366,-1220.69\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;) -->\n<g id=\"node4\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"299\" cy=\"-957.96\" rx=\"26.83\" ry=\"26.83\"/>\n<text text-anchor=\"start\" x=\"276\" y=\"-954.36\" font-family=\"Times,serif\" font-size=\"12.00\">nucleolus</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;) -->\n<g id=\"edge10\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M294.57,-1165.24C295.54,-1120.41 297.46,-1030.59 298.43,-985.73\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;) -->\n<g id=\"node10\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"373\" cy=\"-957.96\" rx=\"28.91\" ry=\"28.91\"/>\n<text text-anchor=\"start\" x=\"355.35\" y=\"-966.36\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n<text text-anchor=\"start\" x=\"354\" y=\"-954.36\" font-family=\"Times,serif\" font-size=\"12.00\">splicing</text>\n<text text-anchor=\"start\" x=\"355.01\" y=\"-942.36\" font-family=\"Times,serif\" font-size=\"12.00\">speckle</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;) -->\n<g id=\"edge16\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M302.6,-1166.58C317.6,-1122.35 348.39,-1031.53 363.8,-986.1\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node5\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"299\" cy=\"-707.84\" rx=\"38.33\" ry=\"38.33\"/>\n<text text-anchor=\"start\" x=\"286.33\" y=\"-722.24\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n<text text-anchor=\"start\" x=\"273.34\" y=\"-710.24\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n<text text-anchor=\"start\" x=\"278.34\" y=\"-698.24\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"283.34\" y=\"-686.24\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge11\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M299,-930.4C299,-886.27 299,-797.67 299,-746.9\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node8\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"413\" cy=\"-707.84\" rx=\"58.11\" ry=\"58.11\"/>\n<text text-anchor=\"start\" x=\"373.35\" y=\"-716.24\" font-family=\"Times,serif\" font-size=\"12.00\">ribonucleprotein</text>\n<text text-anchor=\"start\" x=\"392.34\" y=\"-704.24\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"397.34\" y=\"-692.24\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge14\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M310.14,-932.71C328.29,-893.21 364.66,-814.05 388.92,-761.24\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node9\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"532\" cy=\"-707.84\" rx=\"42.57\" ry=\"42.57\"/>\n<text text-anchor=\"start\" x=\"503.34\" y=\"-716.24\" font-family=\"Times,serif\" font-size=\"12.00\">splicosomal</text>\n<text text-anchor=\"start\" x=\"511.34\" y=\"-704.24\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"516.34\" y=\"-692.24\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge15\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M313.98,-934.94C319.94,-927.14 327.23,-918.68 335,-912.09 347.44,-901.53 354.23,-904.25 367,-894.09 425.33,-847.7 430.79,-826.16 480,-770.19 488.11,-760.97 496.79,-750.83 504.69,-741.51\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n<g id=\"node6\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"344\" cy=\"-535.14\" rx=\"38.33\" ry=\"38.33\"/>\n<text text-anchor=\"start\" x=\"331.33\" y=\"-549.54\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n<text text-anchor=\"start\" x=\"318.34\" y=\"-537.54\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n<text text-anchor=\"start\" x=\"323.34\" y=\"-525.54\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"341\" y=\"-513.54\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n<g id=\"edge12\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M308.76,-669.8C316.31,-641.17 326.71,-601.72 334.25,-573.11\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node7\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"436\" cy=\"-535.14\" rx=\"35.98\" ry=\"35.98\"/>\n<text text-anchor=\"start\" x=\"423.33\" y=\"-549.54\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n<text text-anchor=\"start\" x=\"417\" y=\"-537.54\" font-family=\"Times,serif\" font-size=\"12.00\">splicing</text>\n<text text-anchor=\"start\" x=\"415.34\" y=\"-525.54\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"420.34\" y=\"-513.54\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge13\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M321.98,-676.4C329.56,-666.48 338.07,-655.46 346,-645.48 368.32,-617.37 394.46,-585.76 412.79,-563.78\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;) -->\n<g id=\"edge51\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M285.11,-659.1C273.93,-610.65 264.83,-535.22 297,-480.8 378.78,-342.48 582.5,-285.66 662.25,-268.22\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"281.69,-659.85 287.46,-668.74 288.49,-658.2 281.69,-659.85\"/>\n<text text-anchor=\"start\" x=\"334.83\" y=\"-454.8\" font-family=\"Times,serif\" font-size=\"10.00\">RNA</text>\n<text text-anchor=\"start\" x=\"324\" y=\"-444.8\" font-family=\"Times,serif\" font-size=\"10.00\">processing</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;) -->\n<g id=\"edge67\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M325.85,-665.48C331.86,-658.18 338.66,-651.12 346,-645.48 358.88,-635.58 466.92,-597.58 481,-589.48 498.75,-579.27 517.09,-565.13 530.55,-553.98\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"323.46,-662.85 320.09,-672.9 328.98,-667.14 323.46,-662.85\"/>\n<text text-anchor=\"start\" x=\"435\" y=\"-619.48\" font-family=\"Times,serif\" font-size=\"10.00\">processing</text>\n<text text-anchor=\"start\" x=\"439.17\" y=\"-609.48\" font-family=\"Times,serif\" font-size=\"10.00\">complex</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;) -->\n<g id=\"node12\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"587\" cy=\"-841.14\" rx=\"52.95\" ry=\"52.95\"/>\n<text text-anchor=\"start\" x=\"566.68\" y=\"-867.54\" font-family=\"Times,serif\" font-size=\"12.00\">negative</text>\n<text text-anchor=\"start\" x=\"562.67\" y=\"-855.54\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n<text text-anchor=\"start\" x=\"582\" y=\"-843.54\" font-family=\"Times,serif\" font-size=\"12.00\">of</text>\n<text text-anchor=\"start\" x=\"574.33\" y=\"-831.54\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n<text text-anchor=\"start\" x=\"557.33\" y=\"-819.54\" font-family=\"Times,serif\" font-size=\"12.00\">biosynthesis</text>\n<text text-anchor=\"start\" x=\"569.01\" y=\"-807.54\" font-family=\"Times,serif\" font-size=\"12.00\">process</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;) -->\n<g id=\"edge18\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M577.6,-1141.1C579.67,-1075.34 583.25,-961.4 585.34,-894.79\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;) -->\n<g id=\"edge53\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M594.65,-776.68C603.74,-704.64 620.27,-583.76 640.46,-480.8 655.07,-406.29 662.29,-388.43 681,-314.84 684,-303.04 687.5,-289.8 690.25,-279.51\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"591.21,-775.95 593.44,-786.31 598.16,-776.83 591.21,-775.95\"/>\n<text text-anchor=\"start\" x=\"643.33\" y=\"-537.14\" font-family=\"Times,serif\" font-size=\"10.00\">negative</text>\n<text text-anchor=\"start\" x=\"640\" y=\"-527.14\" font-family=\"Times,serif\" font-size=\"10.00\">regulation</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node14\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"198\" cy=\"-957.96\" rx=\"45.87\" ry=\"45.87\"/>\n<text text-anchor=\"start\" x=\"167.01\" y=\"-972.36\" font-family=\"Times,serif\" font-size=\"12.00\">chromosome</text>\n<text text-anchor=\"start\" x=\"168.12\" y=\"-960.36\" font-family=\"Times,serif\" font-size=\"12.00\">organization</text>\n<text text-anchor=\"start\" x=\"177.34\" y=\"-948.36\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"182.34\" y=\"-936.36\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge20\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M198,-1144.25C198,-1103.37 198,-1044.9 198,-1004.63\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node15\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"198\" cy=\"-707.84\" rx=\"44.3\" ry=\"44.3\"/>\n<text text-anchor=\"start\" x=\"173.67\" y=\"-722.24\" font-family=\"Times,serif\" font-size=\"12.00\">chromatin</text>\n<text text-anchor=\"start\" x=\"168.12\" y=\"-710.24\" font-family=\"Times,serif\" font-size=\"12.00\">organization</text>\n<text text-anchor=\"start\" x=\"177.34\" y=\"-698.24\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"182.34\" y=\"-686.24\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge21\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M198,-911.16C198,-866.07 198,-797.67 198,-753.12\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;) -->\n<g id=\"edge61\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M165.28,-910.22C141.42,-873.77 110.4,-820.97 93.34,-770.19 67.29,-692.62 62.2,-593.81 61.22,-553.97\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"162.21,-911.92 170.65,-918.32 168.04,-908.06 162.21,-911.92\"/>\n<text text-anchor=\"start\" x=\"94\" y=\"-704.84\" font-family=\"Times,serif\" font-size=\"10.00\">chromosome</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node16\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"198\" cy=\"-535.14\" rx=\"31.25\" ry=\"31.25\"/>\n<text text-anchor=\"start\" x=\"186.33\" y=\"-543.54\" font-family=\"Times,serif\" font-size=\"12.00\">HAT</text>\n<text text-anchor=\"start\" x=\"177.34\" y=\"-531.54\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"182.34\" y=\"-519.54\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge22\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M198,-662.88C198,-632.95 198,-594.05 198,-567.29\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;) -->\n<g id=\"edge60\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M163.27,-663.57C134.78,-628.07 95.89,-579.61 75.24,-553.88\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"160.34,-665.5 169.33,-671.11 165.8,-661.12 160.34,-665.5\"/>\n<text text-anchor=\"start\" x=\"134\" y=\"-614.48\" font-family=\"Times,serif\" font-size=\"10.00\">chromatin</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;) -->\n<g id=\"node17\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"198\" cy=\"-388.82\" rx=\"31.25\" ry=\"31.25\"/>\n<text text-anchor=\"start\" x=\"183.33\" y=\"-397.22\" font-family=\"Times,serif\" font-size=\"12.00\">NuA4</text>\n<text text-anchor=\"start\" x=\"186.33\" y=\"-385.22\" font-family=\"Times,serif\" font-size=\"12.00\">HAT</text>\n<text text-anchor=\"start\" x=\"177.34\" y=\"-373.22\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;) -->\n<g id=\"edge23\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M198,-503.18C198,-478.83 198,-445.07 198,-420.73\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;) -->\n<g id=\"node19\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"459\" cy=\"-957.96\" rx=\"36.44\" ry=\"36.44\"/>\n<text text-anchor=\"start\" x=\"434.67\" y=\"-966.36\" font-family=\"Times,serif\" font-size=\"12.00\">chromatin</text>\n<text text-anchor=\"start\" x=\"434.67\" y=\"-954.36\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n<text text-anchor=\"start\" x=\"438.34\" y=\"-942.36\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;) -->\n<g id=\"edge25\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M459,-1144.25C459,-1100.05 459,-1035.26 459,-995.17\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;) -->\n<g id=\"edge59\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M67.02,-1025.17C65.38,-944.21 62.14,-772.54 61,-627.48 60.8,-601.95 60.85,-572.4 60.92,-553.99\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"63.52,-1025.01 67.22,-1034.93 70.52,-1024.86 63.52,-1025.01\"/>\n<text text-anchor=\"start\" x=\"65\" y=\"-838.14\" font-family=\"Times,serif\" font-size=\"10.00\">DNA</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node23\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"901\" cy=\"-1072.87\" rx=\"51.05\" ry=\"51.05\"/>\n<text text-anchor=\"start\" x=\"866.34\" y=\"-1087.27\" font-family=\"Times,serif\" font-size=\"12.00\">transcriptional</text>\n<text text-anchor=\"start\" x=\"876.67\" y=\"-1075.27\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n<text text-anchor=\"start\" x=\"880.34\" y=\"-1063.27\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"885.34\" y=\"-1051.27\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge29\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M923.86,-1280.35C919.6,-1241.81 911.89,-1172.19 906.61,-1124.54\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n<g id=\"node25\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"980\" cy=\"-1192.96\" rx=\"43.04\" ry=\"43.04\"/>\n<text text-anchor=\"start\" x=\"951.01\" y=\"-1195.36\" font-family=\"Times,serif\" font-size=\"12.00\">cytoplasmic</text>\n<text text-anchor=\"start\" x=\"955.45\" y=\"-1183.36\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n<g id=\"edge31\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M939.03,-1282.65C945.81,-1267.97 954.41,-1249.35 961.98,-1232.96\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;) -->\n<g id=\"node42\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1202\" cy=\"-957.96\" rx=\"36.76\" ry=\"36.76\"/>\n<text text-anchor=\"start\" x=\"1180.01\" y=\"-960.36\" font-family=\"Times,serif\" font-size=\"12.00\">secretory</text>\n<text text-anchor=\"start\" x=\"1177.45\" y=\"-948.36\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;) -->\n<g id=\"edge48\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M953.6,-1296.8C976.42,-1285.64 1009.15,-1267.18 1032,-1244.01 1107.85,-1167.07 1163.79,-1049.92 1188.29,-992.74\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;) -->\n<g id=\"node24\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"878\" cy=\"-707.84\" rx=\"51.05\" ry=\"51.05\"/>\n<text text-anchor=\"start\" x=\"843.34\" y=\"-716.24\" font-family=\"Times,serif\" font-size=\"12.00\">transcriptional</text>\n<text text-anchor=\"start\" x=\"853.67\" y=\"-704.24\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n<text text-anchor=\"start\" x=\"852.34\" y=\"-692.24\" font-family=\"Times,serif\" font-size=\"12.00\">complexes</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;) -->\n<g id=\"edge30\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M897.78,-1021.01C893.39,-951.83 885.6,-828.8 881.22,-759.65\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;) -->\n<g id=\"edge54\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M836.68,-659.94C820.29,-639.43 802.43,-614.39 790,-589.48 775.59,-560.6 716.83,-343.34 699.71,-279.47\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"833.85,-662.01 842.88,-667.56 839.28,-657.59 833.85,-662.01\"/>\n<text text-anchor=\"start\" x=\"749\" y=\"-449.8\" font-family=\"Times,serif\" font-size=\"10.00\">regulation</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;)&#45;&gt;(&#39;cell&#39;, &#39;signalling&#39;) -->\n<g id=\"edge63\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;)&#45;&gt;(&#39;cell&#39;, &#39;signalling&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M918.14,-659.75C924.35,-654.34 931.03,-649.38 938,-645.48 1035.45,-590.89 1081.22,-635.5 1183,-589.48 1202.91,-580.47 1222.86,-565.68 1237.02,-553.95\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"916.03,-656.93 911.09,-666.3 920.79,-662.06 916.03,-656.93\"/>\n<text text-anchor=\"start\" x=\"1130.33\" y=\"-619.48\" font-family=\"Times,serif\" font-size=\"10.00\">trancript</text>\n<text text-anchor=\"start\" x=\"1127\" y=\"-609.48\" font-family=\"Times,serif\" font-size=\"10.00\">regulation</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n<g id=\"node26\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"984\" cy=\"-957.96\" rx=\"43.04\" ry=\"43.04\"/>\n<text text-anchor=\"start\" x=\"961.67\" y=\"-972.36\" font-family=\"Times,serif\" font-size=\"12.00\">subgroup</text>\n<text text-anchor=\"start\" x=\"979\" y=\"-960.36\" font-family=\"Times,serif\" font-size=\"12.00\">of</text>\n<text text-anchor=\"start\" x=\"955.01\" y=\"-948.36\" font-family=\"Times,serif\" font-size=\"12.00\">cytoplasmic</text>\n<text text-anchor=\"start\" x=\"959.45\" y=\"-936.36\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n<g id=\"edge32\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M980.74,-1148.96C981.46,-1107.01 982.55,-1043.58 983.27,-1001.72\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"node27\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1340\" cy=\"-707.84\" rx=\"62.36\" ry=\"62.36\"/>\n<text text-anchor=\"start\" x=\"1297.35\" y=\"-716.24\" font-family=\"Times,serif\" font-size=\"12.00\">ribonucleoprotein</text>\n<text text-anchor=\"start\" x=\"1319.34\" y=\"-704.24\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"1324.34\" y=\"-692.24\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n<g id=\"edge33\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1020.71,-934.2C1075.14,-900.09 1179.95,-833.01 1265,-770.19 1273.93,-763.6 1283.24,-756.32 1292.14,-749.16\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;) -->\n<g id=\"node39\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"984\" cy=\"-707.84\" rx=\"36.76\" ry=\"36.76\"/>\n<text text-anchor=\"start\" x=\"960.34\" y=\"-710.24\" font-family=\"Times,serif\" font-size=\"12.00\">metabolic</text>\n<text text-anchor=\"start\" x=\"959.45\" y=\"-698.24\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;) -->\n<g id=\"edge45\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M984,-914.02C984,-866.31 984,-790.09 984,-745.2\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;) -->\n<g id=\"node28\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1343\" cy=\"-535.14\" rx=\"40.69\" ry=\"40.69\"/>\n<text text-anchor=\"start\" x=\"1320.67\" y=\"-543.54\" font-family=\"Times,serif\" font-size=\"12.00\">ribosome</text>\n<text text-anchor=\"start\" x=\"1317.67\" y=\"-531.54\" font-family=\"Times,serif\" font-size=\"12.00\">biogenesis</text>\n<text text-anchor=\"start\" x=\"1315.67\" y=\"-519.54\" font-family=\"Times,serif\" font-size=\"12.00\">community</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;) -->\n<g id=\"edge34\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1341.09,-644.73C1341.49,-622.04 1341.93,-596.98 1342.29,-576.66\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n<g id=\"node37\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1440\" cy=\"-535.14\" rx=\"38.33\" ry=\"38.33\"/>\n<text text-anchor=\"start\" x=\"1427.33\" y=\"-549.54\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n<text text-anchor=\"start\" x=\"1414.34\" y=\"-537.54\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n<text text-anchor=\"start\" x=\"1419.34\" y=\"-525.54\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"1437\" y=\"-513.54\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n<g id=\"edge43\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1371.37,-653.29C1387.52,-625.72 1406.7,-592.98 1420.67,-569.13\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;) -->\n<g id=\"node29\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1050\" cy=\"-388.82\" rx=\"35.98\" ry=\"35.98\"/>\n<text text-anchor=\"start\" x=\"1032.68\" y=\"-403.22\" font-family=\"Times,serif\" font-size=\"12.00\">nucleic</text>\n<text text-anchor=\"start\" x=\"1040.01\" y=\"-391.22\" font-family=\"Times,serif\" font-size=\"12.00\">acid</text>\n<text text-anchor=\"start\" x=\"1031.67\" y=\"-379.22\" font-family=\"Times,serif\" font-size=\"12.00\">binding</text>\n<text text-anchor=\"start\" x=\"1029.34\" y=\"-367.22\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;) -->\n<g id=\"edge35\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1317.63,-502.27C1310.3,-494.49 1301.87,-486.71 1293,-480.8 1226.47,-436.45 1135.38,-409.86 1085.77,-397.73\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;) -->\n<g id=\"node30\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1343\" cy=\"-388.82\" rx=\"26.1\" ry=\"26.1\"/>\n<text text-anchor=\"start\" x=\"1320.67\" y=\"-385.22\" font-family=\"Times,serif\" font-size=\"12.00\">ribosome</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;) -->\n<g id=\"edge36\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1343,-493.55C1343,-468.54 1343,-437.23 1343,-415.58\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;) -->\n<g id=\"edge52\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M1005.03,-371.86C932.47,-346.12 791.28,-296.03 727.57,-273.43\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"1003.62,-375.08 1014.21,-375.12 1005.96,-368.48 1003.62,-375.08\"/>\n<text text-anchor=\"start\" x=\"893\" y=\"-326.84\" font-family=\"Times,serif\" font-size=\"10.00\">nucleic</text>\n<text text-anchor=\"start\" x=\"897.16\" y=\"-316.84\" font-family=\"Times,serif\" font-size=\"10.00\">acids</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;) -->\n<g id=\"node31\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"969\" cy=\"-260.87\" rx=\"35.97\" ry=\"35.97\"/>\n<text text-anchor=\"start\" x=\"945\" y=\"-263.27\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n<text text-anchor=\"start\" x=\"951.33\" y=\"-251.27\" font-family=\"Times,serif\" font-size=\"12.00\">subunit</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;) -->\n<g id=\"edge37\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1327.55,-367.04C1313.83,-350.19 1292.25,-327.27 1268,-314.84 1181.44,-270.49 1064.34,-262.55 1005.69,-261.58\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)&#45;&gt;(&#39;cell&#39;, &#39;translation&#39;) -->\n<g id=\"edge55\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)&#45;&gt;(&#39;cell&#39;, &#39;translation&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M1370.41,-362.4C1374.51,-359.03 1378.78,-355.74 1383,-352.84 1430.87,-319.92 1492.06,-290.17 1527.06,-274.19\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"1368.26,-359.64 1362.94,-368.81 1372.81,-364.95 1368.26,-359.64\"/>\n<text text-anchor=\"start\" x=\"1442\" y=\"-321.84\" font-family=\"Times,serif\" font-size=\"10.00\">ribosomes</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;) -->\n<g id=\"node32\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1121\" cy=\"-134.95\" rx=\"52.95\" ry=\"52.95\"/>\n<text text-anchor=\"start\" x=\"1085.68\" y=\"-161.35\" font-family=\"Times,serif\" font-size=\"12.00\">cotranslational</text>\n<text text-anchor=\"start\" x=\"1104\" y=\"-149.35\" font-family=\"Times,serif\" font-size=\"12.00\">protein</text>\n<text text-anchor=\"start\" x=\"1099.78\" y=\"-137.35\" font-family=\"Times,serif\" font-size=\"12.00\">targeting</text>\n<text text-anchor=\"start\" x=\"1116.33\" y=\"-125.35\" font-family=\"Times,serif\" font-size=\"12.00\">to</text>\n<text text-anchor=\"start\" x=\"1095.68\" y=\"-113.35\" font-family=\"Times,serif\" font-size=\"12.00\">membrane</text>\n<text text-anchor=\"start\" x=\"1104.33\" y=\"-101.35\" font-family=\"Times,serif\" font-size=\"12.00\">system</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;) -->\n<g id=\"edge38\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M997.51,-237.34C1015.28,-223.24 1038.56,-204.62 1059,-187.9 1065.95,-182.2 1073.31,-176.11 1080.42,-170.17\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;) -->\n<g id=\"node33\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"728\" cy=\"-134.95\" rx=\"52\" ry=\"52\"/>\n<text text-anchor=\"start\" x=\"692.67\" y=\"-149.35\" font-family=\"Times,serif\" font-size=\"12.00\">mitochondirial</text>\n<text text-anchor=\"start\" x=\"716.11\" y=\"-137.35\" font-family=\"Times,serif\" font-size=\"12.00\">large</text>\n<text text-anchor=\"start\" x=\"704\" y=\"-125.35\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n<text text-anchor=\"start\" x=\"710.33\" y=\"-113.35\" font-family=\"Times,serif\" font-size=\"12.00\">subunit</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;) -->\n<g id=\"edge39\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M933.94,-249.6C896.47,-237.8 836.11,-216.32 789,-187.9 781.14,-183.15 773.28,-177.4 765.94,-171.48\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;) -->\n<g id=\"node34\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"834\" cy=\"-134.95\" rx=\"35.97\" ry=\"35.97\"/>\n<text text-anchor=\"start\" x=\"810.67\" y=\"-143.35\" font-family=\"Times,serif\" font-size=\"12.00\">mito&#45;cyto</text>\n<text text-anchor=\"start\" x=\"810\" y=\"-131.35\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n<text text-anchor=\"start\" x=\"818.01\" y=\"-119.35\" font-family=\"Times,serif\" font-size=\"12.00\">cluster</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;) -->\n<g id=\"edge40\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M939.92,-238.76C921.72,-225.04 898.18,-206.34 879,-187.9 871.34,-180.54 863.57,-172.01 856.69,-164.03\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;) -->\n<g id=\"node35\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"924\" cy=\"-134.95\" rx=\"35.97\" ry=\"35.97\"/>\n<text text-anchor=\"start\" x=\"900\" y=\"-143.35\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n<text text-anchor=\"start\" x=\"903.34\" y=\"-131.35\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"921\" y=\"-119.35\" font-family=\"Times,serif\" font-size=\"12.00\">5</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;) -->\n<g id=\"edge41\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M956.7,-226.01C950.35,-208.52 942.63,-187.25 936.28,-169.77\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;) -->\n<g id=\"node36\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1014\" cy=\"-134.95\" rx=\"35.97\" ry=\"35.97\"/>\n<text text-anchor=\"start\" x=\"990\" y=\"-143.35\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n<text text-anchor=\"start\" x=\"993.34\" y=\"-131.35\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"1011\" y=\"-119.35\" font-family=\"Times,serif\" font-size=\"12.00\">2</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;) -->\n<g id=\"edge42\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M981.3,-226.01C987.65,-208.52 995.37,-187.25 1001.72,-169.77\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;) -->\n<g id=\"edge64\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M1125.41,-70.52C1128,-64.36 1131.55,-58.68 1136.35,-54 1162.02,-28.98 1268.76,-21.8 1320.13,-19.77\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"1122.15,-69.22 1122.36,-79.81 1128.8,-71.4 1122.15,-69.22\"/>\n<text text-anchor=\"start\" x=\"1136\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">cotranslation</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n<g id=\"node38\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1428\" cy=\"-388.82\" rx=\"35.98\" ry=\"35.98\"/>\n<text text-anchor=\"start\" x=\"1415.33\" y=\"-403.22\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n<text text-anchor=\"start\" x=\"1409\" y=\"-391.22\" font-family=\"Times,serif\" font-size=\"12.00\">splicing</text>\n<text text-anchor=\"start\" x=\"1407.34\" y=\"-379.22\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n<text text-anchor=\"start\" x=\"1425\" y=\"-367.22\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n<g id=\"edge44\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1436.85,-496.22C1435.03,-474.34 1432.75,-446.99 1430.97,-425.5\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;) -->\n<g id=\"node40\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"839\" cy=\"-535.14\" rx=\"40.38\" ry=\"40.38\"/>\n<text text-anchor=\"start\" x=\"803.67\" y=\"-531.54\" font-family=\"Times,serif\" font-size=\"12.00\">mitochondrion</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;) -->\n<g id=\"edge46\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M964.79,-675.38C957.35,-664.95 948.16,-653.87 938,-645.48 923.9,-633.84 914.83,-639.45 901,-627.48 883.18,-612.05 868.17,-590.36 857.38,-571.99\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;) -->\n<g id=\"node41\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1011\" cy=\"-535.14\" rx=\"36.76\" ry=\"36.76\"/>\n<text text-anchor=\"start\" x=\"988.67\" y=\"-549.54\" font-family=\"Times,serif\" font-size=\"12.00\">subgroup</text>\n<text text-anchor=\"start\" x=\"1006\" y=\"-537.54\" font-family=\"Times,serif\" font-size=\"12.00\">of</text>\n<text text-anchor=\"start\" x=\"987.34\" y=\"-525.54\" font-family=\"Times,serif\" font-size=\"12.00\">metabolic</text>\n<text text-anchor=\"start\" x=\"986.45\" y=\"-513.54\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;) -->\n<g id=\"edge47\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M989.73,-670.64C994.31,-641.67 1000.7,-601.24 1005.28,-572.29\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;) -->\n<g id=\"edge58\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M958.94,-665.7C946.65,-643.79 932.68,-615.96 924.08,-589.48 903.69,-526.71 919.42,-506.17 901,-442.8 897.47,-430.67 891.85,-417.64 887.04,-407.54\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"955.87,-667.38 963.87,-674.32 961.94,-663.91 955.87,-667.38\"/>\n<text text-anchor=\"start\" x=\"924\" y=\"-532.14\" font-family=\"Times,serif\" font-size=\"10.00\">organelles</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;) -->\n<g id=\"edge57\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M835.75,-482.79C836.26,-469.5 838.02,-455.4 842.12,-442.8 846.31,-429.91 854.66,-417.28 862.25,-407.58\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"832.25,-482.63 835.62,-492.68 839.25,-482.72 832.25,-482.63\"/>\n<text text-anchor=\"start\" x=\"842\" y=\"-449.8\" font-family=\"Times,serif\" font-size=\"10.00\">mitochondrion</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n<g id=\"node43\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1202\" cy=\"-707.84\" rx=\"54.34\" ry=\"54.34\"/>\n<text text-anchor=\"start\" x=\"1165.02\" y=\"-716.24\" font-family=\"Times,serif\" font-size=\"12.00\">transmembrane</text>\n<text text-anchor=\"start\" x=\"1180.67\" y=\"-704.24\" font-family=\"Times,serif\" font-size=\"12.00\">transport</text>\n<text text-anchor=\"start\" x=\"1183\" y=\"-692.24\" font-family=\"Times,serif\" font-size=\"12.00\">systems</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n<g id=\"edge49\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1202,-920.35C1202,-878.83 1202,-810.72 1202,-763.04\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;) -->\n<g id=\"edge65\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M1241.85,-929.82C1287.88,-896.74 1363.37,-836.94 1411,-770.19 1528,-606.25 1500,-527.25 1500,-325.84 1500,-325.84 1500,-325.84 1500,-133.95 1500,-72.41 1419.76,-39.61 1375.88,-26.28\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"1239.86,-926.94 1233.73,-935.58 1243.91,-932.65 1239.86,-926.94\"/>\n<text text-anchor=\"start\" x=\"1501\" y=\"-385.82\" font-family=\"Times,serif\" font-size=\"10.00\">secretory</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n<g id=\"node44\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1120\" cy=\"-535.14\" rx=\"54.34\" ry=\"54.34\"/>\n<text text-anchor=\"start\" x=\"1112.33\" y=\"-549.54\" font-family=\"Times,serif\" font-size=\"12.00\">ion</text>\n<text text-anchor=\"start\" x=\"1083.02\" y=\"-537.54\" font-family=\"Times,serif\" font-size=\"12.00\">transmembrane</text>\n<text text-anchor=\"start\" x=\"1098.67\" y=\"-525.54\" font-family=\"Times,serif\" font-size=\"12.00\">transport</text>\n<text text-anchor=\"start\" x=\"1101\" y=\"-513.54\" font-family=\"Times,serif\" font-size=\"12.00\">systems</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n<g id=\"edge50\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1159.76,-672.48C1147.02,-659.76 1134.44,-644.29 1127,-627.48 1121.91,-615.99 1119.3,-602.89 1118.12,-590.28\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)&#45;&gt;(&#39;cell&#39;, &#39;signalling&#39;) -->\n<g id=\"edge62\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)&#45;&gt;(&#39;cell&#39;, &#39;signalling&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M1222.1,-644.46C1232.47,-612.28 1244.35,-575.39 1251.24,-554.03\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"1218.85,-643.13 1219.12,-653.72 1225.51,-645.27 1218.85,-643.13\"/>\n<text text-anchor=\"start\" x=\"1232\" y=\"-614.48\" font-family=\"Times,serif\" font-size=\"10.00\">transport</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;) -->\n<g id=\"edge66\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M1202,-641.27C1202,-609.52 1202,-570.85 1202,-536.14 1202,-536.14 1202,-536.14 1202,-133.95 1202,-74.43 1277.83,-40.99 1320.19,-26.96\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"1198.5,-641.12 1202,-651.12 1205.5,-641.12 1198.5,-641.12\"/>\n<text text-anchor=\"start\" x=\"1202\" y=\"-326.84\" font-family=\"Times,serif\" font-size=\"10.00\">transmembrane</text>\n<text text-anchor=\"start\" x=\"1215.05\" y=\"-316.84\" font-family=\"Times,serif\" font-size=\"10.00\">transport</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;) -->\n<g id=\"edge56\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M1070.98,-490.8C1066.35,-487.28 1061.66,-483.89 1057,-480.8 1007.64,-448.01 944.9,-418.65 908.32,-402.62\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"1068.52,-493.33 1078.55,-496.76 1072.85,-487.83 1068.52,-493.33\"/>\n<text text-anchor=\"start\" x=\"1035.38\" y=\"-454.8\" font-family=\"Times,serif\" font-size=\"10.00\">ion</text>\n<text text-anchor=\"start\" x=\"1024\" y=\"-444.8\" font-family=\"Times,serif\" font-size=\"10.00\">transport</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;translation&#39;, &#39;m&#39;, &#39;e&#39;, &#39;m&#39;, &#39;b&#39;, &#39;r&#39;, &#39;a&#39;, &#39;n&#39;, &#39;e&#39;)_output -->\n<!-- (&#39;cell&#39;, &#39;translation&#39;, &#39;m&#39;, &#39;e&#39;, &#39;m&#39;, &#39;b&#39;, &#39;r&#39;, &#39;a&#39;, &#39;n&#39;, &#39;e&#39;)_output&#45;&gt;(&#39;cell&#39;, &#39;translation&#39;) -->\n<g id=\"edge68\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;translation&#39;, &#39;m&#39;, &#39;e&#39;, &#39;m&#39;, &#39;b&#39;, &#39;r&#39;, &#39;a&#39;, &#39;n&#39;, &#39;e&#39;)_output&#45;&gt;(&#39;cell&#39;, &#39;translation&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M1556.82,-367.71C1553.29,-357.96 1549.56,-345.98 1547.8,-334.84 1544.83,-316.11 1547.76,-294.46 1550.72,-279.64\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"1553.49,-368.79 1560.35,-376.87 1560.02,-366.28 1553.49,-368.79\"/>\n<text text-anchor=\"start\" x=\"1547\" y=\"-321.84\" font-family=\"Times,serif\" font-size=\"10.00\">membrane</text>\n</g>\n</g>\n</svg>\n",
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x12a2daee0>"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 125
  },
  {
   "cell_type": "code",
   "id": "fe3008ad-498e-449f-8521-6ea29fd903f1",
   "metadata": {
    "tags": [],
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:17.408551Z",
     "start_time": "2025-04-06T16:43:17.363604Z"
    }
   },
   "source": [
    "new_process = {\n",
    "    'hypothesized mechanism': {\n",
    "        '_type': 'step',\n",
    "        '_inputs': {\n",
    "            '1': 'Any',\n",
    "            '2': 'Any',\n",
    "        }\n",
    "    }\n",
    "}\n",
    "\n",
    "new_process = replace_regex_recursive(new_process)\n",
    "\n",
    "# plot\n",
    "plot_settingsx = copy.deepcopy(plot_settings)\n",
    "plot_settingsx['label_margin'] = '0.03'\n",
    "plot_settingsx['node_border_colors'] = {\n",
    "    ('hypothesized<br/>mechanism',): 'blue',\n",
    "}\n",
    "plot_settingsx['port_labels'] = False\n",
    "plot_bigraph(new_process, **plot_settingsx, filename='hypothesized_mechanism')"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/hypothesized_mechanism\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n<!-- Generated by graphviz version 9.0.0 (0)\n -->\n<!-- Title: bigraph Pages: 1 -->\n<svg width=\"74pt\" height=\"100pt\"\n viewBox=\"0.00 0.00 75.64 103.08\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 99.08)\">\n<title>bigraph</title>\n<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-99.08 71.64,-99.08 71.64,4 -4,4\"/>\n<!-- (&#39;hypothesized&lt;br/&gt;mechanism&#39;,) -->\n<g id=\"node1\" class=\"node\">\n<title>(&#39;hypothesized&lt;br/&gt;mechanism&#39;,)</title>\n<polygon fill=\"none\" stroke=\"blue\" stroke-width=\"2\" points=\"67.64,-36 0,-36 0,0 67.64,0 67.64,-36\"/>\n<text text-anchor=\"start\" x=\"2.16\" y=\"-20.4\" font-family=\"Times,serif\" font-size=\"12.00\">hypothesized</text>\n<text text-anchor=\"start\" x=\"6.49\" y=\"-8.4\" font-family=\"Times,serif\" font-size=\"12.00\">mechanism</text>\n</g>\n<!-- (&#39;hypothesized&lt;br/&gt;mechanism&#39;, &#39;2&#39;)_input -->\n<!-- (&#39;hypothesized&lt;br/&gt;mechanism&#39;, &#39;2&#39;)_input&#45;&gt;(&#39;hypothesized&lt;br/&gt;mechanism&#39;,) -->\n<g id=\"edge1\" class=\"edge\">\n<title>(&#39;hypothesized&lt;br/&gt;mechanism&#39;, &#39;2&#39;)_input&#45;&gt;(&#39;hypothesized&lt;br/&gt;mechanism&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M17.21,-71.76C19.3,-65.11 22.09,-56.25 24.79,-47.68\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"28.09,-48.84 27.76,-38.26 21.42,-46.74 28.09,-48.84\"/>\n</g>\n<!-- (&#39;hypothesized&lt;br/&gt;mechanism&#39;, &#39;1&#39;)_input -->\n<!-- (&#39;hypothesized&lt;br/&gt;mechanism&#39;, &#39;1&#39;)_input&#45;&gt;(&#39;hypothesized&lt;br/&gt;mechanism&#39;,) -->\n<g id=\"edge2\" class=\"edge\">\n<title>(&#39;hypothesized&lt;br/&gt;mechanism&#39;, &#39;1&#39;)_input&#45;&gt;(&#39;hypothesized&lt;br/&gt;mechanism&#39;,)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M51.26,-71.76C49.06,-65.11 46.13,-56.25 43.3,-47.68\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"46.64,-46.65 40.18,-38.25 39.99,-48.84 46.64,-46.65\"/>\n</g>\n</g>\n</svg>\n",
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x12a393d60>"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 126
  },
  {
   "cell_type": "markdown",
   "id": "ae52f73c-e384-4880-b18f-47b4c25ce203",
   "metadata": {},
   "source": [
    "### Cell structure and function\n",
    "Processes represent their functions, we show a more minimal cell structure with added processes."
   ]
  },
  {
   "cell_type": "code",
   "id": "b39417ce-d9a1-4985-abce-ce22df816897",
   "metadata": {
    "tags": [],
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:17.470889Z",
     "start_time": "2025-04-06T16:43:17.420247Z"
    }
   },
   "source": [
    "core = VisualizeTypes()\n",
    "core.register(key='concentrations', schema='float')\n",
    "core.register(key='sequences', schema='float')\n",
    "core.register(key='membrane', schema={\n",
    "    'transporters': 'concentrations',\n",
    "    'lipids': 'concentrations',\n",
    "    'transmembrane transport': {\n",
    "        '_type': 'process',\n",
    "        '_inputs': {},\n",
    "        '_outputs': {\n",
    "            'transporters': 'concentrations',\n",
    "            'internal': 'concentrations',\n",
    "            'external': 'concentrations'\n",
    "        }\n",
    "    }\n",
    "})\n",
    "\n",
    "core.register(key='cytoplasm', schema={\n",
    "    'metabolites': 'concentrations',\n",
    "    'ribosomal complexes': 'concentrations',\n",
    "    'transcript regulation complex': {\n",
    "        'transcripts': 'concentrations'},\n",
    "    'translation': {\n",
    "        '_type': 'process',\n",
    "        '_outputs': {\n",
    "            'p1': 'concentrations',\n",
    "            'p2': 'concentrations'}}})\n",
    "\n",
    "core.register(\n",
    "    key='nucleoid',\n",
    "    schema={\n",
    "        'chromosome': {\n",
    "            'genes': 'sequences'}})\n",
    "\n",
    "core.register(\n",
    "    key='cell',\n",
    "    schema={\n",
    "        'membrane': 'membrane',\n",
    "        'cytoplasm': 'cytoplasm',\n",
    "        'nucleoid': 'nucleoid'})\n",
    "\n",
    "# state\n",
    "cell_struct_state = {\n",
    "    'cell': {\n",
    "        'membrane': {\n",
    "            'transmembrane transport': {\n",
    "                'outputs': {\n",
    "                    'transporters': ['transporters'],\n",
    "                    'internal': ['..', 'cytoplasm', 'metabolites']}}},\n",
    "        'cytoplasm': {\n",
    "            'translation': {\n",
    "                'outputs': {\n",
    "                    'p1': ['ribosomal complexes'],\n",
    "                    'p2': ['transcript regulation complex', 'transcripts']}}}}}\n",
    "\n",
    "plot_bigraph(\n",
    "    cell_struct_state,\n",
    "    schema={'cell': 'cell'},\n",
    "    core=core,\n",
    "    remove_process_place_edges=True,\n",
    "    out_dir='out',\n",
    "    filename='cell')"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/cell\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n<!-- Generated by graphviz version 9.0.0 (0)\n -->\n<!-- Title: bigraph Pages: 1 -->\n<svg width=\"669pt\" height=\"658pt\"\n viewBox=\"0.00 0.00 687.63 676.98\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 672.98)\">\n<title>bigraph</title>\n<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-672.98 683.63,-672.98 683.63,4 -4,4\"/>\n<!-- (&#39;cell&#39;,) -->\n<g id=\"node1\" class=\"node\">\n<title>(&#39;cell&#39;,)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"372.74\" cy=\"-650.98\" rx=\"18\" ry=\"18\"/>\n<text text-anchor=\"start\" x=\"364.08\" y=\"-647.38\" font-family=\"Times,serif\" font-size=\"12.00\">cell</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;membrane&#39;) -->\n<g id=\"node2\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;membrane&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"137.74\" cy=\"-561.79\" rx=\"34.19\" ry=\"34.19\"/>\n<text text-anchor=\"start\" x=\"112.42\" y=\"-558.19\" font-family=\"Times,serif\" font-size=\"12.00\">membrane</text>\n</g>\n<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;) -->\n<g id=\"edge1\" class=\"edge\">\n<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M355.46,-643.57C316.35,-629.06 220.52,-593.5 170.05,-574.78\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;) -->\n<g id=\"node5\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"372.74\" cy=\"-561.79\" rx=\"33.41\" ry=\"33.41\"/>\n<text text-anchor=\"start\" x=\"348.08\" y=\"-558.19\" font-family=\"Times,serif\" font-size=\"12.00\">cytoplasm</text>\n</g>\n<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;) -->\n<g id=\"edge5\" class=\"edge\">\n<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M372.74,-632.34C372.74,-621.89 372.74,-608.31 372.74,-595.93\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleoid&#39;) -->\n<g id=\"node10\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleoid&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"540.74\" cy=\"-561.79\" rx=\"28.68\" ry=\"28.68\"/>\n<text text-anchor=\"start\" x=\"520.08\" y=\"-558.19\" font-family=\"Times,serif\" font-size=\"12.00\">nucleoid</text>\n</g>\n<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;nucleoid&#39;) -->\n<g id=\"edge11\" class=\"edge\">\n<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;nucleoid&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M389.15,-641.46C418.46,-626.25 480,-594.31 514.99,-576.15\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;membrane&#39;, &#39;transporters&#39;) -->\n<g id=\"node3\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;membrane&#39;, &#39;transporters&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"37.74\" cy=\"-402.83\" rx=\"37.74\" ry=\"37.74\"/>\n<text text-anchor=\"start\" x=\"9.42\" y=\"-399.23\" font-family=\"Times,serif\" font-size=\"12.00\">transporters</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;membrane&#39;)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;, &#39;transporters&#39;) -->\n<g id=\"edge2\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;membrane&#39;)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;, &#39;transporters&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M116.12,-534.14C106.1,-521.3 94.32,-505.5 84.74,-490.6 73.8,-473.6 62.96,-453.83 54.45,-437.47\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;membrane&#39;, &#39;lipids&#39;) -->\n<g id=\"node4\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;membrane&#39;, &#39;lipids&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"113.74\" cy=\"-402.83\" rx=\"20.02\" ry=\"20.02\"/>\n<text text-anchor=\"start\" x=\"100.4\" y=\"-399.23\" font-family=\"Times,serif\" font-size=\"12.00\">lipids</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;membrane&#39;)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;, &#39;lipids&#39;) -->\n<g id=\"edge3\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;membrane&#39;)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;, &#39;lipids&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M132.59,-527.12C127.77,-495.59 120.73,-449.56 116.76,-423.58\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane transport&#39;) -->\n<g id=\"node13\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane transport&#39;)</title>\n<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"242.15,-133.8 115.33,-133.8 115.33,-97.8 242.15,-97.8 242.15,-133.8\"/>\n<text text-anchor=\"start\" x=\"118.93\" y=\"-112.2\" font-family=\"Times,serif\" font-size=\"12.00\">transmembrane transport</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;membrane&#39;)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane transport&#39;) -->\n<!-- (&#39;cell&#39;, &#39;membrane&#39;, &#39;transporters&#39;)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane transport&#39;) -->\n<g id=\"edge14\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;membrane&#39;, &#39;transporters&#39;)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane transport&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M40.67,-352.93C45.88,-302.27 60.36,-222.9 100.53,-167.6 110.53,-153.85 125.48,-142.83 139.64,-134.62\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"37.2,-352.43 39.76,-362.72 44.17,-353.09 37.2,-352.43\"/>\n<text text-anchor=\"start\" x=\"100.74\" y=\"-229.33\" font-family=\"Times,serif\" font-size=\"10.00\">transporters</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;metabolites&#39;) -->\n<g id=\"node6\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;metabolites&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"226.74\" cy=\"-402.83\" rx=\"36.95\" ry=\"36.95\"/>\n<text text-anchor=\"start\" x=\"199.08\" y=\"-399.23\" font-family=\"Times,serif\" font-size=\"12.00\">metabolites</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;metabolites&#39;) -->\n<g id=\"edge6\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;metabolites&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M343.22,-545.13C321.66,-532.58 292.81,-513.34 272.74,-490.6 259.09,-475.13 248.06,-454.9 240.24,-437.87\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;ribosomal complexes&#39;) -->\n<g id=\"node7\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;ribosomal complexes&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"390.74\" cy=\"-232.33\" rx=\"64.73\" ry=\"64.73\"/>\n<text text-anchor=\"start\" x=\"339.58\" y=\"-228.73\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal complexes</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;ribosomal complexes&#39;) -->\n<g id=\"edge7\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;ribosomal complexes&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M363.88,-528.9C360.99,-516.98 358.17,-503.27 356.74,-490.6 348.02,-413.07 341.46,-391.57 356.74,-315.06 358.27,-307.38 360.55,-299.51 363.2,-291.84\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript regulation complex&#39;) -->\n<g id=\"node8\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript regulation complex&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"453.74\" cy=\"-402.83\" rx=\"87.77\" ry=\"87.77\"/>\n<text text-anchor=\"start\" x=\"383.1\" y=\"-399.23\" font-family=\"Times,serif\" font-size=\"12.00\">transcript regulation complex</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript regulation complex&#39;) -->\n<g id=\"edge8\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript regulation complex&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M387.98,-531.27C395.26,-517.15 404.37,-499.51 413.45,-481.9\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;) -->\n<g id=\"node14\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;)</title>\n<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"499.67,-36 441.81,-36 441.81,0 499.67,0 499.67,-36\"/>\n<text text-anchor=\"start\" x=\"445.41\" y=\"-14.4\" font-family=\"Times,serif\" font-size=\"12.00\">translation</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;) -->\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;metabolites&#39;)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane transport&#39;) -->\n<g id=\"edge15\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;metabolites&#39;)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane transport&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M218.75,-354.37C207.89,-289.89 188.99,-177.69 181.72,-134.48\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"215.25,-354.67 220.36,-363.95 222.15,-353.51 215.25,-354.67\"/>\n<text text-anchor=\"start\" x=\"208.74\" y=\"-229.33\" font-family=\"Times,serif\" font-size=\"10.00\">internal</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;ribosomal complexes&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;) -->\n<g id=\"edge16\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;ribosomal complexes&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M396.67,-155.61C400.58,-131.28 407.05,-104.75 417.74,-82 425.82,-64.79 439.81,-48.46 451.39,-36.74\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"393.21,-155.09 395.22,-165.49 400.14,-156.1 393.21,-155.09\"/>\n<text text-anchor=\"start\" x=\"417.74\" y=\"-112.8\" font-family=\"Times,serif\" font-size=\"10.00\">p1</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript regulation complex&#39;, &#39;transcripts&#39;) -->\n<g id=\"node9\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript regulation complex&#39;, &#39;transcripts&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"470.74\" cy=\"-115.8\" rx=\"33.8\" ry=\"33.8\"/>\n<text text-anchor=\"start\" x=\"445.75\" y=\"-112.2\" font-family=\"Times,serif\" font-size=\"12.00\">transcripts</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript regulation complex&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript regulation complex&#39;, &#39;transcripts&#39;) -->\n<g id=\"edge9\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript regulation complex&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript regulation complex&#39;, &#39;transcripts&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M458.96,-314.38C462.26,-258.9 466.36,-190.27 468.75,-150.23\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript regulation complex&#39;, &#39;transcripts&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;) -->\n<g id=\"edge17\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript regulation complex&#39;, &#39;transcripts&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M470.74,-69.6C470.74,-58.01 470.74,-46.2 470.74,-36.91\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"467.24,-69.51 470.74,-79.51 474.24,-69.51 467.24,-69.51\"/>\n<text text-anchor=\"start\" x=\"470.74\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">p2</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleoid&#39;, &#39;chromosome&#39;) -->\n<g id=\"node11\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleoid&#39;, &#39;chromosome&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"638.74\" cy=\"-402.83\" rx=\"40.89\" ry=\"40.89\"/>\n<text text-anchor=\"start\" x=\"607.75\" y=\"-399.23\" font-family=\"Times,serif\" font-size=\"12.00\">chromosome</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleoid&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleoid&#39;, &#39;chromosome&#39;) -->\n<g id=\"edge12\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleoid&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleoid&#39;, &#39;chromosome&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M560.25,-539.69C568.03,-530.76 576.78,-520.02 583.74,-509.6 598.34,-487.75 611.99,-461.54 622.04,-440.69\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleoid&#39;, &#39;chromosome&#39;, &#39;genes&#39;) -->\n<g id=\"node12\" class=\"node\">\n<title>(&#39;cell&#39;, &#39;nucleoid&#39;, &#39;chromosome&#39;, &#39;genes&#39;)</title>\n<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"638.74\" cy=\"-115.8\" rx=\"20.41\" ry=\"20.41\"/>\n<text text-anchor=\"start\" x=\"625.08\" y=\"-112.2\" font-family=\"Times,serif\" font-size=\"12.00\">genes</text>\n</g>\n<!-- (&#39;cell&#39;, &#39;nucleoid&#39;, &#39;chromosome&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleoid&#39;, &#39;chromosome&#39;, &#39;genes&#39;) -->\n<g id=\"edge13\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;nucleoid&#39;, &#39;chromosome&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleoid&#39;, &#39;chromosome&#39;, &#39;genes&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M638.74,-360.99C638.74,-299.11 638.74,-183.87 638.74,-137.2\"/>\n</g>\n<!-- (&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane transport&#39;, &#39;e&#39;, &#39;x&#39;, &#39;t&#39;, &#39;e&#39;, &#39;r&#39;, &#39;n&#39;, &#39;a&#39;, &#39;l&#39;)_output -->\n<!-- (&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane transport&#39;, &#39;e&#39;, &#39;x&#39;, &#39;t&#39;, &#39;e&#39;, &#39;r&#39;, &#39;n&#39;, &#39;a&#39;, &#39;l&#39;)_output&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane transport&#39;) -->\n<g id=\"edge18\" class=\"edge\">\n<title>(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane transport&#39;, &#39;e&#39;, &#39;x&#39;, &#39;t&#39;, &#39;e&#39;, &#39;r&#39;, &#39;n&#39;, &#39;a&#39;, &#39;l&#39;)_output&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane transport&#39;)</title>\n<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M292.45,-377.41C285.02,-324.8 265.86,-202.66 243.74,-167.6 235.41,-154.4 222.39,-143.23 210.18,-134.76\"/>\n<polygon fill=\"black\" stroke=\"black\" points=\"288.96,-377.72 293.8,-387.15 295.89,-376.76 288.96,-377.72\"/>\n<text text-anchor=\"start\" x=\"279.74\" y=\"-229.33\" font-family=\"Times,serif\" font-size=\"10.00\">external</text>\n</g>\n</g>\n</svg>\n",
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x12a5047f0>"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 127
  },
  {
   "cell_type": "markdown",
   "id": "188eb4a6-c526-4b38-a22d-97f7fb8b7244",
   "metadata": {},
   "source": [
    "### Whole-cell *E. coli* model\n",
    "\n",
    "We here plot the processes and wiring of a whole-cell model of *E. coli* from [Macklin et al. Simultaneous cross-evaluation of heterogeneous E. coli datasets via mechanistic simulation. Science. 2020.](https://pubmed.ncbi.nlm.nih.gov/32703847/)"
   ]
  },
  {
   "cell_type": "code",
   "id": "16960d1b-6e35-421e-bb7d-3efe5052e4fc",
   "metadata": {
    "tags": [],
    "ExecuteTime": {
     "end_time": "2025-04-06T16:43:17.550367Z",
     "start_time": "2025-04-06T16:43:17.483364Z"
    }
   },
   "source": [
    "ecoli = { \n",
    "    'chromosome-structure': { \n",
    "        'wires': {\n",
    "            'fragmentBases': ('bulk',),\n",
    "            'molecules': ('bulk',),\n",
    "            'active_tfs': ('bulk',),\n",
    "            'subunits': ('bulk',),\n",
    "            'amino_acids': ('bulk',),\n",
    "            'active_replisomes': ('unique', 'active_replisome'),\n",
    "            'oriCs': ('unique', 'oriC'),\n",
    "            'chromosome_domains': ('unique', 'chromosome_domain'),\n",
    "            'active_RNAPs': ('unique', 'active_RNAP'),\n",
    "            'RNAs': ('unique', 'RNA'),\n",
    "            'active_ribosome': ('unique', 'active_ribosome'),\n",
    "            'full_chromosomes': ('unique', 'full_chromosome'),\n",
    "            'promoters': ('unique', 'promoter'),\n",
    "            'DnaA_boxes': ('unique', 'DnaA_box')\n",
    "        }\n",
    "    },\n",
    "    'metabolism': { \n",
    "        'wires': {\n",
    "            'metabolites': ('bulk',),\n",
    "            'catalysts': ('bulk',),\n",
    "            'kinetics_enzymes': ('bulk',),\n",
    "            'kinetics_substrates': ('bulk',),\n",
    "            'amino_acids': ('bulk',),\n",
    "            'environment': ('environment',),\n",
    "            'amino_acids_total': ('bulk',)\n",
    "        }\n",
    "    },\n",
    "    'tf-binding': { \n",
    "        'wires': {\n",
    "            'promoters': ('unique', 'promoter'),\n",
    "            'active_tfs': ('bulk',),\n",
    "            'active_tfs_total': ('bulk',),\n",
    "            'inactive_tfs_total': ('bulk',),\n",
    "        }\n",
    "    },\n",
    "    'transcript-initiation': { \n",
    "        'wires': {\n",
    "            'environment': ('environment',),\n",
    "            'full_chromosomes': ('unique', 'full_chromosome'),\n",
    "            'RNAs': ('unique', 'RNA'),\n",
    "            'active_RNAPs': ('unique', 'active_RNAP'),\n",
    "            'promoters': ('unique', 'promoter'),\n",
    "            'molecules': ('bulk',),\n",
    "        }\n",
    "    },\n",
    "    'transcript-elongation': { \n",
    "        'wires': {\n",
    "            'environment': ('environment',),\n",
    "            'RNAs': ('unique', 'RNA'),\n",
    "            'active_RNAPs': ('unique', 'active_RNAP'),\n",
    "            'molecules': ('bulk',),\n",
    "            'bulk_RNAs': ('bulk',),\n",
    "            'ntps': ('bulk',),\n",
    "        }\n",
    "    },\n",
    "    'rna-degradation': { \n",
    "        'wires': {\n",
    "            'charged_trna': ('bulk',),\n",
    "            'bulk_RNAs': ('bulk',),\n",
    "            'nmps': ('bulk',),\n",
    "            'fragmentMetabolites': ('bulk',),\n",
    "            'fragmentBases': ('bulk',),\n",
    "            'endoRnases': ('bulk',),\n",
    "            'exoRnases': ('bulk',),\n",
    "            'subunits': ('bulk',),\n",
    "            'molecules': ('bulk',),\n",
    "            'RNAs': ('unique', 'RNA'),\n",
    "            'active_ribosome': ('unique', 'active_ribosome'),\n",
    "        }\n",
    "    },\n",
    "    'polypeptide-initiation': { \n",
    "        'wires': {\n",
    "            'environment': ('environment',),\n",
    "            'active_ribosome': ('unique', 'active_ribosome'),\n",
    "            'RNA': ('unique', 'RNA'),\n",
    "            'subunits': ('bulk',)\n",
    "        }\n",
    "    },\n",
    "    'polypeptide-elongation': { \n",
    "        'wires': {\n",
    "            'environment': ('environment',),\n",
    "            'active_ribosome': ('unique', 'active_ribosome'),\n",
    "            'molecules': ('bulk',),\n",
    "            'monomers': ('bulk',),\n",
    "            'amino_acids': ('bulk',),\n",
    "            'ppgpp_reaction_metabolites': ('bulk',),\n",
    "            'uncharged_trna': ('bulk',),\n",
    "            'charged_trna': ('bulk',),\n",
    "            'charging_molecules': ('bulk',),\n",
    "            'synthetases': ('bulk',),\n",
    "            'subunits': ('bulk',),\n",
    "            'molecules_total': ('bulk',),\n",
    "            'amino_acids_total': ('bulk',),\n",
    "            'charged_trna_total': ('bulk',),\n",
    "            'uncharged_trna_total': ('bulk',)\n",
    "        }\n",
    "    },\n",
    "    'complexation': {\n",
    "        'wires': {\n",
    "            'molecules': ('bulk',), \n",
    "        }\n",
    "    },\n",
    "    'two-component-system': { \n",
    "        'wires': {\n",
    "            'molecules': ('bulk',)\n",
    "        }\n",
    "    },\n",
    "    'equilibrium': {\n",
    "        'wires': {\n",
    "            'molecules': ('bulk',)\n",
    "        }\n",
    "    },\n",
    "    'protein-degradation': { \n",
    "        'wires': {\n",
    "            'metabolites': ('bulk',),\n",
    "            'proteins': ('bulk',)\n",
    "        }\n",
    "    },\n",
    "    'chromosome-replication': { \n",
    "        'wires': {\n",
    "            'replisome_trimers': ('bulk',),\n",
    "            'replisome_monomers': ('bulk',),\n",
    "            'dntps': ('bulk',),\n",
    "            'ppi': ('bulk',),\n",
    "            'active_replisomes': ('unique', 'active_replisome'),\n",
    "            'oriCs': ('unique', 'oriC'),\n",
    "            'chromosome_domains': ('unique', 'chromosome_domain'),\n",
    "            'full_chromosomes': ('unique', 'full_chromosome'),\n",
    "            'environment': ('environment',)\n",
    "        }\n",
    "    },\n",
    "    'unique': {\n",
    "        'chromosome_domain': {},\n",
    "        'full_chromosome': {},\n",
    "        'oriC': {},\n",
    "        'active_replisome': {},\n",
    "        'RNA': {},\n",
    "        'active_ribosome': {},\n",
    "        'DnaA_box': {},\n",
    "        'promoter': {},\n",
    "    },\n",
    "    'bulk': {},\n",
    "    'environment': {},\n",
    "}\n",
    "node_groups = [[ #line up the top nodes for visual effect\n",
    "        ('unique',), ('bulk',), ('environment',), \n",
    "    ],\n",
    "]\n",
    "# plot settings\n",
    "plot_settings_ecoli = {\n",
    "    'remove_process_place_edges': True,\n",
    "    'rankdir': 'RL',\n",
    "    'dpi': '100',\n",
    "    'out_dir': 'out',\n",
    "}\n",
    "fig = plot_bigraph(ecoli, **plot_settings_ecoli, node_groups=node_groups, filename='ecoli')"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/ecoli\n"
     ]
    }
   ],
   "execution_count": 128
  },
  {
   "cell_type": "markdown",
   "id": "54e1102d-521c-49cc-8ed2-be26c2b84c3d",
   "metadata": {
    "tags": []
   },
   "source": [
    "[See E. coli wiring diagram](https://github.com/vivarium-collective/bigraph-viz/blob/main/doc/_static/ecoli.png?raw=true)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
