{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5338cfab-cf15-4c63-a091-c1cbb51bf8dd",
   "metadata": {},
   "source": [
    "# Process Bigraph Schema\n",
    "This tutorial covers Process Bigraph Schema, starting with single `store` and increasing in detail as you scroll towards the bottom. It uses the libraries [bigraph-schema](https://github.com/vivarium-collective/bigraph-schema) for the schema validation and type registry and [bigraph-viz](https://github.com/vivarium-collective/bigraph-viz) for visualization.\n",
    "\n",
    "## Bigraph-schema and Bigraph-viz\n",
    "\n",
    "[Bigraph-schema](https://github.com/vivarium-collective/bigraph-schema) is a serializable type schema for Process Bigraph simulations. From this library, we will use the `validate_schema` method to check our schemas, and `fill` to fill in missing details to prepare the schema for simulation. We will also use the `type_registry` to access different schemas, and register new ones.\n",
    "\n",
    "[Bigraph-viz](https://github.com/vivarium-collective/bigraph-viz) is an easy-to-use plotting tool for compositional bigraph schema, built on top of Graphviz. It reads raw Compositional Bigraph Schemas and generates a figure that can be displayed in a notebook or saved to file. Bigraph-viz's main plotting function is `plot_bigraph` for the bigraph structure. `plot_flow` and `plot_multitimestep` make plots of process orchestration."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "df69e36f-3da8-4bfb-8bc0-0aeb04a6f7d7",
   "metadata": {},
   "source": [
    "### Installation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "63fb7d1a-cd49-4aac-8d4f-31bb54496a51",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-e git+https://github.com/vivarium-collective/bigraph-schema.git@01a8525182a2de039a16223f086852ee602ac3a3#egg=bigraph_schema\n",
      "-e git+https://github.com/vivarium-collective/bigraph-viz.git@c264e1854c88a26c4092eb0097eec472a5aa629f#egg=bigraph_viz\n",
      "-e git+https://github.com/vivarium-collective/process-bigraph.git@830e3ae01fe4e9c6d81ce9c29c47868b2c74a1d1#egg=process_bigraph\n"
     ]
    }
   ],
   "source": [
    "# !pip install bigraph-schema\n",
    "# !pip install bigraph-viz\n",
    "!pip freeze | grep bigraph"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f700afa2-bfee-49bf-a4b1-e23f0540461d",
   "metadata": {},
   "source": [
    "### Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2ab3cb32-e007-45c4-807e-b1b442ac1608",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# from bigraph_viz import plot_bigraph, plot_flow, plot_multitimestep, pf\n",
    "from bigraph_viz.dict_utils import replace_regex_recursive\n",
    "from bigraph_viz import plot_bigraph, TypeSystem\n",
    "\n",
    "plot_settings = {\n",
    "    'remove_process_place_edges': True\n",
    "}\n",
    "save_images = False\n",
    "if save_images:\n",
    "    plot_settings.update({'out_dir': 'out','dpi': '250'})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e716c23-5fbe-4c08-a868-09611ca40ec4",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Stores"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a6e9cc2-e0df-4f73-b4c3-eddb1055b088",
   "metadata": {},
   "source": [
    "### Simple stores\n",
    "A store can be declared in a JSON dictionary with the store's name mapped to its `value`. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d711b952-089a-4938-817d-d1ccad9de56e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/simple_store\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 9.0.0 (0)\n",
       " -->\n",
       "<!-- Title: bigraph Pages: 1 -->\n",
       "<svg width=\"74pt\" height=\"74pt\"\n",
       " viewBox=\"0.00 0.00 76.40 76.40\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 72.4)\">\n",
       "<title>bigraph</title>\n",
       "<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-72.4 72.4,-72.4 72.4,4 -4,4\"/>\n",
       "<!-- (&#39;store1&#39;,) -->\n",
       "<g id=\"node1\" class=\"node\">\n",
       "<title>(&#39;store1&#39;,)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"34.2\" cy=\"-34.2\" rx=\"34.2\" ry=\"34.2\"/>\n",
       "<text text-anchor=\"start\" x=\"8.87\" y=\"-30.6\" font-family=\"Times,serif\" font-size=\"12.00\">store1: 1.0</text>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x10a474cd0>"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "simple_store_state = {\n",
    "    'store1': 1.0,\n",
    "}\n",
    "plot_bigraph(simple_store_state, **plot_settings, show_values=True, filename='simple_store')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "db1cead7-e691-4291-920e-827d69c00727",
   "metadata": {},
   "source": [
    "To include values *and* types in a node label, you can instead have the store name map to a dict with `_value` and `_type` keys."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f3e5fa71-7ffc-4a8f-a1a0-dc547c886f84",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/store\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 9.0.0 (0)\n",
       " -->\n",
       "<!-- Title: bigraph Pages: 1 -->\n",
       "<svg width=\"58pt\" height=\"58pt\"\n",
       " viewBox=\"0.00 0.00 59.66 59.66\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 55.66)\">\n",
       "<title>bigraph</title>\n",
       "<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-55.66 55.66,-55.66 55.66,4 -4,4\"/>\n",
       "<!-- (&#39;store1&#39;,) -->\n",
       "<g id=\"node1\" class=\"node\">\n",
       "<title>(&#39;store1&#39;,)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"25.83\" cy=\"-25.83\" rx=\"25.83\" ry=\"25.83\"/>\n",
       "<text text-anchor=\"start\" x=\"11.16\" y=\"-28.23\" font-family=\"Times,serif\" font-size=\"12.00\">store1</text>\n",
       "<text text-anchor=\"start\" x=\"11.16\" y=\"-16.23\" font-family=\"Times,serif\" font-size=\"12.00\">[float]</text>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x106da06a0>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "typed_store_spec = {\n",
    "    'store1': {\n",
    "        '_value': 1.0, \n",
    "        '_type': 'float',\n",
    "    },\n",
    "}\n",
    "plot_bigraph(typed_store_spec, **plot_settings, show_values=True, show_types=True, filename='store')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "de1011cb-903b-4c15-baed-a62f4afdd6b9",
   "metadata": {},
   "source": [
    "### Hierarchy\n",
    "A hierarchy is a place graph of nested stores. Stores can be placed within stores using JSON dictionary nesting."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0f3f0824-f89d-4c6e-830e-2c5a321f7b33",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/hierarchy\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 9.0.0 (0)\n",
       " -->\n",
       "<!-- Title: bigraph Pages: 1 -->\n",
       "<svg width=\"235pt\" height=\"343pt\"\n",
       " viewBox=\"0.00 0.00 241.78 352.56\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 348.56)\">\n",
       "<title>bigraph</title>\n",
       "<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-348.56 237.78,-348.56 237.78,4 -4,4\"/>\n",
       "<!-- (&#39;store1&#39;,) -->\n",
       "<g id=\"node1\" class=\"node\">\n",
       "<title>(&#39;store1&#39;,)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"150.23\" cy=\"-322.97\" rx=\"21.59\" ry=\"21.59\"/>\n",
       "<text text-anchor=\"start\" x=\"135.57\" y=\"-319.37\" font-family=\"Times,serif\" font-size=\"12.00\">store1</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.1&#39;) -->\n",
       "<g id=\"node2\" class=\"node\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"114.23\" cy=\"-238.47\" rx=\"26.91\" ry=\"26.91\"/>\n",
       "<text text-anchor=\"start\" x=\"95.07\" y=\"-234.87\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;) -->\n",
       "<g id=\"edge5\" class=\"edge\">\n",
       "<title>(&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M141.7,-302.42C136.67,-290.88 130.25,-276.19 124.89,-263.9\"/>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.2&#39;) -->\n",
       "<g id=\"node7\" class=\"node\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.2&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"186.23\" cy=\"-238.47\" rx=\"26.91\" ry=\"26.91\"/>\n",
       "<text text-anchor=\"start\" x=\"167.07\" y=\"-234.87\" font-family=\"Times,serif\" font-size=\"12.00\">store1.2</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.2&#39;) -->\n",
       "<g id=\"edge6\" class=\"edge\">\n",
       "<title>(&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.2&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M158.76,-302.42C163.79,-290.88 170.2,-276.19 175.57,-263.9\"/>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.1&#39;) -->\n",
       "<g id=\"node3\" class=\"node\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"32.23\" cy=\"-143.33\" rx=\"32.23\" ry=\"32.23\"/>\n",
       "<text text-anchor=\"start\" x=\"8.57\" y=\"-139.73\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1.1</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.1&#39;) -->\n",
       "<g id=\"edge2\" class=\"edge\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M96.44,-217.26C83.81,-202.92 66.81,-183.61 53.48,-168.46\"/>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.2&#39;) -->\n",
       "<g id=\"node4\" class=\"node\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.2&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"114.23\" cy=\"-143.33\" rx=\"32.23\" ry=\"32.23\"/>\n",
       "<text text-anchor=\"start\" x=\"90.57\" y=\"-139.73\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1.2</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.2&#39;) -->\n",
       "<g id=\"edge3\" class=\"edge\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.2&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M114.23,-211.03C114.23,-200.27 114.23,-187.74 114.23,-176.46\"/>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;) -->\n",
       "<g id=\"node5\" class=\"node\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"196.23\" cy=\"-143.33\" rx=\"32.23\" ry=\"32.23\"/>\n",
       "<text text-anchor=\"start\" x=\"172.57\" y=\"-139.73\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1.3</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;) -->\n",
       "<g id=\"edge4\" class=\"edge\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M132.02,-217.26C144.65,-202.92 161.65,-183.61 174.98,-168.46\"/>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;, &#39;store1.1.3.1&#39;) -->\n",
       "<g id=\"node6\" class=\"node\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;, &#39;store1.1.3.1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"196.23\" cy=\"-37.55\" rx=\"37.55\" ry=\"37.55\"/>\n",
       "<text text-anchor=\"start\" x=\"168.07\" y=\"-33.95\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1.3.1</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;, &#39;store1.1.3.1&#39;) -->\n",
       "<g id=\"edge1\" class=\"edge\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;, &#39;store1.1.3&#39;, &#39;store1.1.3.1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M196.23,-110.34C196.23,-99.5 196.23,-87.3 196.23,-76.07\"/>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x10a3647f0>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hierarchy_spec = {\n",
    "    'store1': {\n",
    "        'store1.1': {\n",
    "            'store1.1.1': 'Any',\n",
    "            'store1.1.2': 'Any',\n",
    "            'store1.1.3': {\n",
    "                'store1.1.3.1': 'Any',\n",
    "            },\n",
    "        },\n",
    "        'store1.2': 'Any',\n",
    "    },\n",
    "}\n",
    "plot_bigraph(hierarchy_spec, **plot_settings, filename='hierarchy')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c825e07c-8cd5-4df1-8958-aa0f36d61045",
   "metadata": {},
   "source": [
    "## Processes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47ea1a83-aeac-4048-9593-7b30588b0ad8",
   "metadata": {},
   "source": [
    "### Single process\n",
    "Processes require ports, which are declared with the `_ports` key mapped to a dict that has port names matched to types."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7c74237b-0d66-4e8f-a8ca-79174fadad8b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/process\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 9.0.0 (0)\n",
       " -->\n",
       "<!-- Title: bigraph Pages: 1 -->\n",
       "<svg width=\"142pt\" height=\"78pt\"\n",
       " viewBox=\"0.00 0.00 146.26 80.15\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 76.15)\">\n",
       "<title>bigraph</title>\n",
       "<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-76.15 142.26,-76.15 142.26,4 -4,4\"/>\n",
       "<!-- (&#39;process1&#39;,) -->\n",
       "<g id=\"node1\" class=\"node\">\n",
       "<title>(&#39;process1&#39;,)</title>\n",
       "<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"54,-53.58 0,-53.58 0,-17.58 54,-17.58 54,-53.58\"/>\n",
       "<text text-anchor=\"start\" x=\"6.01\" y=\"-31.98\" font-family=\"Times,serif\" font-size=\"12.00\">process1</text>\n",
       "</g>\n",
       "<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;) -->\n",
       "<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)&#45;&gt;(&#39;process1&#39;,) -->\n",
       "<g id=\"edge1\" class=\"edge\">\n",
       "<title>(&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)&#45;&gt;(&#39;process1&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M110.78,-55.47C99.18,-52.69 81.54,-48.45 65.52,-44.6\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"66.8,-41.3 56.26,-42.37 65.17,-48.11 66.8,-41.3\"/>\n",
       "<text text-anchor=\"start\" x=\"72\" y=\"-52.58\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n",
       "</g>\n",
       "<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;) -->\n",
       "<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)&#45;&gt;(&#39;process1&#39;,) -->\n",
       "<g id=\"edge2\" class=\"edge\">\n",
       "<title>(&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)&#45;&gt;(&#39;process1&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M110.78,-16.54C99.29,-19.19 81.88,-23.19 65.99,-26.84\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"65.23,-23.43 56.26,-29.08 66.79,-30.25 65.23,-23.43\"/>\n",
       "<text text-anchor=\"start\" x=\"72\" y=\"-26.58\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x106d5ed90>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "process_spec = {\n",
    "    'process1': {\n",
    "        '_type': 'edge',\n",
    "        '_inputs': {\n",
    "            'port1': 'Any',\n",
    "            'port2': 'Any',\n",
    "        },\n",
    "    },\n",
    "}\n",
    "plot_bigraph(process_spec, **plot_settings, rankdir='RL', filename='process')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "68a2c480-32fe-43d3-b618-8fc6c647b8ca",
   "metadata": {},
   "source": [
    "### Multiple processes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "8c2457a3-fc47-4850-8b9e-9ee98e81567d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/multiple_processes\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 9.0.0 (0)\n",
       " -->\n",
       "<!-- Title: bigraph Pages: 1 -->\n",
       "<svg width=\"266pt\" height=\"114pt\"\n",
       " viewBox=\"0.00 0.00 273.15 117.15\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 113.15)\">\n",
       "<title>bigraph</title>\n",
       "<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-113.15 269.15,-113.15 269.15,4 -4,4\"/>\n",
       "<!-- (&#39;process1&#39;,) -->\n",
       "<g id=\"node1\" class=\"node\">\n",
       "<title>(&#39;process1&#39;,)</title>\n",
       "<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"80.58,-109.15 26.58,-109.15 26.58,-73.15 80.58,-73.15 80.58,-109.15\"/>\n",
       "<text text-anchor=\"start\" x=\"32.58\" y=\"-87.55\" font-family=\"Times,serif\" font-size=\"12.00\">process1</text>\n",
       "</g>\n",
       "<!-- (&#39;process2&#39;,) -->\n",
       "<g id=\"node2\" class=\"node\">\n",
       "<title>(&#39;process2&#39;,)</title>\n",
       "<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"170.58,-109.15 116.58,-109.15 116.58,-73.15 170.58,-73.15 170.58,-109.15\"/>\n",
       "<text text-anchor=\"start\" x=\"122.58\" y=\"-87.55\" font-family=\"Times,serif\" font-size=\"12.00\">process2</text>\n",
       "</g>\n",
       "<!-- (&#39;process3&#39;,) -->\n",
       "<g id=\"node3\" class=\"node\">\n",
       "<title>(&#39;process3&#39;,)</title>\n",
       "<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"251.58,-109.15 197.58,-109.15 197.58,-73.15 251.58,-73.15 251.58,-109.15\"/>\n",
       "<text text-anchor=\"start\" x=\"203.58\" y=\"-87.55\" font-family=\"Times,serif\" font-size=\"12.00\">process3</text>\n",
       "</g>\n",
       "<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;) -->\n",
       "<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)&#45;&gt;(&#39;process1&#39;,) -->\n",
       "<g id=\"edge1\" class=\"edge\">\n",
       "<title>(&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)&#45;&gt;(&#39;process1&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M18.16,-27.29C21.26,-35.43 25.66,-46.13 30.47,-55.15 31.81,-57.68 33.29,-60.26 34.83,-62.82\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"31.78,-64.55 40.1,-71.11 37.69,-60.79 31.78,-64.55\"/>\n",
       "<text text-anchor=\"start\" x=\"30.58\" y=\"-47.15\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n",
       "</g>\n",
       "<!-- (&#39;process2&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;) -->\n",
       "<!-- (&#39;process2&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)&#45;&gt;(&#39;process2&#39;,) -->\n",
       "<g id=\"edge2\" class=\"edge\">\n",
       "<title>(&#39;process2&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)&#45;&gt;(&#39;process2&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M115.64,-27.61C117.72,-35.69 120.76,-46.2 124.47,-55.15 125.44,-57.49 126.52,-59.88 127.66,-62.25\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"124.49,-63.74 132.2,-71.01 130.71,-60.52 124.49,-63.74\"/>\n",
       "<text text-anchor=\"start\" x=\"123.58\" y=\"-47.15\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n",
       "</g>\n",
       "<!-- (&#39;process3&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;) -->\n",
       "<!-- (&#39;process3&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)&#45;&gt;(&#39;process3&#39;,) -->\n",
       "<g id=\"edge3\" class=\"edge\">\n",
       "<title>(&#39;process3&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;1&#39;)&#45;&gt;(&#39;process3&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M205.92,-27.92C205.84,-35.92 206.3,-46.25 208.47,-55.15 209.02,-57.44 209.73,-59.76 210.54,-62.05\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"207.21,-63.16 214.26,-71.07 213.68,-60.49 207.21,-63.16\"/>\n",
       "<text text-anchor=\"start\" x=\"207.58\" y=\"-47.15\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n",
       "</g>\n",
       "<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;) -->\n",
       "<!-- (&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)&#45;&gt;(&#39;process1&#39;,) -->\n",
       "<g id=\"edge4\" class=\"edge\">\n",
       "<title>(&#39;process1&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)&#45;&gt;(&#39;process1&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M56.94,-39.31C56.23,-50.04 55.41,-62.38 54.76,-72.21\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"60.41,-39.81 57.58,-29.6 53.43,-39.35 60.41,-39.81\"/>\n",
       "<text text-anchor=\"start\" x=\"55.58\" y=\"-47.15\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n",
       "</g>\n",
       "<!-- (&#39;process2&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;) -->\n",
       "<!-- (&#39;process2&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)&#45;&gt;(&#39;process2&#39;,) -->\n",
       "<g id=\"edge5\" class=\"edge\">\n",
       "<title>(&#39;process2&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)&#45;&gt;(&#39;process2&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M153.05,-39.04C151.04,-49.88 148.72,-62.39 146.88,-72.33\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"156.48,-39.7 154.86,-29.23 149.6,-38.42 156.48,-39.7\"/>\n",
       "<text text-anchor=\"start\" x=\"151.58\" y=\"-47.15\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n",
       "</g>\n",
       "<!-- (&#39;process3&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;) -->\n",
       "<!-- (&#39;process3&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)&#45;&gt;(&#39;process3&#39;,) -->\n",
       "<g id=\"edge6\" class=\"edge\">\n",
       "<title>(&#39;process3&#39;, &#39;p&#39;, &#39;o&#39;, &#39;r&#39;, &#39;t&#39;, &#39;2&#39;)&#45;&gt;(&#39;process3&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M243.21,-37.99C239.28,-49 234.66,-61.93 231,-72.16\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"246.42,-39.42 246.49,-28.82 239.83,-37.06 246.42,-39.42\"/>\n",
       "<text text-anchor=\"start\" x=\"239.58\" y=\"-47.15\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x10a474f70>"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "process_schema = {\n",
    "    '_type': 'process',\n",
    "    '_inputs': {\n",
    "        'port1': 'Any',\n",
    "    },\n",
    "    '_outputs': {\n",
    "        'port2': 'Any'\n",
    "    },\n",
    "}\n",
    "\n",
    "processes_spec = {\n",
    "    'process1': process_schema,\n",
    "    'process2': process_schema,\n",
    "    'process3': process_schema,\n",
    "}\n",
    "plot_bigraph(processes_spec, **plot_settings, rankdir='BT', filename='multiple_processes')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "15e5f5a6-8dd4-41c3-ba37-33ed9e3e0830",
   "metadata": {},
   "source": [
    "## Wires\n",
    "To connect a process's ports to stores, add wiring with a `_wires` key that maps port names to relative paths in the store hierarchy. The ports used by the wires need to match the ports in the schema."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a6ecad5d-8948-4636-910c-79284a3e8679",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/wires\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 9.0.0 (0)\n",
       " -->\n",
       "<!-- Title: bigraph Pages: 1 -->\n",
       "<svg width=\"168pt\" height=\"129pt\"\n",
       " viewBox=\"0.00 0.00 173.18 133.18\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 129.18)\">\n",
       "<title>bigraph</title>\n",
       "<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-129.18 169.18,-129.18 169.18,4 -4,4\"/>\n",
       "<!-- (&#39;node1&#39;,) -->\n",
       "<g id=\"node1\" class=\"node\">\n",
       "<title>(&#39;node1&#39;,)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"21.59\" cy=\"-103.59\" rx=\"21.59\" ry=\"21.59\"/>\n",
       "<text text-anchor=\"start\" x=\"6.93\" y=\"-99.99\" font-family=\"Times,serif\" font-size=\"12.00\">node1</text>\n",
       "</g>\n",
       "<!-- (&#39;process1&#39;,) -->\n",
       "<g id=\"node4\" class=\"node\">\n",
       "<title>(&#39;process1&#39;,)</title>\n",
       "<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"109.59,-36 55.59,-36 55.59,0 109.59,0 109.59,-36\"/>\n",
       "<text text-anchor=\"start\" x=\"61.6\" y=\"-14.4\" font-family=\"Times,serif\" font-size=\"12.00\">process1</text>\n",
       "</g>\n",
       "<!-- (&#39;node1&#39;,)&#45;&gt;(&#39;process1&#39;,) -->\n",
       "<g id=\"edge1\" class=\"edge\">\n",
       "<title>(&#39;node1&#39;,)&#45;&gt;(&#39;process1&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M34.23,-85.27C42.54,-73.88 53.59,-58.75 63.02,-45.81\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"65.59,-48.24 68.65,-38.1 59.93,-44.11 65.59,-48.24\"/>\n",
       "<text text-anchor=\"start\" x=\"55.59\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n",
       "</g>\n",
       "<!-- (&#39;node2&#39;,) -->\n",
       "<g id=\"node2\" class=\"node\">\n",
       "<title>(&#39;node2&#39;,)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"82.59\" cy=\"-103.59\" rx=\"21.59\" ry=\"21.59\"/>\n",
       "<text text-anchor=\"start\" x=\"67.93\" y=\"-99.99\" font-family=\"Times,serif\" font-size=\"12.00\">node2</text>\n",
       "</g>\n",
       "<!-- (&#39;node2&#39;,)&#45;&gt;(&#39;process1&#39;,) -->\n",
       "<g id=\"edge2\" class=\"edge\">\n",
       "<title>(&#39;node2&#39;,)&#45;&gt;(&#39;process1&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M82.59,-81.06C82.59,-71.08 82.59,-59.07 82.59,-48.28\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"86.09,-48.3 82.59,-38.3 79.09,-48.3 86.09,-48.3\"/>\n",
       "<text text-anchor=\"start\" x=\"82.59\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n",
       "</g>\n",
       "<!-- (&#39;node3&#39;,) -->\n",
       "<g id=\"node3\" class=\"node\">\n",
       "<title>(&#39;node3&#39;,)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"143.59\" cy=\"-103.59\" rx=\"21.59\" ry=\"21.59\"/>\n",
       "<text text-anchor=\"start\" x=\"128.93\" y=\"-99.99\" font-family=\"Times,serif\" font-size=\"12.00\">node3</text>\n",
       "</g>\n",
       "<!-- (&#39;node3&#39;,)&#45;&gt;(&#39;process1&#39;,) -->\n",
       "<g id=\"edge3\" class=\"edge\">\n",
       "<title>(&#39;node3&#39;,)&#45;&gt;(&#39;process1&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M130.95,-85.27C122.64,-73.88 111.59,-58.75 102.16,-45.81\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"105.25,-44.11 96.53,-38.1 99.59,-48.24 105.25,-44.11\"/>\n",
       "<text text-anchor=\"start\" x=\"114.59\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port3</text>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x10a474e20>"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "connected_process_spec = {\n",
    "    'process1': {\n",
    "        '_type': 'edge',\n",
    "        '_inputs': {\n",
    "            'port1': 'float',\n",
    "            'port2': 'int',\n",
    "            'port3': 'int',\n",
    "        },\n",
    "        'inputs': {\n",
    "            'port1': ['node1'],\n",
    "            'port2': ['node2'],\n",
    "            'port3': ['node3'],\n",
    "        }\n",
    "    },\n",
    "    'node1': 'float',  # TODO -- shouldn't these fill?\n",
    "    'node2': 'int',\n",
    "    'node3': 'int',\n",
    "}\n",
    "plot_bigraph(connected_process_spec, **plot_settings, filename='wires')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e037285d-116a-404d-96ec-ba47318a2e8b",
   "metadata": {},
   "source": [
    "### Advanced wiring\n",
    "**TODO**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01b17201-2489-45c8-a06b-9fdc29a714a7",
   "metadata": {},
   "source": [
    "## Composites"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "daf5d428-db23-4985-80ff-6b93d22973aa",
   "metadata": {},
   "source": [
    "### Flat composite\n",
    "A flat composite connects stores with multiple connected processes, without nesting."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "781a84e6-bbe2-4708-b441-47efc611b46f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/flat_composite\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 9.0.0 (0)\n",
       " -->\n",
       "<!-- Title: bigraph Pages: 1 -->\n",
       "<svg width=\"168pt\" height=\"137pt\"\n",
       " viewBox=\"0.00 0.00 172.93 140.82\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 136.82)\">\n",
       "<title>bigraph</title>\n",
       "<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-136.82 168.93,-136.82 168.93,4 -4,4\"/>\n",
       "<!-- (&#39;store1.1&#39;,) -->\n",
       "<g id=\"node1\" class=\"node\">\n",
       "<title>(&#39;store1.1&#39;,)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"138.02\" cy=\"-105.91\" rx=\"26.91\" ry=\"26.91\"/>\n",
       "<text text-anchor=\"start\" x=\"118.86\" y=\"-102.31\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1</text>\n",
       "</g>\n",
       "<!-- (&#39;process1&#39;,) -->\n",
       "<g id=\"node3\" class=\"node\">\n",
       "<title>(&#39;process1&#39;,)</title>\n",
       "<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"54,-118.91 0,-118.91 0,-82.91 54,-82.91 54,-118.91\"/>\n",
       "<text text-anchor=\"start\" x=\"6.01\" y=\"-97.31\" font-family=\"Times,serif\" font-size=\"12.00\">process1</text>\n",
       "</g>\n",
       "<!-- (&#39;store1.1&#39;,)&#45;&gt;(&#39;process1&#39;,) -->\n",
       "<g id=\"edge3\" class=\"edge\">\n",
       "<title>(&#39;store1.1&#39;,)&#45;&gt;(&#39;process1&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M99.1,-104.17C84.49,-103.5 68.12,-102.75 54.77,-102.14\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"98.69,-107.66 108.84,-104.62 99.01,-100.66 98.69,-107.66\"/>\n",
       "<text text-anchor=\"start\" x=\"72\" y=\"-104.91\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n",
       "</g>\n",
       "<!-- (&#39;process2&#39;,) -->\n",
       "<g id=\"node4\" class=\"node\">\n",
       "<title>(&#39;process2&#39;,)</title>\n",
       "<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"54,-53.91 0,-53.91 0,-17.91 54,-17.91 54,-53.91\"/>\n",
       "<text text-anchor=\"start\" x=\"6.01\" y=\"-32.31\" font-family=\"Times,serif\" font-size=\"12.00\">process2</text>\n",
       "</g>\n",
       "<!-- (&#39;store1.1&#39;,)&#45;&gt;(&#39;process2&#39;,) -->\n",
       "<g id=\"edge1\" class=\"edge\">\n",
       "<title>(&#39;store1.1&#39;,)&#45;&gt;(&#39;process2&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M114.43,-91.41C99.94,-82.1 80.79,-69.81 64.21,-59.16\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"66.37,-56.39 56.07,-53.93 62.59,-62.28 66.37,-56.39\"/>\n",
       "<text text-anchor=\"start\" x=\"72\" y=\"-78.91\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n",
       "</g>\n",
       "<!-- (&#39;store1.2&#39;,) -->\n",
       "<g id=\"node2\" class=\"node\">\n",
       "<title>(&#39;store1.2&#39;,)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"138.02\" cy=\"-26.91\" rx=\"26.91\" ry=\"26.91\"/>\n",
       "<text text-anchor=\"start\" x=\"118.86\" y=\"-23.31\" font-family=\"Times,serif\" font-size=\"12.00\">store1.2</text>\n",
       "</g>\n",
       "<!-- (&#39;store1.2&#39;,)&#45;&gt;(&#39;process1&#39;,) -->\n",
       "<g id=\"edge4\" class=\"edge\">\n",
       "<title>(&#39;store1.2&#39;,)&#45;&gt;(&#39;process1&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M100.28,-36.89C90.61,-40.45 80.51,-45.07 72,-50.91 59.8,-59.28 48.8,-71.78 40.85,-82.19\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"101.12,-40.3 109.48,-33.79 98.88,-33.67 101.12,-40.3\"/>\n",
       "<text text-anchor=\"start\" x=\"72\" y=\"-52.91\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n",
       "</g>\n",
       "<!-- (&#39;store1.2&#39;,)&#45;&gt;(&#39;process2&#39;,) -->\n",
       "<g id=\"edge2\" class=\"edge\">\n",
       "<title>(&#39;store1.2&#39;,)&#45;&gt;(&#39;process2&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M111.91,-17.52C99.8,-14.2 85.1,-11.94 72,-14.91 69.83,-15.4 67.65,-16.01 65.47,-16.71\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"64.26,-13.43 56.19,-20.29 66.78,-19.96 64.26,-13.43\"/>\n",
       "<text text-anchor=\"start\" x=\"72\" y=\"-16.91\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x106d5ea00>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "flat_composite_spec = {\n",
    "    'store1.1': 'float',\n",
    "    'store1.2': 'int',\n",
    "    'process1': {\n",
    "        '_type': 'process',\n",
    "        'outputs': {\n",
    "            'port1': ['store1.1'],\n",
    "            'port2': ['store1.2'],\n",
    "        }\n",
    "    },\n",
    "    'process2': {\n",
    "        '_type': 'process',\n",
    "        'inputs': {\n",
    "            'port1': ['store1.1'],\n",
    "            'port2': ['store1.2'],\n",
    "        }\n",
    "    },\n",
    "}\n",
    "plot_bigraph(flat_composite_spec, **plot_settings, rankdir='RL', filename='flat_composite')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "398509c7-cf5c-4cce-a179-3a39752b0dde",
   "metadata": {},
   "source": [
    "### Nested composite\n",
    "A nested composite has store hierarchies with multiple connected processes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "701e759d-7f9b-41bb-911c-a8208c892462",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/nested_composite\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 9.0.0 (0)\n",
       " -->\n",
       "<!-- Title: bigraph Pages: 1 -->\n",
       "<svg width=\"277pt\" height=\"227pt\"\n",
       " viewBox=\"0.00 0.00 285.00 233.00\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 229)\">\n",
       "<title>bigraph</title>\n",
       "<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-229 281,-229 281,4 -4,4\"/>\n",
       "<!-- (&#39;store1&#39;,) -->\n",
       "<g id=\"node1\" class=\"node\">\n",
       "<title>(&#39;store1&#39;,)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"140\" cy=\"-203.41\" rx=\"21.59\" ry=\"21.59\"/>\n",
       "<text text-anchor=\"start\" x=\"125.34\" y=\"-199.81\" font-family=\"Times,serif\" font-size=\"12.00\">store1</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.1&#39;) -->\n",
       "<g id=\"node2\" class=\"node\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"140\" cy=\"-108.91\" rx=\"26.91\" ry=\"26.91\"/>\n",
       "<text text-anchor=\"start\" x=\"120.84\" y=\"-105.31\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;) -->\n",
       "<g id=\"edge1\" class=\"edge\">\n",
       "<title>(&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M140,-180.96C140,-167.72 140,-150.64 140,-136.54\"/>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.2&#39;) -->\n",
       "<g id=\"node3\" class=\"node\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.2&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"68\" cy=\"-108.91\" rx=\"26.91\" ry=\"26.91\"/>\n",
       "<text text-anchor=\"start\" x=\"48.84\" y=\"-105.31\" font-family=\"Times,serif\" font-size=\"12.00\">store1.2</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.2&#39;) -->\n",
       "<g id=\"edge2\" class=\"edge\">\n",
       "<title>(&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;store1.2&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M126.79,-185.44C114.86,-170.11 97.24,-147.48 84.41,-130.99\"/>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;process1&#39;) -->\n",
       "<g id=\"node4\" class=\"node\">\n",
       "<title>(&#39;store1&#39;, &#39;process1&#39;)</title>\n",
       "<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"54,-36 0,-36 0,0 54,0 54,-36\"/>\n",
       "<text text-anchor=\"start\" x=\"6.01\" y=\"-14.4\" font-family=\"Times,serif\" font-size=\"12.00\">process1</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;process1&#39;) -->\n",
       "<g id=\"edge3\" class=\"edge\">\n",
       "<title>(&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;process1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M119,-195.69C93.86,-186.48 52.59,-167.36 32,-135.82 12.36,-105.73 17.49,-61.41 22.52,-36.91\"/>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;process2&#39;) -->\n",
       "<g id=\"node5\" class=\"node\">\n",
       "<title>(&#39;store1&#39;, &#39;process2&#39;)</title>\n",
       "<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"186,-36 132,-36 132,0 186,0 186,-36\"/>\n",
       "<text text-anchor=\"start\" x=\"138.01\" y=\"-14.4\" font-family=\"Times,serif\" font-size=\"12.00\">process2</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;process2&#39;) -->\n",
       "<g id=\"edge4\" class=\"edge\">\n",
       "<title>(&#39;store1&#39;,)&#45;&gt;(&#39;store1&#39;, &#39;process2&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M152.62,-184.75C161.03,-171.87 171.36,-153.62 176,-135.82 185.17,-100.63 185.56,-89.35 177,-54 175.59,-48.16 173.02,-42.19 170.25,-36.89\"/>\n",
       "</g>\n",
       "<!-- (&#39;process3&#39;,) -->\n",
       "<g id=\"node6\" class=\"node\">\n",
       "<title>(&#39;process3&#39;,)</title>\n",
       "<polygon fill=\"none\" stroke=\"black\" stroke-width=\"2\" points=\"277,-126.91 223,-126.91 223,-90.91 277,-90.91 277,-126.91\"/>\n",
       "<text text-anchor=\"start\" x=\"229.01\" y=\"-105.31\" font-family=\"Times,serif\" font-size=\"12.00\">process3</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;,)&#45;&gt;(&#39;process3&#39;,) -->\n",
       "<g id=\"edge7\" class=\"edge\">\n",
       "<title>(&#39;store1&#39;,)&#45;&gt;(&#39;process3&#39;,)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M160.11,-193.87C174.67,-187.08 194.29,-176.6 209,-163.82 217.89,-156.09 226.11,-146.16 232.81,-136.92\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"235.53,-139.14 238.35,-128.92 229.78,-135.15 235.53,-139.14\"/>\n",
       "<text text-anchor=\"start\" x=\"218\" y=\"-155.82\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process1&#39;) -->\n",
       "<g id=\"edge5\" class=\"edge\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M118.75,-91.19C101.82,-77.87 77.88,-59.03 58.69,-43.94\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"60.92,-41.24 50.9,-37.8 56.59,-46.74 60.92,-41.24\"/>\n",
       "<text text-anchor=\"start\" x=\"84\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process2&#39;) -->\n",
       "<g id=\"edge8\" class=\"edge\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.1&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process2&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M147.93,-70.78C150.44,-59.04 153.1,-46.6 155.19,-36.85\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"144.58,-69.72 145.91,-80.23 151.43,-71.19 144.58,-69.72\"/>\n",
       "<text text-anchor=\"start\" x=\"152\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port1</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.2&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process1&#39;) -->\n",
       "<g id=\"edge6\" class=\"edge\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.2&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M47.01,-90.82C39.53,-83.44 31.95,-74.16 27.89,-64 25.91,-59.05 24.93,-53.54 24.54,-48.13\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"28.04,-48.17 24.52,-38.18 21.04,-48.19 28.04,-48.17\"/>\n",
       "<text text-anchor=\"start\" x=\"28\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n",
       "</g>\n",
       "<!-- (&#39;store1&#39;, &#39;store1.2&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process2&#39;) -->\n",
       "<g id=\"edge9\" class=\"edge\">\n",
       "<title>(&#39;store1&#39;, &#39;store1.2&#39;)&#45;&gt;(&#39;store1&#39;, &#39;process2&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M95.51,-81.03C110.13,-66.74 127.71,-49.57 140.66,-36.92\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"93.09,-78.5 88.38,-88 97.98,-83.51 93.09,-78.5\"/>\n",
       "<text text-anchor=\"start\" x=\"122\" y=\"-56\" font-family=\"Times,serif\" font-size=\"10.00\">port2</text>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x106d5e4f0>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nested_composite_spec = {\n",
    "    'store1': {\n",
    "        'store1.1': 'float',\n",
    "        'store1.2': 'int',\n",
    "        'process1': {\n",
    "            '_type': 'process',\n",
    "            'inputs': {\n",
    "                'port1': ['store1.1'],\n",
    "                'port2': ['store1.2'],\n",
    "            }\n",
    "        },\n",
    "        'process2': {\n",
    "            '_type': 'process',\n",
    "            'outputs': {\n",
    "                'port1': ['store1.1'],\n",
    "                'port2': ['store1.2'],\n",
    "            }\n",
    "        },\n",
    "    },\n",
    "    'process3': {\n",
    "        '_type': 'process',\n",
    "        'inputs': {\n",
    "            'port1': ['store1'],\n",
    "        }\n",
    "    }\n",
    "}\n",
    "plot_bigraph(nested_composite_spec, \n",
    "             # **plot_settings, \n",
    "             filename='nested_composite')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c8c1704-2331-44eb-bd05-72e36ccb6858",
   "metadata": {},
   "source": [
    "### Composite process\n",
    "Composite processes are processes with internal stores and internal processes, which can run as their own simulation. The composite process has schema, allowing it to connect to a super-simulation. This allows for improved distributed computation, with composite processes able to run on their own computer and syncronize to the super-simulation via message passing."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "638c5dc7-4cfd-46e4-b66a-8c42ca98ff1d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/composite_process\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 9.0.0 (0)\n",
       " -->\n",
       "<!-- Title: bigraph Pages: 1 -->\n",
       "<svg width=\"655pt\" height=\"370pt\"\n",
       " viewBox=\"0.00 0.00 674.15 380.90\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 376.9)\">\n",
       "<title>bigraph</title>\n",
       "<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-376.9 670.15,-376.9 670.15,4 -4,4\"/>\n",
       "<!-- (&#39;composite_process&#39;,) -->\n",
       "<g id=\"node1\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;,)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"173.23\" cy=\"-314.67\" rx=\"58.23\" ry=\"58.23\"/>\n",
       "<text text-anchor=\"start\" x=\"127.57\" y=\"-311.07\" font-family=\"Times,serif\" font-size=\"12.00\">composite_process</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;store1.1&#39;) -->\n",
       "<g id=\"node2\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;store1.1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"64.23\" cy=\"-191.37\" rx=\"26.91\" ry=\"26.91\"/>\n",
       "<text text-anchor=\"start\" x=\"45.06\" y=\"-187.77\" font-family=\"Times,serif\" font-size=\"12.00\">store1.1</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;,)&#45;&gt;(&#39;composite_process&#39;, &#39;store1.1&#39;) -->\n",
       "<g id=\"edge15\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;,)&#45;&gt;(&#39;composite_process&#39;, &#39;store1.1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M134.18,-270.22C116.47,-250.51 96.38,-228.15 82.29,-212.47\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;store1.2&#39;) -->\n",
       "<g id=\"node3\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;store1.2&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"136.23\" cy=\"-191.37\" rx=\"26.91\" ry=\"26.91\"/>\n",
       "<text text-anchor=\"start\" x=\"117.06\" y=\"-187.77\" font-family=\"Times,serif\" font-size=\"12.00\">store1.2</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;,)&#45;&gt;(&#39;composite_process&#39;, &#39;store1.2&#39;) -->\n",
       "<g id=\"edge16\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;,)&#45;&gt;(&#39;composite_process&#39;, &#39;store1.2&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M156.24,-257.99C151.99,-244.04 147.63,-229.74 144.06,-218.06\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process1&#39;) -->\n",
       "<g id=\"node4\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"210.23\" cy=\"-191.37\" rx=\"29.07\" ry=\"29.07\"/>\n",
       "<text text-anchor=\"start\" x=\"189.23\" y=\"-187.77\" font-family=\"Times,serif\" font-size=\"12.00\">process1</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;,)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;) -->\n",
       "<g id=\"edge17\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;,)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M190.21,-257.99C194.24,-244.79 198.36,-231.27 201.81,-219.96\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process2&#39;) -->\n",
       "<g id=\"node12\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process2&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"475.23\" cy=\"-191.37\" rx=\"29.07\" ry=\"29.07\"/>\n",
       "<text text-anchor=\"start\" x=\"454.23\" y=\"-187.77\" font-family=\"Times,serif\" font-size=\"12.00\">process2</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;,)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;) -->\n",
       "<g id=\"edge18\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;,)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M227.57,-291.85C292.44,-265.79 398.37,-223.24 447.98,-203.31\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;) -->\n",
       "<g id=\"node5\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"47.23\" cy=\"-100.38\" rx=\"25.14\" ry=\"25.14\"/>\n",
       "<text text-anchor=\"start\" x=\"29.56\" y=\"-96.78\" font-family=\"Times,serif\" font-size=\"12.00\">outputs</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process1&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;) -->\n",
       "<g id=\"edge3\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process1&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M187.56,-172.15C182.66,-168.63 177.39,-165.16 172.23,-162.29 134.19,-141.19 119.3,-147.32 81.23,-126.29 76.49,-123.68 71.65,-120.49 67.16,-117.28\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process1&#39;, &#39;inputs&#39;) -->\n",
       "<g id=\"node8\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process1&#39;, &#39;inputs&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"112.23\" cy=\"-100.38\" rx=\"21.6\" ry=\"21.6\"/>\n",
       "<text text-anchor=\"start\" x=\"97.56\" y=\"-96.78\" font-family=\"Times,serif\" font-size=\"12.00\">inputs</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process1&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;, &#39;inputs&#39;) -->\n",
       "<g id=\"edge4\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process1&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;, &#39;inputs&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M188.48,-170.62C170.32,-154.13 144.84,-130.99 128.43,-116.09\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process1&#39;, &#39;address&#39;) -->\n",
       "<g id=\"node9\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process1&#39;, &#39;address&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"177.23\" cy=\"-100.38\" rx=\"25.53\" ry=\"25.53\"/>\n",
       "<text text-anchor=\"start\" x=\"159.23\" y=\"-96.78\" font-family=\"Times,serif\" font-size=\"12.00\">address</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process1&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;, &#39;address&#39;) -->\n",
       "<g id=\"edge5\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process1&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;, &#39;address&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M200.15,-163.18C195.64,-151.02 190.37,-136.81 186.02,-125.08\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process1&#39;, &#39;config&#39;) -->\n",
       "<g id=\"node10\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process1&#39;, &#39;config&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"243.23\" cy=\"-100.38\" rx=\"21.99\" ry=\"21.99\"/>\n",
       "<text text-anchor=\"start\" x=\"228.23\" y=\"-96.78\" font-family=\"Times,serif\" font-size=\"12.00\">config</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process1&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;, &#39;config&#39;) -->\n",
       "<g id=\"edge6\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process1&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;, &#39;config&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M220.31,-163.18C225.24,-149.89 231.07,-134.16 235.62,-121.9\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process1&#39;, &#39;interval&#39;) -->\n",
       "<g id=\"node11\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process1&#39;, &#39;interval&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"309.23\" cy=\"-100.38\" rx=\"25.92\" ry=\"25.92\"/>\n",
       "<text text-anchor=\"start\" x=\"290.9\" y=\"-96.78\" font-family=\"Times,serif\" font-size=\"12.00\">interval</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process1&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;, &#39;interval&#39;) -->\n",
       "<g id=\"edge7\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process1&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;, &#39;interval&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M231.71,-171.06C249,-155.52 273.08,-133.87 289.8,-118.84\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;, &#39;port1&#39;) -->\n",
       "<g id=\"node6\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;, &#39;port1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"19.23\" cy=\"-19.23\" rx=\"19.23\" ry=\"19.23\"/>\n",
       "<text text-anchor=\"start\" x=\"6.56\" y=\"-15.63\" font-family=\"Times,serif\" font-size=\"12.00\">port1</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;, &#39;port1&#39;) -->\n",
       "<g id=\"edge1\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;, &#39;port1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M38.83,-75.63C34.57,-63.61 29.5,-49.25 25.58,-38.17\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;, &#39;port2&#39;) -->\n",
       "<g id=\"node7\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;, &#39;port2&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"75.23\" cy=\"-19.23\" rx=\"19.23\" ry=\"19.23\"/>\n",
       "<text text-anchor=\"start\" x=\"62.56\" y=\"-15.63\" font-family=\"Times,serif\" font-size=\"12.00\">port2</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;, &#39;port2&#39;) -->\n",
       "<g id=\"edge2\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process1&#39;, &#39;outputs&#39;, &#39;port2&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M55.63,-75.63C59.88,-63.61 64.96,-49.25 68.88,-38.17\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;) -->\n",
       "<g id=\"node13\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"378.23\" cy=\"-100.38\" rx=\"25.14\" ry=\"25.14\"/>\n",
       "<text text-anchor=\"start\" x=\"360.56\" y=\"-96.78\" font-family=\"Times,serif\" font-size=\"12.00\">outputs</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process2&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;) -->\n",
       "<g id=\"edge10\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process2&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M453.7,-170.62C436.62,-154.95 413,-133.28 396.75,-118.37\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process2&#39;, &#39;inputs&#39;) -->\n",
       "<g id=\"node16\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process2&#39;, &#39;inputs&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"443.23\" cy=\"-100.38\" rx=\"21.6\" ry=\"21.6\"/>\n",
       "<text text-anchor=\"start\" x=\"428.56\" y=\"-96.78\" font-family=\"Times,serif\" font-size=\"12.00\">inputs</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process2&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;, &#39;inputs&#39;) -->\n",
       "<g id=\"edge11\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process2&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;, &#39;inputs&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M465.45,-163.18C460.64,-149.79 454.93,-133.92 450.51,-121.61\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process2&#39;, &#39;address&#39;) -->\n",
       "<g id=\"node17\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process2&#39;, &#39;address&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"508.23\" cy=\"-100.38\" rx=\"25.53\" ry=\"25.53\"/>\n",
       "<text text-anchor=\"start\" x=\"490.23\" y=\"-96.78\" font-family=\"Times,serif\" font-size=\"12.00\">address</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process2&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;, &#39;address&#39;) -->\n",
       "<g id=\"edge12\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process2&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;, &#39;address&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M485.31,-163.18C489.82,-151.02 495.09,-136.81 499.44,-125.08\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process2&#39;, &#39;config&#39;) -->\n",
       "<g id=\"node18\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process2&#39;, &#39;config&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"574.23\" cy=\"-100.38\" rx=\"21.99\" ry=\"21.99\"/>\n",
       "<text text-anchor=\"start\" x=\"559.23\" y=\"-96.78\" font-family=\"Times,serif\" font-size=\"12.00\">config</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process2&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;, &#39;config&#39;) -->\n",
       "<g id=\"edge13\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process2&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;, &#39;config&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M496.71,-171.06C515.02,-154.6 540.96,-131.28 557.69,-116.24\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process2&#39;, &#39;interval&#39;) -->\n",
       "<g id=\"node19\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process2&#39;, &#39;interval&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"640.23\" cy=\"-100.38\" rx=\"25.92\" ry=\"25.92\"/>\n",
       "<text text-anchor=\"start\" x=\"621.9\" y=\"-96.78\" font-family=\"Times,serif\" font-size=\"12.00\">interval</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process2&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;, &#39;interval&#39;) -->\n",
       "<g id=\"edge14\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process2&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;, &#39;interval&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M502.28,-178.58C529.05,-166.56 570.88,-146.81 605.23,-126.29 609.97,-123.46 614.89,-120.19 619.5,-116.96\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;, &#39;port1&#39;) -->\n",
       "<g id=\"node14\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;, &#39;port1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"350.23\" cy=\"-19.23\" rx=\"19.23\" ry=\"19.23\"/>\n",
       "<text text-anchor=\"start\" x=\"337.56\" y=\"-15.63\" font-family=\"Times,serif\" font-size=\"12.00\">port1</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;, &#39;port1&#39;) -->\n",
       "<g id=\"edge8\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;, &#39;port1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M369.83,-75.63C365.57,-63.61 360.5,-49.25 356.58,-38.17\"/>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;, &#39;port2&#39;) -->\n",
       "<g id=\"node15\" class=\"node\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;, &#39;port2&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"406.23\" cy=\"-19.23\" rx=\"19.23\" ry=\"19.23\"/>\n",
       "<text text-anchor=\"start\" x=\"393.56\" y=\"-15.63\" font-family=\"Times,serif\" font-size=\"12.00\">port2</text>\n",
       "</g>\n",
       "<!-- (&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;, &#39;port2&#39;) -->\n",
       "<g id=\"edge9\" class=\"edge\">\n",
       "<title>(&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;)&#45;&gt;(&#39;composite_process&#39;, &#39;process2&#39;, &#39;outputs&#39;, &#39;port2&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M386.63,-75.63C390.88,-63.61 395.96,-49.25 399.88,-38.17\"/>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x10a3646a0>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "composite_process_spec = {\n",
    "    'composite_process': {\n",
    "        'store1.1': 'any',\n",
    "        'store1.2': 'any',\n",
    "        'process1': {\n",
    "            '_type': 'process',\n",
    "            'outputs': {\n",
    "                'port1': 'store1.1',\n",
    "                'port2': 'store1.2',\n",
    "            }\n",
    "        },\n",
    "        'process2': {\n",
    "            '_type': 'process',\n",
    "            'outputs': {\n",
    "                'port1': 'store1.1',\n",
    "                'port2': 'store1.2',\n",
    "            }\n",
    "        },\n",
    "        '_inputs': {\n",
    "            'port1': 'Any', \n",
    "            'port2': 'Any', \n",
    "        },\n",
    "        # '_tunnels': {\n",
    "        #     'port1': 'store1.1',\n",
    "        #     'port2': 'store1.2',\n",
    "        # }\n",
    "    }\n",
    "}\n",
    "plot_bigraph(composite_process_spec, **plot_settings, filename='composite_process')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "847cef26-65a1-4da6-9dd3-eb5839277970",
   "metadata": {},
   "source": [
    "## Orchestration"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e76c97a8-3037-464a-929e-ba4414c73c34",
   "metadata": {},
   "outputs": [],
   "source": [
    "plot_settings2={}\n",
    "if save_images:\n",
    "    plot_settings2.update({\n",
    "        'out_dir': 'out',\n",
    "        'dpi': '250'\n",
    "    })"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "48975890-06a4-462a-9b1d-6bddd51779fe",
   "metadata": {},
   "source": [
    "### Multi-timestepping\n",
    "\n",
    "Temporal processes can be run by multi-timestepping, with each process updating the state according to its `sync_state` state, which it can update adaptively."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f0eff7b0-f0e2-4d56-8b16-ef0fc1ce5aba",
   "metadata": {},
   "outputs": [],
   "source": [
    "# multitimestep_spec = {\n",
    "#     'temporal process2': {\n",
    "#         '_ports': {'port1': 'Any'},\n",
    "#         'wires': {'port1': 'state'},\n",
    "#         '_sync_step': 1.0,\n",
    "#     },\n",
    "#     'temporal process1': {\n",
    "#         '_ports': {'port1': 'Any'},\n",
    "#         'wires': {'port1': 'state'},\n",
    "#         '_sync_step': 0.5,\n",
    "#     },\n",
    "# }\n",
    "# multitimestep_spec = replace_regex_recursive(multitimestep_spec)\n",
    "# plot_multitimestep(multitimestep_spec, total_time=2.0, **plot_settings2, filename='multitimestep')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a00cf677-ebac-4309-8ac9-a5b2855724af",
   "metadata": {},
   "source": [
    "### Flows\n",
    "\n",
    "A directed acyclic graph (DAG)  a type of workflow that declares the dependencies between processes. A flow run is not temporal, and the order is entirely based on dependencies. It can triggered between time steps, running the steps in the order as determined by their dependencies."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "db45c99b-b97b-47bc-a73e-c669882f3969",
   "metadata": {},
   "outputs": [],
   "source": [
    "# flow = {\n",
    "#     'step1': {\n",
    "#         '_type': 'step_process',\n",
    "#         '_ports': {},\n",
    "#         '_depends_on': [], \n",
    "#     },\n",
    "#     'step2': {\n",
    "#         '_type': 'step_process',\n",
    "#         '_ports': {},\n",
    "#         '_depends_on': 'step1', \n",
    "#     },\n",
    "#     'step3': {\n",
    "#         '_type': 'step_process',\n",
    "#         '_ports': {},\n",
    "#         '_depends_on': [], \n",
    "#     },\n",
    "#     'step4': {\n",
    "#         '_type': 'step_process',\n",
    "#         '_ports': {},\n",
    "#         '_depends_on': ['step2', 'step3'], \n",
    "#     },\n",
    "# }\n",
    "\n",
    "# plot_flow(flow, **plot_settings2, filename='flow')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5882581a-c5ef-40b5-acbd-efddad269807",
   "metadata": {},
   "source": [
    "## Examples"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff81178f-36e4-4200-982a-ed8155b87934",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Multiscale map of a cell\n",
    "\n",
    "Bigraphs can represent the hierarchical nesting of cellular structures. Below is a multiscale map of a cell from [Qin et al. A multi-scale map of cell structure fusing protein images and interactions. Nature. 2021.](https://pubmed.ncbi.nlm.nih.gov/34819669/)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "47bdf8e5-0bdd-4cac-b57e-2449c75fa178",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/cell_hierarchy\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 9.0.0 (0)\n",
       " -->\n",
       "<!-- Title: bigraph Pages: 1 -->\n",
       "<svg width=\"667pt\" height=\"700pt\"\n",
       " viewBox=\"0.00 0.00 686.09 720.00\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(0.629041 0.629041) rotate(0) translate(4 1108.81)\">\n",
       "<title>bigraph</title>\n",
       "<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-1108.81 1056.4,-1108.81 1056.4,4 -4,4\"/>\n",
       "<!-- (&#39;cell&#39;,) -->\n",
       "<g id=\"node1\" class=\"node\">\n",
       "<title>(&#39;cell&#39;,)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"424.39\" cy=\"-1086.81\" rx=\"18\" ry=\"18\"/>\n",
       "<text text-anchor=\"start\" x=\"415.73\" y=\"-1083.21\" font-family=\"Times,serif\" font-size=\"12.00\">cell</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;) -->\n",
       "<g id=\"node2\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"286.39\" cy=\"-999.39\" rx=\"33.41\" ry=\"33.41\"/>\n",
       "<text text-anchor=\"start\" x=\"261.73\" y=\"-995.79\" font-family=\"Times,serif\" font-size=\"12.00\">cytoplasm</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;) -->\n",
       "<g id=\"edge42\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M408.83,-1076.17C386.14,-1062.13 343.56,-1035.78 315.09,-1018.15\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;) -->\n",
       "<g id=\"node25\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"654.39\" cy=\"-999.39\" rx=\"25.92\" ry=\"25.92\"/>\n",
       "<text text-anchor=\"start\" x=\"636.07\" y=\"-995.79\" font-family=\"Times,serif\" font-size=\"12.00\">nucleus</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;) -->\n",
       "<g id=\"edge43\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M441.68,-1079.39C482.11,-1064.37 582.95,-1026.93 629.78,-1009.54\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"node3\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"115.39\" cy=\"-875.88\" rx=\"39.81\" ry=\"39.81\"/>\n",
       "<text text-anchor=\"start\" x=\"93.41\" y=\"-878.28\" font-family=\"Times,serif\" font-size=\"12.00\">secretory</text>\n",
       "<text text-anchor=\"start\" x=\"90.84\" y=\"-866.28\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"edge20\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M256.74,-982.85C233.07,-969.92 199.5,-950.39 172.39,-929.98 162.45,-922.5 152.3,-913.54 143.36,-905.13\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"node6\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"227.39\" cy=\"-875.88\" rx=\"46.09\" ry=\"46.09\"/>\n",
       "<text text-anchor=\"start\" x=\"198.4\" y=\"-878.28\" font-family=\"Times,serif\" font-size=\"12.00\">cytoplasmic</text>\n",
       "<text text-anchor=\"start\" x=\"202.84\" y=\"-866.28\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"edge21\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M271.81,-968.36C264.44,-953.18 255.38,-934.53 247.37,-918.03\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node23\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"345.39\" cy=\"-875.88\" rx=\"54.1\" ry=\"54.1\"/>\n",
       "<text text-anchor=\"start\" x=\"310.74\" y=\"-890.28\" font-family=\"Times,serif\" font-size=\"12.00\">transcriptional</text>\n",
       "<text text-anchor=\"start\" x=\"321.07\" y=\"-878.28\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n",
       "<text text-anchor=\"start\" x=\"324.73\" y=\"-866.28\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"329.73\" y=\"-854.28\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge22\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M300.98,-968.36C307.26,-955.43 314.76,-939.98 321.8,-925.49\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n",
       "<g id=\"node4\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"89.39\" cy=\"-728.39\" rx=\"57.39\" ry=\"57.39\"/>\n",
       "<text text-anchor=\"start\" x=\"52.41\" y=\"-736.79\" font-family=\"Times,serif\" font-size=\"12.00\">transmembrane</text>\n",
       "<text text-anchor=\"start\" x=\"68.07\" y=\"-724.79\" font-family=\"Times,serif\" font-size=\"12.00\">transport</text>\n",
       "<text text-anchor=\"start\" x=\"70.39\" y=\"-712.79\" font-family=\"Times,serif\" font-size=\"12.00\">systems</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n",
       "<g id=\"edge2\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M108.43,-835.89C105.67,-820.49 102.46,-802.49 99.46,-785.72\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n",
       "<g id=\"node5\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"57.39\" cy=\"-569.58\" rx=\"57.39\" ry=\"57.39\"/>\n",
       "<text text-anchor=\"start\" x=\"49.73\" y=\"-583.98\" font-family=\"Times,serif\" font-size=\"12.00\">ion</text>\n",
       "<text text-anchor=\"start\" x=\"20.41\" y=\"-571.98\" font-family=\"Times,serif\" font-size=\"12.00\">transmembrane</text>\n",
       "<text text-anchor=\"start\" x=\"36.07\" y=\"-559.98\" font-family=\"Times,serif\" font-size=\"12.00\">transport</text>\n",
       "<text text-anchor=\"start\" x=\"38.39\" y=\"-547.98\" font-family=\"Times,serif\" font-size=\"12.00\">systems</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n",
       "<g id=\"edge1\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M77.93,-671.22C74.97,-656.72 71.79,-641.11 68.83,-626.62\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"node7\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"227.39\" cy=\"-728.39\" rx=\"46.09\" ry=\"46.09\"/>\n",
       "<text text-anchor=\"start\" x=\"205.06\" y=\"-742.79\" font-family=\"Times,serif\" font-size=\"12.00\">subgroup</text>\n",
       "<text text-anchor=\"start\" x=\"222.4\" y=\"-730.79\" font-family=\"Times,serif\" font-size=\"12.00\">of</text>\n",
       "<text text-anchor=\"start\" x=\"198.4\" y=\"-718.79\" font-family=\"Times,serif\" font-size=\"12.00\">cytoplasmic</text>\n",
       "<text text-anchor=\"start\" x=\"202.84\" y=\"-706.79\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"edge18\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M227.39,-828.85C227.39,-811.7 227.39,-792.25 227.39,-775.12\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"node8\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"172.39\" cy=\"-569.58\" rx=\"39.81\" ry=\"39.81\"/>\n",
       "<text text-anchor=\"start\" x=\"148.74\" y=\"-571.98\" font-family=\"Times,serif\" font-size=\"12.00\">metabolic</text>\n",
       "<text text-anchor=\"start\" x=\"147.84\" y=\"-559.98\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"edge16\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M212.22,-684.12C203.86,-660.28 193.58,-630.97 185.56,-608.11\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node11\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"295.39\" cy=\"-569.58\" rx=\"65.41\" ry=\"65.41\"/>\n",
       "<text text-anchor=\"start\" x=\"252.74\" y=\"-577.98\" font-family=\"Times,serif\" font-size=\"12.00\">ribonucleoprotein</text>\n",
       "<text text-anchor=\"start\" x=\"274.73\" y=\"-565.98\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"279.73\" y=\"-553.98\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge17\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M245.62,-685.37C252.94,-668.49 261.51,-648.73 269.45,-630.4\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"node9\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"73.39\" cy=\"-422.15\" rx=\"39.81\" ry=\"39.81\"/>\n",
       "<text text-anchor=\"start\" x=\"51.06\" y=\"-436.55\" font-family=\"Times,serif\" font-size=\"12.00\">subgroup</text>\n",
       "<text text-anchor=\"start\" x=\"68.4\" y=\"-424.55\" font-family=\"Times,serif\" font-size=\"12.00\">of</text>\n",
       "<text text-anchor=\"start\" x=\"49.74\" y=\"-412.55\" font-family=\"Times,serif\" font-size=\"12.00\">metabolic</text>\n",
       "<text text-anchor=\"start\" x=\"48.84\" y=\"-400.55\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"edge3\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M149.93,-535.58C133.73,-511.78 111.95,-479.79 95.77,-456.02\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;) -->\n",
       "<g id=\"node10\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"177.39\" cy=\"-422.15\" rx=\"46.02\" ry=\"46.02\"/>\n",
       "<text text-anchor=\"start\" x=\"142.07\" y=\"-418.55\" font-family=\"Times,serif\" font-size=\"12.00\">mitochondrion</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;) -->\n",
       "<g id=\"edge4\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M173.76,-528.84C174.4,-510.34 175.16,-488.22 175.82,-469.01\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"node12\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"291.39\" cy=\"-422.15\" rx=\"41.38\" ry=\"41.38\"/>\n",
       "<text text-anchor=\"start\" x=\"278.73\" y=\"-436.55\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n",
       "<text text-anchor=\"start\" x=\"265.73\" y=\"-424.55\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n",
       "<text text-anchor=\"start\" x=\"270.73\" y=\"-412.55\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"288.39\" y=\"-400.55\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"edge14\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M293.61,-503.52C293.24,-490.21 292.86,-476.55 292.53,-464.36\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;) -->\n",
       "<g id=\"node14\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"394.39\" cy=\"-422.15\" rx=\"43.74\" ry=\"43.74\"/>\n",
       "<text text-anchor=\"start\" x=\"372.06\" y=\"-430.55\" font-family=\"Times,serif\" font-size=\"12.00\">ribosome</text>\n",
       "<text text-anchor=\"start\" x=\"369.06\" y=\"-418.55\" font-family=\"Times,serif\" font-size=\"12.00\">biogenesis</text>\n",
       "<text text-anchor=\"start\" x=\"367.06\" y=\"-406.55\" font-family=\"Times,serif\" font-size=\"12.00\">community</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;) -->\n",
       "<g id=\"edge15\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M332.26,-514.43C344.79,-496.01 358.49,-475.9 369.82,-459.25\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"node13\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"291.39\" cy=\"-301.09\" rx=\"39.03\" ry=\"39.03\"/>\n",
       "<text text-anchor=\"start\" x=\"278.73\" y=\"-315.49\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n",
       "<text text-anchor=\"start\" x=\"272.4\" y=\"-303.49\" font-family=\"Times,serif\" font-size=\"12.00\">splicing</text>\n",
       "<text text-anchor=\"start\" x=\"270.73\" y=\"-291.49\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"288.39\" y=\"-279.49\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"edge5\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M291.39,-379.84C291.39,-367.22 291.39,-353.38 291.39,-340.92\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;) -->\n",
       "<g id=\"node15\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"387.39\" cy=\"-301.09\" rx=\"30.65\" ry=\"30.65\"/>\n",
       "<text text-anchor=\"start\" x=\"365.06\" y=\"-297.49\" font-family=\"Times,serif\" font-size=\"12.00\">ribosome</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;) -->\n",
       "<g id=\"edge12\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M391.83,-377.49C390.95,-362.49 389.98,-346.05 389.17,-332.35\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"node22\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"475.39\" cy=\"-301.09\" rx=\"39.03\" ry=\"39.03\"/>\n",
       "<text text-anchor=\"start\" x=\"458.07\" y=\"-315.49\" font-family=\"Times,serif\" font-size=\"12.00\">nucleic</text>\n",
       "<text text-anchor=\"start\" x=\"465.4\" y=\"-303.49\" font-family=\"Times,serif\" font-size=\"12.00\">acid</text>\n",
       "<text text-anchor=\"start\" x=\"457.06\" y=\"-291.49\" font-family=\"Times,serif\" font-size=\"12.00\">binding</text>\n",
       "<text text-anchor=\"start\" x=\"454.73\" y=\"-279.49\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"edge13\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M418.92,-385.1C429.91,-368.95 442.76,-350.05 453.43,-334.37\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;) -->\n",
       "<g id=\"node16\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"387.39\" cy=\"-187.03\" rx=\"39.03\" ry=\"39.03\"/>\n",
       "<text text-anchor=\"start\" x=\"363.4\" y=\"-189.43\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n",
       "<text text-anchor=\"start\" x=\"369.73\" y=\"-177.43\" font-family=\"Times,serif\" font-size=\"12.00\">subunit</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;) -->\n",
       "<g id=\"edge11\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M387.39,-269.47C387.39,-256.38 387.39,-240.95 387.39,-227.05\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;) -->\n",
       "<g id=\"node17\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"162.39\" cy=\"-56\" rx=\"55.05\" ry=\"55.05\"/>\n",
       "<text text-anchor=\"start\" x=\"127.07\" y=\"-70.4\" font-family=\"Times,serif\" font-size=\"12.00\">mitochondirial</text>\n",
       "<text text-anchor=\"start\" x=\"150.51\" y=\"-58.4\" font-family=\"Times,serif\" font-size=\"12.00\">large</text>\n",
       "<text text-anchor=\"start\" x=\"138.4\" y=\"-46.4\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n",
       "<text text-anchor=\"start\" x=\"144.73\" y=\"-34.4\" font-family=\"Times,serif\" font-size=\"12.00\">subunit</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;) -->\n",
       "<g id=\"edge6\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M350.4,-172.46C316.76,-159.32 266.4,-137.67 226.39,-112.01 218.38,-106.86 210.31,-100.79 202.73,-94.58\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;) -->\n",
       "<g id=\"node18\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"274.39\" cy=\"-56\" rx=\"39.03\" ry=\"39.03\"/>\n",
       "<text text-anchor=\"start\" x=\"251.07\" y=\"-64.4\" font-family=\"Times,serif\" font-size=\"12.00\">mito&#45;cyto</text>\n",
       "<text text-anchor=\"start\" x=\"250.4\" y=\"-52.4\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n",
       "<text text-anchor=\"start\" x=\"258.4\" y=\"-40.4\" font-family=\"Times,serif\" font-size=\"12.00\">cluster</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;) -->\n",
       "<g id=\"edge7\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M361.75,-156.87C349.78,-143.25 335.35,-126.81 322.39,-112.01 315.13,-103.71 307.26,-94.7 300.04,-86.42\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;) -->\n",
       "<g id=\"node19\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"387.39\" cy=\"-56\" rx=\"56\" ry=\"56\"/>\n",
       "<text text-anchor=\"start\" x=\"352.07\" y=\"-82.4\" font-family=\"Times,serif\" font-size=\"12.00\">cotranslational</text>\n",
       "<text text-anchor=\"start\" x=\"370.4\" y=\"-70.4\" font-family=\"Times,serif\" font-size=\"12.00\">protein</text>\n",
       "<text text-anchor=\"start\" x=\"366.18\" y=\"-58.4\" font-family=\"Times,serif\" font-size=\"12.00\">targeting</text>\n",
       "<text text-anchor=\"start\" x=\"382.73\" y=\"-46.4\" font-family=\"Times,serif\" font-size=\"12.00\">to</text>\n",
       "<text text-anchor=\"start\" x=\"362.07\" y=\"-34.4\" font-family=\"Times,serif\" font-size=\"12.00\">membrane</text>\n",
       "<text text-anchor=\"start\" x=\"370.73\" y=\"-22.4\" font-family=\"Times,serif\" font-size=\"12.00\">system</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;) -->\n",
       "<g id=\"edge8\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M387.39,-147.32C387.39,-136.48 387.39,-124.48 387.39,-112.86\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;) -->\n",
       "<g id=\"node20\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"500.39\" cy=\"-56\" rx=\"39.03\" ry=\"39.03\"/>\n",
       "<text text-anchor=\"start\" x=\"476.4\" y=\"-64.4\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n",
       "<text text-anchor=\"start\" x=\"479.73\" y=\"-52.4\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"497.39\" y=\"-40.4\" font-family=\"Times,serif\" font-size=\"12.00\">5</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;) -->\n",
       "<g id=\"edge9\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M413.04,-156.87C425.01,-143.25 439.44,-126.81 452.39,-112.01 459.66,-103.71 467.53,-94.7 474.75,-86.42\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;) -->\n",
       "<g id=\"node21\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"596.39\" cy=\"-56\" rx=\"39.03\" ry=\"39.03\"/>\n",
       "<text text-anchor=\"start\" x=\"572.4\" y=\"-64.4\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n",
       "<text text-anchor=\"start\" x=\"575.73\" y=\"-52.4\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"593.39\" y=\"-40.4\" font-family=\"Times,serif\" font-size=\"12.00\">2</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;) -->\n",
       "<g id=\"edge10\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M424.97,-173.89C459.24,-161.68 510.21,-140.56 548.39,-112.01 557.33,-105.33 565.79,-96.74 573.05,-88.35\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;) -->\n",
       "<g id=\"node24\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"345.39\" cy=\"-728.39\" rx=\"54.1\" ry=\"54.1\"/>\n",
       "<text text-anchor=\"start\" x=\"310.74\" y=\"-736.79\" font-family=\"Times,serif\" font-size=\"12.00\">transcriptional</text>\n",
       "<text text-anchor=\"start\" x=\"321.07\" y=\"-724.79\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n",
       "<text text-anchor=\"start\" x=\"319.74\" y=\"-712.79\" font-family=\"Times,serif\" font-size=\"12.00\">complexes</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;) -->\n",
       "<g id=\"edge19\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M345.39,-821.12C345.39,-808.81 345.39,-795.73 345.39,-783.42\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;) -->\n",
       "<g id=\"node26\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"489.39\" cy=\"-875.88\" rx=\"38.55\" ry=\"38.55\"/>\n",
       "<text text-anchor=\"start\" x=\"476.4\" y=\"-884.28\" font-family=\"Times,serif\" font-size=\"12.00\">DNA</text>\n",
       "<text text-anchor=\"start\" x=\"465.74\" y=\"-872.28\" font-family=\"Times,serif\" font-size=\"12.00\">metabolic</text>\n",
       "<text text-anchor=\"start\" x=\"467.06\" y=\"-860.28\" font-family=\"Times,serif\" font-size=\"12.00\">assembly</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;) -->\n",
       "<g id=\"edge36\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M630.62,-987.22C605.97,-975.03 566.98,-954.03 537.39,-929.98 529,-923.17 520.77,-914.85 513.56,-906.84\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;) -->\n",
       "<g id=\"node27\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"576.39\" cy=\"-875.88\" rx=\"30.06\" ry=\"30.06\"/>\n",
       "<text text-anchor=\"start\" x=\"558.74\" y=\"-878.28\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n",
       "<text text-anchor=\"start\" x=\"561.4\" y=\"-866.28\" font-family=\"Times,serif\" font-size=\"12.00\">lumen</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;) -->\n",
       "<g id=\"edge37\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M640.45,-976.67C627.04,-955.78 606.78,-924.22 592.65,-902.2\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;) -->\n",
       "<g id=\"node35\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"654.39\" cy=\"-875.88\" rx=\"30.06\" ry=\"30.06\"/>\n",
       "<text text-anchor=\"start\" x=\"636.74\" y=\"-878.28\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n",
       "<text text-anchor=\"start\" x=\"642.39\" y=\"-866.28\" font-family=\"Times,serif\" font-size=\"12.00\">body</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;) -->\n",
       "<g id=\"edge38\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M654.39,-972.63C654.39,-953.4 654.39,-927.08 654.39,-906.9\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"node36\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"753.39\" cy=\"-875.88\" rx=\"50.8\" ry=\"50.8\"/>\n",
       "<text text-anchor=\"start\" x=\"721.07\" y=\"-878.28\" font-family=\"Times,serif\" font-size=\"12.00\">nucleo&#45;plasm</text>\n",
       "<text text-anchor=\"start\" x=\"750.39\" y=\"-866.28\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"edge39\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M670.74,-978.33C684.47,-961.48 704.5,-936.9 721.4,-916.15\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;) -->\n",
       "<g id=\"node41\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"873.39\" cy=\"-875.88\" rx=\"50.8\" ry=\"50.8\"/>\n",
       "<text text-anchor=\"start\" x=\"841.07\" y=\"-878.28\" font-family=\"Times,serif\" font-size=\"12.00\">nucleo&#45;plasm</text>\n",
       "<text text-anchor=\"start\" x=\"870.39\" y=\"-866.28\" font-family=\"Times,serif\" font-size=\"12.00\">2</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;) -->\n",
       "<g id=\"edge40\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M679.57,-990.36C711.99,-979.41 769.37,-957.92 813.39,-929.98 821.5,-924.84 829.59,-918.61 837.09,-912.26\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;) -->\n",
       "<g id=\"node43\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"996.39\" cy=\"-875.88\" rx=\"54.1\" ry=\"54.1\"/>\n",
       "<text text-anchor=\"start\" x=\"978.74\" y=\"-884.28\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n",
       "<text text-anchor=\"start\" x=\"961.74\" y=\"-872.28\" font-family=\"Times,serif\" font-size=\"12.00\">transcriptional</text>\n",
       "<text text-anchor=\"start\" x=\"978.4\" y=\"-860.28\" font-family=\"Times,serif\" font-size=\"12.00\">speckle</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;) -->\n",
       "<g id=\"edge41\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M680.78,-995.97C731.99,-990.43 847.37,-973.72 933.39,-929.98 941.68,-925.77 949.82,-920.27 957.36,-914.42\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;) -->\n",
       "<g id=\"node28\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"561.39\" cy=\"-728.39\" rx=\"31.44\" ry=\"31.44\"/>\n",
       "<text text-anchor=\"start\" x=\"538.4\" y=\"-724.79\" font-family=\"Times,serif\" font-size=\"12.00\">nucleolus</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;) -->\n",
       "<g id=\"edge28\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M573.32,-845.09C570.79,-820.48 567.2,-785.65 564.62,-760.65\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;) -->\n",
       "<g id=\"node34\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"642.39\" cy=\"-728.39\" rx=\"31.96\" ry=\"31.96\"/>\n",
       "<text text-anchor=\"start\" x=\"624.74\" y=\"-736.79\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n",
       "<text text-anchor=\"start\" x=\"623.4\" y=\"-724.79\" font-family=\"Times,serif\" font-size=\"12.00\">splicing</text>\n",
       "<text text-anchor=\"start\" x=\"624.4\" y=\"-712.79\" font-family=\"Times,serif\" font-size=\"12.00\">speckle</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;) -->\n",
       "<g id=\"edge29\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M588.81,-847.52C600.35,-822.08 617.48,-784.32 629.25,-758.38\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node29\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"440.39\" cy=\"-569.58\" rx=\"61.17\" ry=\"61.17\"/>\n",
       "<text text-anchor=\"start\" x=\"400.74\" y=\"-577.98\" font-family=\"Times,serif\" font-size=\"12.00\">ribonucleprotein</text>\n",
       "<text text-anchor=\"start\" x=\"419.73\" y=\"-565.98\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"424.73\" y=\"-553.98\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge25\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M542.22,-702.54C524.96,-680.17 499.11,-646.67 477.77,-619.02\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node30\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"561.39\" cy=\"-569.58\" rx=\"41.38\" ry=\"41.38\"/>\n",
       "<text text-anchor=\"start\" x=\"548.73\" y=\"-583.98\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n",
       "<text text-anchor=\"start\" x=\"535.73\" y=\"-571.98\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n",
       "<text text-anchor=\"start\" x=\"540.73\" y=\"-559.98\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"545.73\" y=\"-547.98\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge26\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M561.39,-696.03C561.39,-671.81 561.39,-638.08 561.39,-611.7\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node33\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"666.39\" cy=\"-569.58\" rx=\"45.63\" ry=\"45.63\"/>\n",
       "<text text-anchor=\"start\" x=\"637.73\" y=\"-577.98\" font-family=\"Times,serif\" font-size=\"12.00\">splicosomal</text>\n",
       "<text text-anchor=\"start\" x=\"645.73\" y=\"-565.98\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"650.73\" y=\"-553.98\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge27\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M578.97,-701.14C595.96,-675.77 621.88,-637.06 641.09,-608.38\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"node31\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"529.39\" cy=\"-422.15\" rx=\"41.38\" ry=\"41.38\"/>\n",
       "<text text-anchor=\"start\" x=\"516.73\" y=\"-436.55\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n",
       "<text text-anchor=\"start\" x=\"503.73\" y=\"-424.55\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n",
       "<text text-anchor=\"start\" x=\"508.73\" y=\"-412.55\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"526.39\" y=\"-400.55\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"edge23\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M552.57,-528.45C548.11,-508.19 542.71,-483.67 538.25,-463.39\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node32\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"627.39\" cy=\"-422.15\" rx=\"39.03\" ry=\"39.03\"/>\n",
       "<text text-anchor=\"start\" x=\"614.73\" y=\"-436.55\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n",
       "<text text-anchor=\"start\" x=\"608.4\" y=\"-424.55\" font-family=\"Times,serif\" font-size=\"12.00\">splicing</text>\n",
       "<text text-anchor=\"start\" x=\"606.73\" y=\"-412.55\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"611.73\" y=\"-400.55\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge24\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M578.39,-531.13C588.55,-508.74 601.38,-480.47 611.36,-458.47\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node37\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"762.39\" cy=\"-728.39\" rx=\"48.92\" ry=\"48.92\"/>\n",
       "<text text-anchor=\"start\" x=\"731.4\" y=\"-742.79\" font-family=\"Times,serif\" font-size=\"12.00\">chromosome</text>\n",
       "<text text-anchor=\"start\" x=\"732.51\" y=\"-730.79\" font-family=\"Times,serif\" font-size=\"12.00\">organization</text>\n",
       "<text text-anchor=\"start\" x=\"741.73\" y=\"-718.79\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"746.73\" y=\"-706.79\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge33\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M756.52,-824.4C757.45,-809.3 758.47,-792.78 759.4,-777.82\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node38\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"777.39\" cy=\"-569.58\" rx=\"47.35\" ry=\"47.35\"/>\n",
       "<text text-anchor=\"start\" x=\"753.07\" y=\"-583.98\" font-family=\"Times,serif\" font-size=\"12.00\">chromatin</text>\n",
       "<text text-anchor=\"start\" x=\"747.51\" y=\"-571.98\" font-family=\"Times,serif\" font-size=\"12.00\">organization</text>\n",
       "<text text-anchor=\"start\" x=\"756.73\" y=\"-559.98\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"761.73\" y=\"-547.98\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge32\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M767.02,-679.06C768.89,-659.53 771.03,-637.07 772.89,-617.67\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node39\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"777.39\" cy=\"-422.15\" rx=\"34.31\" ry=\"34.31\"/>\n",
       "<text text-anchor=\"start\" x=\"765.73\" y=\"-430.55\" font-family=\"Times,serif\" font-size=\"12.00\">HAT</text>\n",
       "<text text-anchor=\"start\" x=\"756.73\" y=\"-418.55\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"761.73\" y=\"-406.55\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge31\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M777.39,-521.36C777.39,-500.46 777.39,-476.34 777.39,-457.24\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"node40\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"777.39\" cy=\"-301.09\" rx=\"34.31\" ry=\"34.31\"/>\n",
       "<text text-anchor=\"start\" x=\"762.73\" y=\"-309.49\" font-family=\"Times,serif\" font-size=\"12.00\">NuA4</text>\n",
       "<text text-anchor=\"start\" x=\"765.73\" y=\"-297.49\" font-family=\"Times,serif\" font-size=\"12.00\">HAT</text>\n",
       "<text text-anchor=\"start\" x=\"756.73\" y=\"-285.49\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"edge30\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M777.39,-387.03C777.39,-371.09 777.39,-352.16 777.39,-336.22\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"node42\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"873.39\" cy=\"-728.39\" rx=\"39.49\" ry=\"39.49\"/>\n",
       "<text text-anchor=\"start\" x=\"849.07\" y=\"-736.79\" font-family=\"Times,serif\" font-size=\"12.00\">chromatin</text>\n",
       "<text text-anchor=\"start\" x=\"849.07\" y=\"-724.79\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n",
       "<text text-anchor=\"start\" x=\"852.73\" y=\"-712.79\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"edge34\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M873.39,-824.4C873.39,-806.18 873.39,-785.9 873.39,-768.77\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;) -->\n",
       "<g id=\"node44\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"996.39\" cy=\"-728.39\" rx=\"56\" ry=\"56\"/>\n",
       "<text text-anchor=\"start\" x=\"976.07\" y=\"-754.79\" font-family=\"Times,serif\" font-size=\"12.00\">negative</text>\n",
       "<text text-anchor=\"start\" x=\"972.07\" y=\"-742.79\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n",
       "<text text-anchor=\"start\" x=\"991.4\" y=\"-730.79\" font-family=\"Times,serif\" font-size=\"12.00\">of</text>\n",
       "<text text-anchor=\"start\" x=\"983.73\" y=\"-718.79\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n",
       "<text text-anchor=\"start\" x=\"966.73\" y=\"-706.79\" font-family=\"Times,serif\" font-size=\"12.00\">biosynthesis</text>\n",
       "<text text-anchor=\"start\" x=\"978.4\" y=\"-694.79\" font-family=\"Times,serif\" font-size=\"12.00\">process</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;) -->\n",
       "<g id=\"edge35\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M996.39,-821.12C996.39,-809.46 996.39,-797.11 996.39,-785.37\"/>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x106c56e80>"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "music_map = {\n",
    "    'cell': {\n",
    "        'cytoplasm': {\n",
    "            'secretory organelles': {\n",
    "                'transmembrane transport systems': {\n",
    "                    'ion transmembrane transport systems': {}}},\n",
    "            'cytoplasmic organelles': {\n",
    "                'subgroup of cytoplasmic organelles': {\n",
    "                    'metabolic organelles': {\n",
    "                        'subgroup of metabolic organelles': {},\n",
    "                        'mitochondrion': {}},\n",
    "                    'ribonucleoprotein complex family': {\n",
    "                        'RNA processing complex 1': {'RNA splicing complex 1': {}},\n",
    "                        'ribosome biogenesis community': {\n",
    "                            'ribosome': {\n",
    "                                'ribosomal subunit': {\n",
    "                                    'mitochondirial large ribosomal subunit': {\n",
    "                                        # 'large ribosomal subunit subcomplex 1': {}\n",
    "                                    },\n",
    "                                    'mito-cyto ribosomal cluster': {},\n",
    "                                    'cotranslational protein targeting to membrane system': {},\n",
    "                                    'ribosomal complex 5': {},\n",
    "                                    'ribosomal complex 2': {}}},\n",
    "                            'nucleic acid binding complex': {}}}}},\n",
    "            'transcriptional regulation complex family': {\n",
    "                'transcriptional regulation complexes': {}}},\n",
    "        'nucleus': {\n",
    "            'DNA metabolic assembly': {},\n",
    "            'nuclear lumen': {\n",
    "                'nucleolus': {\n",
    "                    'ribonucleprotein complex family': {},\n",
    "                    'RNA processing complex family': {\n",
    "                        'RNA processing complex 1': {},\n",
    "                        'RNA splicing complex family': {}},\n",
    "                    'splicosomal complex family': {}},\n",
    "                'nuclear splicing speckle': {}},\n",
    "            'nuclear body': {},\n",
    "            'nucleo-plasm 1': {\n",
    "                'chromosome organization complex family': {\n",
    "                    'chromatin organization complex family': {\n",
    "                        'HAT complex family': {\n",
    "                            'NuA4 HAT complex': {}}}}},\n",
    "            'nucleo-plasm 2': {\n",
    "                'chromatin regulation complex': {}},\n",
    "            'nuclear transcriptional speckle': {\n",
    "                'negative regulation of RNA biosynthesis process': {}}}}}\n",
    "music_map1 = replace_regex_recursive(music_map)\n",
    "plot_bigraph(music_map1, **plot_settings, filename='cell_hierarchy')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "ab431982-d156-4b7a-9918-384a5ea5132a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'cell': {'cytoplasm': {'secretory organelles': {'transmembrane transport systems': {'ion transmembrane transport systems': {}}},\n",
       "   'cytoplasmic organelles': {'subgroup of cytoplasmic organelles': {'metabolic organelles': {'subgroup of metabolic organelles': {},\n",
       "      'mitochondrion': {}},\n",
       "     'ribonucleoprotein complex family': {'RNA processing complex 1': {'RNA splicing complex 1': {}},\n",
       "      'ribosome biogenesis community': {'ribosome': {'ribosomal subunit': {'mitochondirial large ribosomal subunit': {},\n",
       "         'mito-cyto ribosomal cluster': {},\n",
       "         'cotranslational protein targeting to membrane system': {},\n",
       "         'ribosomal complex 5': {},\n",
       "         'ribosomal complex 2': {}}},\n",
       "       'nucleic acid binding complex': {}}}}},\n",
       "   'transcriptional regulation complex family': {'transcriptional regulation complexes': {}}},\n",
       "  'nucleus': {'DNA metabolic assembly': {},\n",
       "   'nuclear lumen': {'nucleolus': {'ribonucleprotein complex family': {},\n",
       "     'RNA processing complex family': {'RNA processing complex 1': {},\n",
       "      'RNA splicing complex family': {}},\n",
       "     'splicosomal complex family': {}},\n",
       "    'nuclear splicing speckle': {}},\n",
       "   'nuclear body': {},\n",
       "   'nucleo-plasm 1': {'chromosome organization complex family': {'chromatin organization complex family': {'HAT complex family': {'NuA4 HAT complex': {}}}}},\n",
       "   'nucleo-plasm 2': {'chromatin regulation complex': {}},\n",
       "   'nuclear transcriptional speckle': {'negative regulation of RNA biosynthesis process': {}}}}}"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "music_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "5a76608a-fff0-4005-9f32-c02afef39d9e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/cell_hierarchy_functions\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 9.0.0 (0)\n",
       " -->\n",
       "<!-- Title: bigraph Pages: 1 -->\n",
       "<svg width=\"1120pt\" height=\"500pt\"\n",
       " viewBox=\"0.00 0.00 1152.00 513.87\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(0.464602 0.464602) rotate(0) translate(4 1071.32)\">\n",
       "<title>bigraph</title>\n",
       "<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-1071.32 2406.67,-1071.32 2406.67,4 -4,4\"/>\n",
       "<!-- (&#39;cell&#39;,) -->\n",
       "<g id=\"node1\" class=\"node\">\n",
       "<title>(&#39;cell&#39;,)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1523.34\" cy=\"-1049.32\" rx=\"18\" ry=\"18\"/>\n",
       "<text text-anchor=\"start\" x=\"1514.68\" y=\"-1045.72\" font-family=\"Times,serif\" font-size=\"12.00\">cell</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;) -->\n",
       "<g id=\"node2\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"274.34\" cy=\"-956.99\" rx=\"28.67\" ry=\"28.67\"/>\n",
       "<text text-anchor=\"start\" x=\"249.68\" y=\"-953.39\" font-family=\"Times,serif\" font-size=\"12.00\">cytoplasm</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;) -->\n",
       "<g id=\"edge66\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1504.7,-1047.33C1414.28,-1042.53 1012.24,-1020.7 683.34,-995.32 540.8,-984.32 370.65,-967.65 303.58,-960.94\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;) -->\n",
       "<g id=\"node25\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"714.34\" cy=\"-956.99\" rx=\"21.71\" ry=\"21.71\"/>\n",
       "<text text-anchor=\"start\" x=\"696.01\" y=\"-953.39\" font-family=\"Times,serif\" font-size=\"12.00\">nucleus</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;) -->\n",
       "<g id=\"edge67\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1504.92,-1046.26C1397.56,-1034.27 853.17,-973.49 736.41,-960.45\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;transcription&#39;) -->\n",
       "<g id=\"node45\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;transcription&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"blue\" stroke-width=\"2\" cx=\"1238.34\" cy=\"-956.99\" rx=\"34.88\" ry=\"34.88\"/>\n",
       "<text text-anchor=\"start\" x=\"1208.02\" y=\"-953.39\" font-family=\"Times,serif\" font-size=\"12.00\">transcription</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;) -->\n",
       "<g id=\"edge68\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1505.94,-1042.8C1459.65,-1028.13 1332.87,-987.95 1271.93,-968.64\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;translation&#39;) -->\n",
       "<g id=\"node51\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;translation&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"blue\" stroke-width=\"2\" cx=\"1422.34\" cy=\"-956.99\" rx=\"29.4\" ry=\"29.4\"/>\n",
       "<text text-anchor=\"start\" x=\"1397.01\" y=\"-953.39\" font-family=\"Times,serif\" font-size=\"12.00\">translation</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;translation&#39;) -->\n",
       "<g id=\"edge69\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;translation&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1509.72,-1036.14C1492.92,-1021.11 1464.15,-995.38 1444.23,-977.57\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;metabolism&#39;) -->\n",
       "<g id=\"node54\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;metabolism&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"blue\" stroke-width=\"2\" cx=\"1523.34\" cy=\"-956.99\" rx=\"32.33\" ry=\"32.33\"/>\n",
       "<text text-anchor=\"start\" x=\"1495.34\" y=\"-953.39\" font-family=\"Times,serif\" font-size=\"12.00\">metabolism</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;) -->\n",
       "<g id=\"edge70\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1523.34,-1030.47C1523.34,-1018.95 1523.34,-1003.64 1523.34,-990.06\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;) -->\n",
       "<g id=\"node59\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"blue\" stroke-width=\"2\" cx=\"1637.34\" cy=\"-956.99\" rx=\"19.94\" ry=\"19.94\"/>\n",
       "<text text-anchor=\"start\" x=\"1628.68\" y=\"-959.39\" font-family=\"Times,serif\" font-size=\"12.00\">cell</text>\n",
       "<text text-anchor=\"start\" x=\"1624.68\" y=\"-947.39\" font-family=\"Times,serif\" font-size=\"12.00\">cycle</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;) -->\n",
       "<g id=\"edge71\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1537.32,-1037.24C1558.72,-1020.28 1599.48,-987.99 1621.79,-970.31\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;signalling&#39;) -->\n",
       "<g id=\"node64\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;signalling&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"blue\" stroke-width=\"2\" cx=\"1805.34\" cy=\"-956.99\" rx=\"27.57\" ry=\"27.57\"/>\n",
       "<text text-anchor=\"start\" x=\"1781.67\" y=\"-953.39\" font-family=\"Times,serif\" font-size=\"12.00\">signalling</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;signalling&#39;) -->\n",
       "<g id=\"edge72\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;signalling&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1540.98,-1042.67C1588.85,-1027.33 1721.43,-984.87 1778.57,-966.57\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;) -->\n",
       "<g id=\"node68\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"blue\" stroke-width=\"2\" cx=\"1971.34\" cy=\"-956.99\" rx=\"32.2\" ry=\"32.2\"/>\n",
       "<text text-anchor=\"start\" x=\"1954.35\" y=\"-959.39\" font-family=\"Times,serif\" font-size=\"12.00\">protein</text>\n",
       "<text text-anchor=\"start\" x=\"1950.01\" y=\"-947.39\" font-family=\"Times,serif\" font-size=\"12.00\">transport</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;) -->\n",
       "<g id=\"edge73\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1541.83,-1046.02C1591.06,-1039.72 1729.28,-1020.97 1842.34,-995.32 1876.22,-987.63 1914.36,-976.25 1940.33,-968.07\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;) -->\n",
       "<g id=\"node73\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"blue\" stroke-width=\"2\" cx=\"2104.34\" cy=\"-956.99\" rx=\"38.33\" ry=\"38.33\"/>\n",
       "<text text-anchor=\"start\" x=\"2091.67\" y=\"-959.39\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n",
       "<text text-anchor=\"start\" x=\"2078.68\" y=\"-947.39\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;) -->\n",
       "<g id=\"edge74\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1541.87,-1048.15C1607.28,-1047.16 1833.06,-1040.46 2012.34,-995.32 2031.9,-990.39 2052.79,-982.17 2069.72,-974.7\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"node3\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"112.34\" cy=\"-831.62\" rx=\"36.76\" ry=\"36.76\"/>\n",
       "<text text-anchor=\"start\" x=\"90.35\" y=\"-834.02\" font-family=\"Times,serif\" font-size=\"12.00\">secretory</text>\n",
       "<text text-anchor=\"start\" x=\"87.79\" y=\"-822.02\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"edge20\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M249.58,-940.69C227.19,-926.51 193.65,-904.39 166.34,-882.66 156.94,-875.18 147.16,-866.45 138.55,-858.39\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"node6\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"218.34\" cy=\"-831.62\" rx=\"43.04\" ry=\"43.04\"/>\n",
       "<text text-anchor=\"start\" x=\"189.35\" y=\"-834.02\" font-family=\"Times,serif\" font-size=\"12.00\">cytoplasmic</text>\n",
       "<text text-anchor=\"start\" x=\"193.79\" y=\"-822.02\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"edge21\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M262.46,-929.83C254.82,-912.99 244.75,-890.8 236.16,-871.88\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node23\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"330.34\" cy=\"-831.62\" rx=\"51.05\" ry=\"51.05\"/>\n",
       "<text text-anchor=\"start\" x=\"295.68\" y=\"-846.02\" font-family=\"Times,serif\" font-size=\"12.00\">transcriptional</text>\n",
       "<text text-anchor=\"start\" x=\"306.02\" y=\"-834.02\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n",
       "<text text-anchor=\"start\" x=\"309.68\" y=\"-822.02\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"314.68\" y=\"-810.02\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge22\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M286.22,-929.83C292.89,-915.11 301.43,-896.31 309.21,-879.17\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n",
       "<g id=\"node4\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"86.34\" cy=\"-690.23\" rx=\"54.34\" ry=\"54.34\"/>\n",
       "<text text-anchor=\"start\" x=\"49.36\" y=\"-698.63\" font-family=\"Times,serif\" font-size=\"12.00\">transmembrane</text>\n",
       "<text text-anchor=\"start\" x=\"65.01\" y=\"-686.63\" font-family=\"Times,serif\" font-size=\"12.00\">transport</text>\n",
       "<text text-anchor=\"start\" x=\"67.34\" y=\"-674.63\" font-family=\"Times,serif\" font-size=\"12.00\">systems</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n",
       "<g id=\"edge2\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M105.64,-794.72C102.8,-779.48 99.43,-761.38 96.3,-744.62\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n",
       "<g id=\"node5\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"54.34\" cy=\"-537.53\" rx=\"54.34\" ry=\"54.34\"/>\n",
       "<text text-anchor=\"start\" x=\"46.67\" y=\"-551.93\" font-family=\"Times,serif\" font-size=\"12.00\">ion</text>\n",
       "<text text-anchor=\"start\" x=\"17.36\" y=\"-539.93\" font-family=\"Times,serif\" font-size=\"12.00\">transmembrane</text>\n",
       "<text text-anchor=\"start\" x=\"33.01\" y=\"-527.93\" font-family=\"Times,serif\" font-size=\"12.00\">transport</text>\n",
       "<text text-anchor=\"start\" x=\"35.34\" y=\"-515.93\" font-family=\"Times,serif\" font-size=\"12.00\">systems</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;) -->\n",
       "<g id=\"edge1\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;secretory&lt;br/&gt;organelles&#39;, &#39;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;, &#39;ion&lt;br/&gt;transmembrane&lt;br/&gt;transport&lt;br/&gt;systems&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M75.06,-636.1C72,-621.69 68.69,-606.1 65.63,-591.69\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"node7\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"218.34\" cy=\"-690.23\" rx=\"43.04\" ry=\"43.04\"/>\n",
       "<text text-anchor=\"start\" x=\"196.01\" y=\"-704.63\" font-family=\"Times,serif\" font-size=\"12.00\">subgroup</text>\n",
       "<text text-anchor=\"start\" x=\"213.34\" y=\"-692.63\" font-family=\"Times,serif\" font-size=\"12.00\">of</text>\n",
       "<text text-anchor=\"start\" x=\"189.35\" y=\"-680.63\" font-family=\"Times,serif\" font-size=\"12.00\">cytoplasmic</text>\n",
       "<text text-anchor=\"start\" x=\"193.79\" y=\"-668.63\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"edge18\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M218.34,-787.66C218.34,-770.54 218.34,-750.89 218.34,-733.81\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"node8\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"163.34\" cy=\"-537.53\" rx=\"36.76\" ry=\"36.76\"/>\n",
       "<text text-anchor=\"start\" x=\"139.68\" y=\"-539.93\" font-family=\"Times,serif\" font-size=\"12.00\">metabolic</text>\n",
       "<text text-anchor=\"start\" x=\"138.79\" y=\"-527.93\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"edge16\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M203.6,-648.85C194.95,-625.13 184.12,-595.48 175.88,-572.89\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node11\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"280.34\" cy=\"-537.53\" rx=\"62.36\" ry=\"62.36\"/>\n",
       "<text text-anchor=\"start\" x=\"237.69\" y=\"-545.93\" font-family=\"Times,serif\" font-size=\"12.00\">ribonucleoprotein</text>\n",
       "<text text-anchor=\"start\" x=\"259.68\" y=\"-533.93\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"264.68\" y=\"-521.93\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge17\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M234.63,-649.64C241.38,-633.24 249.33,-613.91 256.7,-596\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"node9\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"73.34\" cy=\"-398.49\" rx=\"36.76\" ry=\"36.76\"/>\n",
       "<text text-anchor=\"start\" x=\"51.01\" y=\"-412.89\" font-family=\"Times,serif\" font-size=\"12.00\">subgroup</text>\n",
       "<text text-anchor=\"start\" x=\"68.34\" y=\"-400.89\" font-family=\"Times,serif\" font-size=\"12.00\">of</text>\n",
       "<text text-anchor=\"start\" x=\"49.68\" y=\"-388.89\" font-family=\"Times,serif\" font-size=\"12.00\">metabolic</text>\n",
       "<text text-anchor=\"start\" x=\"48.79\" y=\"-376.89\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;) -->\n",
       "<g id=\"edge3\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;metabolic&lt;br/&gt;organelles&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M143.14,-505.78C128.32,-483.21 108.28,-452.69 93.47,-430.14\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;) -->\n",
       "<g id=\"node10\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"168.34\" cy=\"-398.49\" rx=\"40.38\" ry=\"40.38\"/>\n",
       "<text text-anchor=\"start\" x=\"133.01\" y=\"-394.89\" font-family=\"Times,serif\" font-size=\"12.00\">mitochondrion</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;) -->\n",
       "<g id=\"edge4\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;metabolic&lt;br/&gt;organelles&#39;, &#39;mitochondrion&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M164.68,-499.81C165.36,-481.25 166.18,-458.7 166.87,-439.69\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"node12\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"275.34\" cy=\"-398.49\" rx=\"38.33\" ry=\"38.33\"/>\n",
       "<text text-anchor=\"start\" x=\"262.67\" y=\"-412.89\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n",
       "<text text-anchor=\"start\" x=\"249.68\" y=\"-400.89\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n",
       "<text text-anchor=\"start\" x=\"254.68\" y=\"-388.89\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"272.34\" y=\"-376.89\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"edge14\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M278.07,-474.43C277.61,-461.82 277.14,-448.92 276.73,-437.46\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;) -->\n",
       "<g id=\"node14\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"372.34\" cy=\"-398.49\" rx=\"40.69\" ry=\"40.69\"/>\n",
       "<text text-anchor=\"start\" x=\"350.01\" y=\"-406.89\" font-family=\"Times,serif\" font-size=\"12.00\">ribosome</text>\n",
       "<text text-anchor=\"start\" x=\"347.01\" y=\"-394.89\" font-family=\"Times,serif\" font-size=\"12.00\">biogenesis</text>\n",
       "<text text-anchor=\"start\" x=\"345.01\" y=\"-382.89\" font-family=\"Times,serif\" font-size=\"12.00\">community</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;) -->\n",
       "<g id=\"edge15\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M315.12,-484.72C326.72,-467.44 339.34,-448.65 349.76,-433.12\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"node13\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"275.34\" cy=\"-285.82\" rx=\"35.98\" ry=\"35.98\"/>\n",
       "<text text-anchor=\"start\" x=\"262.67\" y=\"-300.22\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n",
       "<text text-anchor=\"start\" x=\"256.34\" y=\"-288.22\" font-family=\"Times,serif\" font-size=\"12.00\">splicing</text>\n",
       "<text text-anchor=\"start\" x=\"254.68\" y=\"-276.22\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"272.34\" y=\"-264.22\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"edge5\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M275.34,-359.4C275.34,-347.52 275.34,-334.46 275.34,-322.74\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;) -->\n",
       "<g id=\"node15\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"364.34\" cy=\"-285.82\" rx=\"26.1\" ry=\"26.1\"/>\n",
       "<text text-anchor=\"start\" x=\"342.01\" y=\"-282.22\" font-family=\"Times,serif\" font-size=\"12.00\">ribosome</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;) -->\n",
       "<g id=\"edge12\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M369.41,-356.91C368.34,-342.1 367.16,-325.78 366.2,-312.61\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"node22\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"444.34\" cy=\"-285.82\" rx=\"35.98\" ry=\"35.98\"/>\n",
       "<text text-anchor=\"start\" x=\"427.02\" y=\"-300.22\" font-family=\"Times,serif\" font-size=\"12.00\">nucleic</text>\n",
       "<text text-anchor=\"start\" x=\"434.35\" y=\"-288.22\" font-family=\"Times,serif\" font-size=\"12.00\">acid</text>\n",
       "<text text-anchor=\"start\" x=\"426.01\" y=\"-276.22\" font-family=\"Times,serif\" font-size=\"12.00\">binding</text>\n",
       "<text text-anchor=\"start\" x=\"423.68\" y=\"-264.22\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"edge13\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;nucleic&lt;br/&gt;acid&lt;br/&gt;binding&lt;br/&gt;complex&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M394.53,-363.38C404.2,-348.52 415.43,-331.25 424.78,-316.88\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;) -->\n",
       "<g id=\"node16\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"364.34\" cy=\"-177.87\" rx=\"35.97\" ry=\"35.97\"/>\n",
       "<text text-anchor=\"start\" x=\"340.34\" y=\"-180.27\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n",
       "<text text-anchor=\"start\" x=\"346.67\" y=\"-168.27\" font-family=\"Times,serif\" font-size=\"12.00\">subunit</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;) -->\n",
       "<g id=\"edge11\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M364.34,-258.9C364.34,-245.55 364.34,-229.06 364.34,-214.52\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;) -->\n",
       "<g id=\"node17\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"151.34\" cy=\"-52.95\" rx=\"52\" ry=\"52\"/>\n",
       "<text text-anchor=\"start\" x=\"116.01\" y=\"-67.35\" font-family=\"Times,serif\" font-size=\"12.00\">mitochondirial</text>\n",
       "<text text-anchor=\"start\" x=\"139.45\" y=\"-55.35\" font-family=\"Times,serif\" font-size=\"12.00\">large</text>\n",
       "<text text-anchor=\"start\" x=\"127.34\" y=\"-43.35\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n",
       "<text text-anchor=\"start\" x=\"133.67\" y=\"-31.35\" font-family=\"Times,serif\" font-size=\"12.00\">subunit</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;) -->\n",
       "<g id=\"edge6\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mitochondirial&lt;br/&gt;large&lt;br/&gt;ribosomal&lt;br/&gt;subunit&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M330.38,-164.08C298.64,-151.44 250.56,-130.46 212.34,-105.9 204.62,-100.93 196.82,-95.08 189.5,-89.12\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;) -->\n",
       "<g id=\"node18\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"257.34\" cy=\"-52.95\" rx=\"35.97\" ry=\"35.97\"/>\n",
       "<text text-anchor=\"start\" x=\"234.01\" y=\"-61.35\" font-family=\"Times,serif\" font-size=\"12.00\">mito&#45;cyto</text>\n",
       "<text text-anchor=\"start\" x=\"233.34\" y=\"-49.35\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n",
       "<text text-anchor=\"start\" x=\"241.35\" y=\"-37.35\" font-family=\"Times,serif\" font-size=\"12.00\">cluster</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;) -->\n",
       "<g id=\"edge7\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;mito&#45;cyto&lt;br/&gt;ribosomal&lt;br/&gt;cluster&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M340.76,-149.93C329.17,-136.68 315.01,-120.46 302.34,-105.9 295.37,-97.88 287.81,-89.16 280.91,-81.2\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;) -->\n",
       "<g id=\"node19\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"364.34\" cy=\"-52.95\" rx=\"52.95\" ry=\"52.95\"/>\n",
       "<text text-anchor=\"start\" x=\"329.02\" y=\"-79.35\" font-family=\"Times,serif\" font-size=\"12.00\">cotranslational</text>\n",
       "<text text-anchor=\"start\" x=\"347.35\" y=\"-67.35\" font-family=\"Times,serif\" font-size=\"12.00\">protein</text>\n",
       "<text text-anchor=\"start\" x=\"343.12\" y=\"-55.35\" font-family=\"Times,serif\" font-size=\"12.00\">targeting</text>\n",
       "<text text-anchor=\"start\" x=\"359.67\" y=\"-43.35\" font-family=\"Times,serif\" font-size=\"12.00\">to</text>\n",
       "<text text-anchor=\"start\" x=\"339.02\" y=\"-31.35\" font-family=\"Times,serif\" font-size=\"12.00\">membrane</text>\n",
       "<text text-anchor=\"start\" x=\"347.67\" y=\"-19.35\" font-family=\"Times,serif\" font-size=\"12.00\">system</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;) -->\n",
       "<g id=\"edge8\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;cotranslational&lt;br/&gt;protein&lt;br/&gt;targeting&lt;br/&gt;to&lt;br/&gt;membrane&lt;br/&gt;system&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M364.34,-140.99C364.34,-130.21 364.34,-118.17 364.34,-106.55\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;) -->\n",
       "<g id=\"node20\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"471.34\" cy=\"-52.95\" rx=\"35.97\" ry=\"35.97\"/>\n",
       "<text text-anchor=\"start\" x=\"447.34\" y=\"-61.35\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n",
       "<text text-anchor=\"start\" x=\"450.68\" y=\"-49.35\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"468.34\" y=\"-37.35\" font-family=\"Times,serif\" font-size=\"12.00\">5</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;) -->\n",
       "<g id=\"edge9\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;5&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M387.92,-149.93C399.51,-136.68 413.67,-120.46 426.34,-105.9 433.31,-97.88 440.87,-89.16 447.77,-81.2\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;) -->\n",
       "<g id=\"node21\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"561.34\" cy=\"-52.95\" rx=\"35.97\" ry=\"35.97\"/>\n",
       "<text text-anchor=\"start\" x=\"537.34\" y=\"-61.35\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n",
       "<text text-anchor=\"start\" x=\"540.68\" y=\"-49.35\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"558.34\" y=\"-37.35\" font-family=\"Times,serif\" font-size=\"12.00\">2</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;) -->\n",
       "<g id=\"edge10\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;subgroup&lt;br/&gt;of&lt;br/&gt;cytoplasmic&lt;br/&gt;organelles&#39;, &#39;ribonucleoprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;ribosome&lt;br/&gt;biogenesis&lt;br/&gt;community&#39;, &#39;ribosome&#39;, &#39;ribosomal&lt;br/&gt;subunit&#39;, &#39;ribosomal&lt;br/&gt;complex&lt;br/&gt;2&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M398.89,-165.49C431.26,-153.77 479.97,-133.37 516.34,-105.9 524.91,-99.42 533.02,-91.08 539.92,-82.98\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;) -->\n",
       "<g id=\"node24\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"330.34\" cy=\"-690.23\" rx=\"51.05\" ry=\"51.05\"/>\n",
       "<text text-anchor=\"start\" x=\"295.68\" y=\"-698.63\" font-family=\"Times,serif\" font-size=\"12.00\">transcriptional</text>\n",
       "<text text-anchor=\"start\" x=\"306.02\" y=\"-686.63\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n",
       "<text text-anchor=\"start\" x=\"304.68\" y=\"-674.63\" font-family=\"Times,serif\" font-size=\"12.00\">complexes</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;) -->\n",
       "<g id=\"edge19\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;transcriptional&lt;br/&gt;regulation&lt;br/&gt;complexes&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M330.34,-779.9C330.34,-767.6 330.34,-754.45 330.34,-742.14\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;) -->\n",
       "<g id=\"node26\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"469.34\" cy=\"-831.62\" rx=\"35.49\" ry=\"35.49\"/>\n",
       "<text text-anchor=\"start\" x=\"456.34\" y=\"-840.02\" font-family=\"Times,serif\" font-size=\"12.00\">DNA</text>\n",
       "<text text-anchor=\"start\" x=\"445.68\" y=\"-828.02\" font-family=\"Times,serif\" font-size=\"12.00\">metabolic</text>\n",
       "<text text-anchor=\"start\" x=\"447.01\" y=\"-816.02\" font-family=\"Times,serif\" font-size=\"12.00\">assembly</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;) -->\n",
       "<g id=\"edge36\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;DNA&lt;br/&gt;metabolic&lt;br/&gt;assembly&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M692.46,-951.76C653.67,-943.48 571.45,-922.39 513.34,-882.66 504.84,-876.85 496.93,-869.03 490.23,-861.3\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;) -->\n",
       "<g id=\"node27\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"549.34\" cy=\"-831.62\" rx=\"27\" ry=\"27\"/>\n",
       "<text text-anchor=\"start\" x=\"531.69\" y=\"-834.02\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n",
       "<text text-anchor=\"start\" x=\"534.34\" y=\"-822.02\" font-family=\"Times,serif\" font-size=\"12.00\">lumen</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;) -->\n",
       "<g id=\"edge37\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M693.84,-947.66C666.84,-936 618.87,-912.8 585.34,-882.66 576.82,-875.01 569.14,-865.01 563.07,-855.96\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;) -->\n",
       "<g id=\"node35\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"621.34\" cy=\"-831.62\" rx=\"27\" ry=\"27\"/>\n",
       "<text text-anchor=\"start\" x=\"603.69\" y=\"-834.02\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n",
       "<text text-anchor=\"start\" x=\"609.34\" y=\"-822.02\" font-family=\"Times,serif\" font-size=\"12.00\">body</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;) -->\n",
       "<g id=\"edge38\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;body&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M700.8,-938.94C689.13,-924.17 671.92,-902.16 657.34,-882.66 650.55,-873.58 643.21,-863.44 636.94,-854.67\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"node36\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"714.34\" cy=\"-831.62\" rx=\"47.75\" ry=\"47.75\"/>\n",
       "<text text-anchor=\"start\" x=\"682.02\" y=\"-834.02\" font-family=\"Times,serif\" font-size=\"12.00\">nucleo&#45;plasm</text>\n",
       "<text text-anchor=\"start\" x=\"711.34\" y=\"-822.02\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"edge39\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M714.34,-934.5C714.34,-919.52 714.34,-898.94 714.34,-880.19\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;) -->\n",
       "<g id=\"node41\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"827.34\" cy=\"-831.62\" rx=\"47.75\" ry=\"47.75\"/>\n",
       "<text text-anchor=\"start\" x=\"795.02\" y=\"-834.02\" font-family=\"Times,serif\" font-size=\"12.00\">nucleo&#45;plasm</text>\n",
       "<text text-anchor=\"start\" x=\"824.34\" y=\"-822.02\" font-family=\"Times,serif\" font-size=\"12.00\">2</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;) -->\n",
       "<g id=\"edge40\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M729.11,-939.86C745.74,-921.7 773.27,-891.65 794.96,-867.97\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;) -->\n",
       "<g id=\"node43\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"944.34\" cy=\"-831.62\" rx=\"51.05\" ry=\"51.05\"/>\n",
       "<text text-anchor=\"start\" x=\"926.69\" y=\"-840.02\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n",
       "<text text-anchor=\"start\" x=\"909.68\" y=\"-828.02\" font-family=\"Times,serif\" font-size=\"12.00\">transcriptional</text>\n",
       "<text text-anchor=\"start\" x=\"926.35\" y=\"-816.02\" font-family=\"Times,serif\" font-size=\"12.00\">speckle</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;) -->\n",
       "<g id=\"edge41\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M735.43,-949.11C768.54,-937.83 834.09,-913.55 884.34,-882.66 891.85,-878.05 899.42,-872.55 906.52,-866.91\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;) -->\n",
       "<g id=\"node28\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"532.34\" cy=\"-690.23\" rx=\"26.83\" ry=\"26.83\"/>\n",
       "<text text-anchor=\"start\" x=\"509.35\" y=\"-686.63\" font-family=\"Times,serif\" font-size=\"12.00\">nucleolus</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;) -->\n",
       "<g id=\"edge28\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M546.1,-804.07C543.08,-779.27 538.59,-742.45 535.57,-717.68\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;) -->\n",
       "<g id=\"node34\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"606.34\" cy=\"-690.23\" rx=\"28.91\" ry=\"28.91\"/>\n",
       "<text text-anchor=\"start\" x=\"588.69\" y=\"-698.63\" font-family=\"Times,serif\" font-size=\"12.00\">nuclear</text>\n",
       "<text text-anchor=\"start\" x=\"587.34\" y=\"-686.63\" font-family=\"Times,serif\" font-size=\"12.00\">splicing</text>\n",
       "<text text-anchor=\"start\" x=\"588.35\" y=\"-674.63\" font-family=\"Times,serif\" font-size=\"12.00\">speckle</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;) -->\n",
       "<g id=\"edge29\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nuclear&lt;br/&gt;splicing&lt;br/&gt;speckle&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M559.53,-805.7C569.65,-780.95 585.11,-743.15 595.46,-717.83\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node29\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"418.34\" cy=\"-537.53\" rx=\"58.11\" ry=\"58.11\"/>\n",
       "<text text-anchor=\"start\" x=\"378.69\" y=\"-545.93\" font-family=\"Times,serif\" font-size=\"12.00\">ribonucleprotein</text>\n",
       "<text text-anchor=\"start\" x=\"397.68\" y=\"-533.93\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"402.68\" y=\"-521.93\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge25\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;ribonucleprotein&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M516.25,-667.96C499.98,-646.46 474.33,-612.54 453.36,-584.82\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node30\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"532.34\" cy=\"-537.53\" rx=\"38.33\" ry=\"38.33\"/>\n",
       "<text text-anchor=\"start\" x=\"519.67\" y=\"-551.93\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n",
       "<text text-anchor=\"start\" x=\"506.68\" y=\"-539.93\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n",
       "<text text-anchor=\"start\" x=\"511.68\" y=\"-527.93\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"516.68\" y=\"-515.93\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge26\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M532.34,-662.64C532.34,-638.81 532.34,-603.58 532.34,-576.74\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node33\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"631.34\" cy=\"-537.53\" rx=\"42.57\" ry=\"42.57\"/>\n",
       "<text text-anchor=\"start\" x=\"602.68\" y=\"-545.93\" font-family=\"Times,serif\" font-size=\"12.00\">splicosomal</text>\n",
       "<text text-anchor=\"start\" x=\"610.68\" y=\"-533.93\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"615.68\" y=\"-521.93\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge27\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;splicosomal&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M546.95,-666.99C563.03,-642.51 589.11,-602.82 608.03,-574.02\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"node31\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"501.34\" cy=\"-398.49\" rx=\"38.33\" ry=\"38.33\"/>\n",
       "<text text-anchor=\"start\" x=\"488.67\" y=\"-412.89\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n",
       "<text text-anchor=\"start\" x=\"475.68\" y=\"-400.89\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n",
       "<text text-anchor=\"start\" x=\"480.68\" y=\"-388.89\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"498.34\" y=\"-376.89\" font-family=\"Times,serif\" font-size=\"12.00\">1</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;) -->\n",
       "<g id=\"edge23\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M523.95,-499.45C519.54,-479.95 514.15,-456.11 509.74,-436.6\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node32\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"593.34\" cy=\"-398.49\" rx=\"35.98\" ry=\"35.98\"/>\n",
       "<text text-anchor=\"start\" x=\"580.67\" y=\"-412.89\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n",
       "<text text-anchor=\"start\" x=\"574.34\" y=\"-400.89\" font-family=\"Times,serif\" font-size=\"12.00\">splicing</text>\n",
       "<text text-anchor=\"start\" x=\"572.68\" y=\"-388.89\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"577.68\" y=\"-376.89\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge24\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;lumen&#39;, &#39;nucleolus&#39;, &#39;RNA&lt;br/&gt;processing&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;RNA&lt;br/&gt;splicing&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M547.89,-501.6C557.42,-480.18 569.54,-452.96 578.87,-432.01\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node37\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"717.34\" cy=\"-690.23\" rx=\"45.87\" ry=\"45.87\"/>\n",
       "<text text-anchor=\"start\" x=\"686.35\" y=\"-704.63\" font-family=\"Times,serif\" font-size=\"12.00\">chromosome</text>\n",
       "<text text-anchor=\"start\" x=\"687.46\" y=\"-692.63\" font-family=\"Times,serif\" font-size=\"12.00\">organization</text>\n",
       "<text text-anchor=\"start\" x=\"696.68\" y=\"-680.63\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"701.68\" y=\"-668.63\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge33\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M715.36,-783.04C715.69,-768.05 716.04,-751.55 716.36,-736.71\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node38\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"736.34\" cy=\"-537.53\" rx=\"44.3\" ry=\"44.3\"/>\n",
       "<text text-anchor=\"start\" x=\"712.02\" y=\"-551.93\" font-family=\"Times,serif\" font-size=\"12.00\">chromatin</text>\n",
       "<text text-anchor=\"start\" x=\"706.46\" y=\"-539.93\" font-family=\"Times,serif\" font-size=\"12.00\">organization</text>\n",
       "<text text-anchor=\"start\" x=\"715.68\" y=\"-527.93\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"720.68\" y=\"-515.93\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge32\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M723.04,-644.02C725.5,-624.51 728.37,-601.78 730.81,-582.42\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"node39\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"736.34\" cy=\"-398.49\" rx=\"31.25\" ry=\"31.25\"/>\n",
       "<text text-anchor=\"start\" x=\"724.67\" y=\"-406.89\" font-family=\"Times,serif\" font-size=\"12.00\">HAT</text>\n",
       "<text text-anchor=\"start\" x=\"715.68\" y=\"-394.89\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "<text text-anchor=\"start\" x=\"720.68\" y=\"-382.89\" font-family=\"Times,serif\" font-size=\"12.00\">family</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;) -->\n",
       "<g id=\"edge31\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M736.34,-492.42C736.34,-472.23 736.34,-448.79 736.34,-430.5\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"node40\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"736.34\" cy=\"-285.82\" rx=\"31.25\" ry=\"31.25\"/>\n",
       "<text text-anchor=\"start\" x=\"721.67\" y=\"-294.22\" font-family=\"Times,serif\" font-size=\"12.00\">NuA4</text>\n",
       "<text text-anchor=\"start\" x=\"724.67\" y=\"-282.22\" font-family=\"Times,serif\" font-size=\"12.00\">HAT</text>\n",
       "<text text-anchor=\"start\" x=\"715.68\" y=\"-270.22\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"edge30\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;1&#39;, &#39;chromosome&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;chromatin&lt;br/&gt;organization&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;HAT&lt;br/&gt;complex&lt;br/&gt;family&#39;, &#39;NuA4&lt;br/&gt;HAT&lt;br/&gt;complex&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M736.34,-366.37C736.34,-351.15 736.34,-332.89 736.34,-317.71\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"node42\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"823.34\" cy=\"-690.23\" rx=\"36.44\" ry=\"36.44\"/>\n",
       "<text text-anchor=\"start\" x=\"799.02\" y=\"-698.63\" font-family=\"Times,serif\" font-size=\"12.00\">chromatin</text>\n",
       "<text text-anchor=\"start\" x=\"799.02\" y=\"-686.63\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n",
       "<text text-anchor=\"start\" x=\"802.68\" y=\"-674.63\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"edge34\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nucleo&#45;plasm&lt;br/&gt;2&#39;, &#39;chromatin&lt;br/&gt;regulation&lt;br/&gt;complex&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M825.97,-783.04C825.45,-764.9 824.87,-744.53 824.38,-727.6\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;) -->\n",
       "<g id=\"node44\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"930.34\" cy=\"-690.23\" rx=\"52.95\" ry=\"52.95\"/>\n",
       "<text text-anchor=\"start\" x=\"910.02\" y=\"-716.63\" font-family=\"Times,serif\" font-size=\"12.00\">negative</text>\n",
       "<text text-anchor=\"start\" x=\"906.02\" y=\"-704.63\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n",
       "<text text-anchor=\"start\" x=\"925.34\" y=\"-692.63\" font-family=\"Times,serif\" font-size=\"12.00\">of</text>\n",
       "<text text-anchor=\"start\" x=\"917.67\" y=\"-680.63\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n",
       "<text text-anchor=\"start\" x=\"900.67\" y=\"-668.63\" font-family=\"Times,serif\" font-size=\"12.00\">biosynthesis</text>\n",
       "<text text-anchor=\"start\" x=\"912.35\" y=\"-656.63\" font-family=\"Times,serif\" font-size=\"12.00\">process</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;) -->\n",
       "<g id=\"edge35\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleus&#39;, &#39;nuclear&lt;br/&gt;transcriptional&lt;br/&gt;speckle&#39;, &#39;negative&lt;br/&gt;regulation&lt;br/&gt;of&lt;br/&gt;RNA&lt;br/&gt;biosynthesis&lt;br/&gt;process&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M939.25,-779.9C938.06,-768.07 936.79,-755.45 935.6,-743.55\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"node46\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1197.34\" cy=\"-831.62\" rx=\"18\" ry=\"18\"/>\n",
       "<text text-anchor=\"start\" x=\"1184.34\" y=\"-828.02\" font-family=\"Times,serif\" font-size=\"12.00\">wires</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;transcription&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"edge46\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;transcription&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1227.35,-922.93C1219.48,-899.23 1209.2,-868.29 1202.94,-849.47\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;RNA&lt;br/&gt;processing&#39;) -->\n",
       "<g id=\"node47\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;RNA&lt;br/&gt;processing&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1039.34\" cy=\"-690.23\" rx=\"38.33\" ry=\"38.33\"/>\n",
       "<text text-anchor=\"start\" x=\"1026.67\" y=\"-692.63\" font-family=\"Times,serif\" font-size=\"12.00\">RNA</text>\n",
       "<text text-anchor=\"start\" x=\"1013.68\" y=\"-680.63\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;RNA&lt;br/&gt;processing&#39;) -->\n",
       "<g id=\"edge42\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;RNA&lt;br/&gt;processing&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1182.45,-820.24C1160.67,-804.79 1119.25,-774.35 1087.34,-744.57 1079.42,-737.18 1071.34,-728.65 1064.12,-720.61\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;regulation&#39;) -->\n",
       "<g id=\"node48\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;regulation&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1124.34\" cy=\"-690.23\" rx=\"28.3\" ry=\"28.3\"/>\n",
       "<text text-anchor=\"start\" x=\"1100.02\" y=\"-686.63\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;regulation&#39;) -->\n",
       "<g id=\"edge43\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;regulation&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1188.97,-814.63C1176.35,-790.53 1152.34,-744.69 1137.41,-716.18\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;nucleic&lt;br/&gt;acids&#39;) -->\n",
       "<g id=\"node49\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;nucleic&lt;br/&gt;acids&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1197.34\" cy=\"-690.23\" rx=\"26.54\" ry=\"26.54\"/>\n",
       "<text text-anchor=\"start\" x=\"1180.02\" y=\"-692.63\" font-family=\"Times,serif\" font-size=\"12.00\">nucleic</text>\n",
       "<text text-anchor=\"start\" x=\"1185.01\" y=\"-680.63\" font-family=\"Times,serif\" font-size=\"12.00\">acids</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;nucleic&lt;br/&gt;acids&#39;) -->\n",
       "<g id=\"edge44\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;nucleic&lt;br/&gt;acids&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1197.34,-812.68C1197.34,-788.56 1197.34,-745.5 1197.34,-717.67\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;negative&lt;br/&gt;regulation&#39;) -->\n",
       "<g id=\"node50\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;negative&lt;br/&gt;regulation&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1278.34\" cy=\"-690.23\" rx=\"36.44\" ry=\"36.44\"/>\n",
       "<text text-anchor=\"start\" x=\"1258.02\" y=\"-692.63\" font-family=\"Times,serif\" font-size=\"12.00\">negative</text>\n",
       "<text text-anchor=\"start\" x=\"1254.02\" y=\"-680.63\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;negative&lt;br/&gt;regulation&#39;) -->\n",
       "<g id=\"edge45\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;transcription&#39;, &#39;wires&#39;, &#39;negative&lt;br/&gt;regulation&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1206.32,-815.17C1219.21,-792.98 1243.26,-751.6 1260,-722.79\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;translation&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"node52\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;translation&#39;, &#39;wires&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1381.34\" cy=\"-831.62\" rx=\"18\" ry=\"18\"/>\n",
       "<text text-anchor=\"start\" x=\"1368.34\" y=\"-828.02\" font-family=\"Times,serif\" font-size=\"12.00\">wires</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;translation&#39;)&#45;&gt;(&#39;cell&#39;, &#39;translation&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"edge48\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;translation&#39;)&#45;&gt;(&#39;cell&#39;, &#39;translation&#39;, &#39;wires&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1413.14,-928.3C1405.11,-904.15 1393.72,-869.88 1386.98,-849.58\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;translation&#39;, &#39;wires&#39;, &#39;ribosomes&#39;) -->\n",
       "<g id=\"node53\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;translation&#39;, &#39;wires&#39;, &#39;ribosomes&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1361.34\" cy=\"-690.23\" rx=\"28.67\" ry=\"28.67\"/>\n",
       "<text text-anchor=\"start\" x=\"1336.68\" y=\"-686.63\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomes</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;translation&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;translation&#39;, &#39;wires&#39;, &#39;ribosomes&#39;) -->\n",
       "<g id=\"edge47\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;translation&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;translation&#39;, &#39;wires&#39;, &#39;ribosomes&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1378.81,-812.96C1375.43,-789.45 1369.42,-747.57 1365.41,-719.6\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"node55\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1523.34\" cy=\"-831.62\" rx=\"18\" ry=\"18\"/>\n",
       "<text text-anchor=\"start\" x=\"1510.34\" y=\"-828.02\" font-family=\"Times,serif\" font-size=\"12.00\">wires</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;metabolism&#39;)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"edge52\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;metabolism&#39;)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1523.34,-923.9C1523.34,-900.48 1523.34,-869.6 1523.34,-850.44\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;, &#39;organelles&#39;) -->\n",
       "<g id=\"node56\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;, &#39;organelles&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1436.34\" cy=\"-690.23\" rx=\"28.54\" ry=\"28.54\"/>\n",
       "<text text-anchor=\"start\" x=\"1411.79\" y=\"-686.63\" font-family=\"Times,serif\" font-size=\"12.00\">organelles</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;, &#39;organelles&#39;) -->\n",
       "<g id=\"edge49\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;, &#39;organelles&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1513.86,-815.43C1498.85,-791.37 1469.45,-744.27 1451.48,-715.49\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;, &#39;mitochondrion&#39;) -->\n",
       "<g id=\"node57\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;, &#39;mitochondrion&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1523.34\" cy=\"-690.23\" rx=\"40.38\" ry=\"40.38\"/>\n",
       "<text text-anchor=\"start\" x=\"1488.01\" y=\"-686.63\" font-family=\"Times,serif\" font-size=\"12.00\">mitochondrion</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;, &#39;mitochondrion&#39;) -->\n",
       "<g id=\"edge50\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;, &#39;mitochondrion&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1523.34,-812.68C1523.34,-792.24 1523.34,-758.19 1523.34,-731.34\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;, &#39;ion&lt;br/&gt;transport&#39;) -->\n",
       "<g id=\"node58\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;, &#39;ion&lt;br/&gt;transport&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1614.34\" cy=\"-690.23\" rx=\"32.2\" ry=\"32.2\"/>\n",
       "<text text-anchor=\"start\" x=\"1606.67\" y=\"-692.63\" font-family=\"Times,serif\" font-size=\"12.00\">ion</text>\n",
       "<text text-anchor=\"start\" x=\"1593.01\" y=\"-680.63\" font-family=\"Times,serif\" font-size=\"12.00\">transport</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;, &#39;ion&lt;br/&gt;transport&#39;) -->\n",
       "<g id=\"edge51\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;metabolism&#39;, &#39;wires&#39;, &#39;ion&lt;br/&gt;transport&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1533.25,-815.43C1548.5,-792.08 1577.92,-747.01 1596.82,-718.07\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"node60\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1718.34\" cy=\"-831.62\" rx=\"18\" ry=\"18\"/>\n",
       "<text text-anchor=\"start\" x=\"1705.34\" y=\"-828.02\" font-family=\"Times,serif\" font-size=\"12.00\">wires</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"edge56\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1648.26,-939.35C1664.16,-915.15 1693.4,-870.6 1708.6,-847.45\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;, &#39;DNA&#39;) -->\n",
       "<g id=\"node61\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;, &#39;DNA&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1682.34\" cy=\"-690.23\" rx=\"18\" ry=\"18\"/>\n",
       "<text text-anchor=\"start\" x=\"1669.34\" y=\"-686.63\" font-family=\"Times,serif\" font-size=\"12.00\">DNA</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;, &#39;DNA&#39;) -->\n",
       "<g id=\"edge53\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;, &#39;DNA&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1713.85,-813.25C1706.92,-786.39 1693.67,-735.11 1686.77,-708.38\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;, &#39;chromosome&#39;) -->\n",
       "<g id=\"node62\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;, &#39;chromosome&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1754.34\" cy=\"-690.23\" rx=\"35.62\" ry=\"35.62\"/>\n",
       "<text text-anchor=\"start\" x=\"1723.35\" y=\"-686.63\" font-family=\"Times,serif\" font-size=\"12.00\">chromosome</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;, &#39;chromosome&#39;) -->\n",
       "<g id=\"edge54\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;, &#39;chromosome&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1722.83,-813.25C1728.46,-791.42 1738.26,-753.48 1745.44,-725.68\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;, &#39;chromatin&#39;) -->\n",
       "<g id=\"node63\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;, &#39;chromatin&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1836.34\" cy=\"-690.23\" rx=\"28.3\" ry=\"28.3\"/>\n",
       "<text text-anchor=\"start\" x=\"1812.02\" y=\"-686.63\" font-family=\"Times,serif\" font-size=\"12.00\">chromatin</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;, &#39;chromatin&#39;) -->\n",
       "<g id=\"edge55\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cell&lt;br/&gt;cycle&#39;, &#39;wires&#39;, &#39;chromatin&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1731.32,-818.04C1747.86,-801.75 1777,-772.15 1799.34,-744.57 1806.93,-735.2 1814.64,-724.34 1821.04,-714.88\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"node65\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1907.34\" cy=\"-831.62\" rx=\"18\" ry=\"18\"/>\n",
       "<text text-anchor=\"start\" x=\"1894.34\" y=\"-828.02\" font-family=\"Times,serif\" font-size=\"12.00\">wires</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;signalling&#39;)&#45;&gt;(&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"edge59\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;signalling&#39;)&#45;&gt;(&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1822.87,-934.78C1843.72,-909.57 1877.93,-868.19 1895.74,-846.65\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;, &#39;transport&#39;) -->\n",
       "<g id=\"node66\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;, &#39;transport&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1907.34\" cy=\"-690.23\" rx=\"25\" ry=\"25\"/>\n",
       "<text text-anchor=\"start\" x=\"1886.01\" y=\"-686.63\" font-family=\"Times,serif\" font-size=\"12.00\">transport</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;, &#39;transport&#39;) -->\n",
       "<g id=\"edge57\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;, &#39;transport&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1907.34,-812.68C1907.34,-788.12 1907.34,-743.92 1907.34,-716.16\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;, &#39;trancript&lt;br/&gt;regulation&#39;) -->\n",
       "<g id=\"node67\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;, &#39;trancript&lt;br/&gt;regulation&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"1986.34\" cy=\"-690.23\" rx=\"36.44\" ry=\"36.44\"/>\n",
       "<text text-anchor=\"start\" x=\"1966.02\" y=\"-692.63\" font-family=\"Times,serif\" font-size=\"12.00\">trancript</text>\n",
       "<text text-anchor=\"start\" x=\"1962.02\" y=\"-680.63\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;, &#39;trancript&lt;br/&gt;regulation&#39;) -->\n",
       "<g id=\"edge58\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;signalling&#39;, &#39;wires&#39;, &#39;trancript&lt;br/&gt;regulation&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1916.25,-814.9C1928.84,-792.68 1952.11,-751.62 1968.37,-722.93\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"node69\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"2144.34\" cy=\"-831.62\" rx=\"18\" ry=\"18\"/>\n",
       "<text text-anchor=\"start\" x=\"2131.34\" y=\"-828.02\" font-family=\"Times,serif\" font-size=\"12.00\">wires</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"edge63\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M1997.62,-937.25C2034.48,-910.96 2100.32,-864.01 2129.6,-843.13\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;, &#39;transmembrane&lt;br/&gt;transport&#39;) -->\n",
       "<g id=\"node70\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;, &#39;transmembrane&lt;br/&gt;transport&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"2095.34\" cy=\"-690.23\" rx=\"54.34\" ry=\"54.34\"/>\n",
       "<text text-anchor=\"start\" x=\"2058.36\" y=\"-692.63\" font-family=\"Times,serif\" font-size=\"12.00\">transmembrane</text>\n",
       "<text text-anchor=\"start\" x=\"2074.01\" y=\"-680.63\" font-family=\"Times,serif\" font-size=\"12.00\">transport</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;, &#39;transmembrane&lt;br/&gt;transport&#39;) -->\n",
       "<g id=\"edge60\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;, &#39;transmembrane&lt;br/&gt;transport&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M2138.43,-813.8C2132.22,-796.14 2122.14,-767.47 2113.29,-742.3\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;, &#39;secretory&#39;) -->\n",
       "<g id=\"node71\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;, &#39;secretory&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"2193.34\" cy=\"-690.23\" rx=\"25.73\" ry=\"25.73\"/>\n",
       "<text text-anchor=\"start\" x=\"2171.35\" y=\"-686.63\" font-family=\"Times,serif\" font-size=\"12.00\">secretory</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;, &#39;secretory&#39;) -->\n",
       "<g id=\"edge61\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;, &#39;secretory&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M2150.25,-813.8C2158.91,-789.19 2175.07,-743.21 2184.95,-715.1\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;, &#39;cotranslation&#39;) -->\n",
       "<g id=\"node72\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;, &#39;cotranslation&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"2272.34\" cy=\"-690.23\" rx=\"35.61\" ry=\"35.61\"/>\n",
       "<text text-anchor=\"start\" x=\"2241.35\" y=\"-686.63\" font-family=\"Times,serif\" font-size=\"12.00\">cotranslation</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;, &#39;cotranslation&#39;) -->\n",
       "<g id=\"edge62\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;protein&lt;br/&gt;transport&#39;, &#39;wires&#39;, &#39;cotranslation&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M2157.21,-818.11C2173.99,-801.64 2204.07,-771.6 2228.34,-744.57 2235.64,-736.44 2243.31,-727.35 2250.16,-719.01\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"node74\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;, &#39;wires&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"2305.34\" cy=\"-831.62\" rx=\"18\" ry=\"18\"/>\n",
       "<text text-anchor=\"start\" x=\"2292.34\" y=\"-828.02\" font-family=\"Times,serif\" font-size=\"12.00\">wires</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;)&#45;&gt;(&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"edge65\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;)&#45;&gt;(&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;, &#39;wires&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M2137.08,-935.89C2180.91,-908.99 2256.92,-862.34 2289.63,-842.26\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;, &#39;wires&#39;, &#39;processing&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"node75\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;, &#39;wires&#39;, &#39;processing&lt;br/&gt;complex&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"2364.34\" cy=\"-690.23\" rx=\"38.33\" ry=\"38.33\"/>\n",
       "<text text-anchor=\"start\" x=\"2338.68\" y=\"-692.63\" font-family=\"Times,serif\" font-size=\"12.00\">processing</text>\n",
       "<text text-anchor=\"start\" x=\"2343.68\" y=\"-680.63\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;, &#39;wires&#39;, &#39;processing&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"edge64\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;, &#39;wires&#39;)&#45;&gt;(&#39;cell&#39;, &#39;RNA&lt;br/&gt;processing&#39;, &#39;wires&#39;, &#39;processing&lt;br/&gt;complex&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M2312.34,-814.08C2321.44,-792.59 2337.56,-754.5 2349.45,-726.41\"/>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x106d46100>"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import copy\n",
    "\n",
    "transcription_process = {\n",
    "    '_ports': {\n",
    "        'RNA processing': 'Any',\n",
    "        'regulation': 'Any',\n",
    "        'nucleic acids': 'Any',\n",
    "        'negative regulation': 'Any'\n",
    "    },\n",
    "    'wires': {\n",
    "        'RNA processing': [\n",
    "            'nucleus', 'nuclear lumen', 'nucleolus', 'RNA processing complex family'],\n",
    "        'regulation': [\n",
    "            'cytoplasm', 'transcriptional regulation complex family', 'transcriptional regulation complexes'],\n",
    "        'nucleic acids': [\n",
    "            'cytoplasm', 'cytoplasmic organelles', 'subgroup of cytoplasmic organelles', \n",
    "            'ribonucleoprotein complex family', 'ribosome biogenesis community', 'nucleic acid binding complex'],\n",
    "        'negative regulation': [\n",
    "            'nucleus', 'nuclear transcriptional speckle', 'negative regulation of RNA biosynthesis process'],\n",
    "    }\n",
    "}\n",
    "\n",
    "translation_process = {\n",
    "    '_ports': {\n",
    "        'ribosomes': 'Any',\n",
    "        'membrane': 'Any'\n",
    "    },\n",
    "    'wires': {\n",
    "        'ribosomes': [\n",
    "            'cytoplasm', 'cytoplasmic organelles', 'subgroup of cytoplasmic organelles', 'ribonucleoprotein complex family', 'ribosome biogenesis community', 'ribosome'],\n",
    "        # 'membrane': ['],\n",
    "    }\n",
    "}\n",
    "\n",
    "metabolism_process = {\n",
    "    '_ports': {\n",
    "        'organelles': 'Any',\n",
    "        'mitochondrion': 'Any',\n",
    "        'ion transport': 'Any',\n",
    "    },\n",
    "    'wires': {\n",
    "        'organelles': [\n",
    "            'cytoplasm', 'cytoplasmic organelles', 'subgroup of cytoplasmic organelles', 'metabolic organelles', ],\n",
    "        'mitochondrion': [\n",
    "            'cytoplasm', 'cytoplasmic organelles', 'subgroup of cytoplasmic organelles', 'metabolic organelles', 'mitochondrion'],\n",
    "        'ion transport': [\n",
    "            'cytoplasm', 'secretory organelles', 'transmembrane transport systems', 'ion transmembrane transport systems'],\n",
    "    }\n",
    "}\n",
    "\n",
    "cell_cycle_process = {\n",
    "    '_ports': {\n",
    "        'DNA': 'Any',\n",
    "        'chromosome': 'Any',\n",
    "        'chromatin': 'Any',\n",
    "    },\n",
    "    'wires': {        \n",
    "        'DNA': ['nucleus', 'DNA metabolic assembly'],\n",
    "        'chromosome': ['nucleus', 'nucleo-plasm 1', 'chromosome organization complex family',],\n",
    "        'chromatin': ['nucleus', 'nucleo-plasm 1', 'chromosome organization complex family', 'chromatin organization complex family'],\n",
    "    }\n",
    "}\n",
    "\n",
    "signalling_process = {\n",
    "    '_ports': {\n",
    "        'transport': 'Any',\n",
    "        'trancript regulation': 'Any',\n",
    "    },\n",
    "    'wires': {        \n",
    "        'transport': [\n",
    "            'cytoplasm', 'secretory organelles', 'transmembrane transport systems'],\n",
    "        'trancript regulation': [\n",
    "            'cytoplasm', 'transcriptional regulation complex family', 'transcriptional regulation complexes'],\n",
    "    }\n",
    "}\n",
    "\n",
    "protein_transport_process = {\n",
    "    '_ports': {\n",
    "        'transmembrane transport': 'Any',\n",
    "        'secretory': 'Any',\n",
    "        'cotranslation': 'Any',\n",
    "    },\n",
    "    'wires': {        \n",
    "        'transmembrane transport': [\n",
    "            'cytoplasm', 'secretory organelles', 'transmembrane transport systems'],\n",
    "        'secretory': [\n",
    "            'cytoplasm', 'secretory organelles'],\n",
    "        'cotranslation': [\n",
    "            'cytoplasm', 'cytoplasmic organelles', 'subgroup of cytoplasmic organelles', 'ribonucleoprotein complex family', \n",
    "            'ribosome biogenesis community', 'ribosome', 'ribosomal subunit', 'cotranslational protein targeting to membrane system']\n",
    "    }\n",
    "}\n",
    "\n",
    "rna_processing_process = {\n",
    "    '_ports': {\n",
    "        'processing complex': 'Any',\n",
    "    },\n",
    "    'wires': {        \n",
    "        'processing complex': [\n",
    "            'nucleus', 'nuclear lumen', 'nucleolus', 'RNA processing complex family'],\n",
    "    }\n",
    "}\n",
    "\n",
    "# compose schema\n",
    "music_map['cell']['transcription'] = transcription_process\n",
    "music_map['cell']['translation'] = translation_process\n",
    "music_map['cell']['metabolism'] = metabolism_process\n",
    "music_map['cell']['cell cycle'] = cell_cycle_process\n",
    "music_map['cell']['signalling'] = signalling_process\n",
    "music_map['cell']['protein transport'] = protein_transport_process\n",
    "music_map['cell']['RNA processing'] = rna_processing_process\n",
    "\n",
    "music_map2 = replace_regex_recursive(music_map)\n",
    "\n",
    "# plot\n",
    "plot_settingsx = copy.deepcopy(plot_settings)\n",
    "plot_settingsx['label_margin'] = '0.02'\n",
    "plot_settingsx['node_border_colors'] = {\n",
    "    ('cell', 'translation'): 'blue',\n",
    "    ('cell', 'transcription'): 'blue',\n",
    "    ('cell', 'metabolism'): 'blue',\n",
    "    ('cell', 'cell<br/>cycle'): 'blue',\n",
    "    ('cell', 'signalling'): 'blue',\n",
    "    ('cell', 'protein<br/>transport'): 'blue',\n",
    "    ('cell', 'RNA<br/>processing'): 'blue',\n",
    "}\n",
    "# plot_settingsx.update({'out_dir': 'out','dpi': '250'})\n",
    "plot_bigraph(music_map2, **plot_settingsx, filename='cell_hierarchy_functions')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "fe3008ad-498e-449f-8521-6ea29fd903f1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/hypothesized_mechanism\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 9.0.0 (0)\n",
       " -->\n",
       "<!-- Title: bigraph Pages: 1 -->\n",
       "<svg width=\"101pt\" height=\"101pt\"\n",
       " viewBox=\"0.00 0.00 103.65 103.65\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 99.65)\">\n",
       "<title>bigraph</title>\n",
       "<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-99.65 99.65,-99.65 99.65,4 -4,4\"/>\n",
       "<!-- (&#39;hypothesized&lt;br/&gt;mechanism&#39;,) -->\n",
       "<g id=\"node1\" class=\"node\">\n",
       "<title>(&#39;hypothesized&lt;br/&gt;mechanism&#39;,)</title>\n",
       "<ellipse fill=\"none\" stroke=\"blue\" stroke-width=\"2\" cx=\"47.83\" cy=\"-47.83\" rx=\"47.83\" ry=\"47.83\"/>\n",
       "<text text-anchor=\"start\" x=\"16.17\" y=\"-50.23\" font-family=\"Times,serif\" font-size=\"12.00\">hypothesized</text>\n",
       "<text text-anchor=\"start\" x=\"20.5\" y=\"-38.23\" font-family=\"Times,serif\" font-size=\"12.00\">mechanism</text>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x10a4da040>"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_process = {\n",
    "    'hypothesized mechanism': {\n",
    "        '_ports': {\n",
    "            '1': 'Any',\n",
    "            '2': 'Any',\n",
    "        }\n",
    "    }\n",
    "}\n",
    "\n",
    "new_process = replace_regex_recursive(new_process)\n",
    "\n",
    "# plot\n",
    "plot_settingsx = copy.deepcopy(plot_settings)\n",
    "plot_settingsx['label_margin'] = '0.03'\n",
    "plot_settingsx['node_border_colors'] = {\n",
    "    ('hypothesized<br/>mechanism',): 'blue',\n",
    "}\n",
    "plot_settingsx['port_labels'] = False\n",
    "plot_bigraph(new_process, **plot_settingsx, filename='hypothesized_mechanism')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ae52f73c-e384-4880-b18f-47b4c25ce203",
   "metadata": {},
   "source": [
    "### Cell structure and function\n",
    "Processes represent their functions, we show a more minimal cell structure with added processes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "b39417ce-d9a1-4985-abce-ce22df816897",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing out/cell\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 9.0.0 (0)\n",
       " -->\n",
       "<!-- Title: bigraph Pages: 1 -->\n",
       "<svg width=\"582pt\" height=\"405pt\"\n",
       " viewBox=\"0.00 0.00 598.29 417.06\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(0.972222 0.972222) rotate(0) translate(4 413.06)\">\n",
       "<title>bigraph</title>\n",
       "<polygon fill=\"white\" stroke=\"none\" points=\"-4,4 -4,-413.06 594.29,-413.06 594.29,4 -4,4\"/>\n",
       "<!-- (&#39;outputs&#39;,) -->\n",
       "<g id=\"node1\" class=\"node\">\n",
       "<title>(&#39;outputs&#39;,)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"70.39\" cy=\"-383.92\" rx=\"25.14\" ry=\"25.14\"/>\n",
       "<text text-anchor=\"start\" x=\"52.73\" y=\"-380.32\" font-family=\"Times,serif\" font-size=\"12.00\">outputs</text>\n",
       "</g>\n",
       "<!-- (&#39;outputs&#39;, &#39;p1&#39;) -->\n",
       "<g id=\"node2\" class=\"node\">\n",
       "<title>(&#39;outputs&#39;, &#39;p1&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"43.39\" cy=\"-288.58\" rx=\"18\" ry=\"18\"/>\n",
       "<text text-anchor=\"start\" x=\"37.39\" y=\"-284.98\" font-family=\"Times,serif\" font-size=\"12.00\">p1</text>\n",
       "</g>\n",
       "<!-- (&#39;outputs&#39;,)&#45;&gt;(&#39;outputs&#39;, &#39;p1&#39;) -->\n",
       "<g id=\"edge1\" class=\"edge\">\n",
       "<title>(&#39;outputs&#39;,)&#45;&gt;(&#39;outputs&#39;, &#39;p1&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M63.44,-358.88C58.72,-342.57 52.6,-321.39 48.36,-306.76\"/>\n",
       "</g>\n",
       "<!-- (&#39;outputs&#39;, &#39;p2&#39;) -->\n",
       "<g id=\"node3\" class=\"node\">\n",
       "<title>(&#39;outputs&#39;, &#39;p2&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"97.39\" cy=\"-288.58\" rx=\"18\" ry=\"18\"/>\n",
       "<text text-anchor=\"start\" x=\"91.39\" y=\"-284.98\" font-family=\"Times,serif\" font-size=\"12.00\">p2</text>\n",
       "</g>\n",
       "<!-- (&#39;outputs&#39;,)&#45;&gt;(&#39;outputs&#39;, &#39;p2&#39;) -->\n",
       "<g id=\"edge2\" class=\"edge\">\n",
       "<title>(&#39;outputs&#39;,)&#45;&gt;(&#39;outputs&#39;, &#39;p2&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M77.35,-358.88C82.07,-342.57 88.19,-321.39 92.43,-306.76\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;,) -->\n",
       "<g id=\"node4\" class=\"node\">\n",
       "<title>(&#39;cell&#39;,)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"314.39\" cy=\"-383.92\" rx=\"18\" ry=\"18\"/>\n",
       "<text text-anchor=\"start\" x=\"305.73\" y=\"-380.32\" font-family=\"Times,serif\" font-size=\"12.00\">cell</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;membrane&#39;) -->\n",
       "<g id=\"node5\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;membrane&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"167.39\" cy=\"-288.58\" rx=\"34.19\" ry=\"34.19\"/>\n",
       "<text text-anchor=\"start\" x=\"142.07\" y=\"-284.98\" font-family=\"Times,serif\" font-size=\"12.00\">membrane</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;) -->\n",
       "<g id=\"edge12\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M298.94,-373.1C274.69,-357.71 227.29,-327.61 196.5,-308.06\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;) -->\n",
       "<g id=\"node8\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"314.39\" cy=\"-288.58\" rx=\"33.41\" ry=\"33.41\"/>\n",
       "<text text-anchor=\"start\" x=\"289.73\" y=\"-284.98\" font-family=\"Times,serif\" font-size=\"12.00\">cytoplasm</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;) -->\n",
       "<g id=\"edge13\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M314.39,-364.92C314.39,-352.95 314.39,-336.87 314.39,-322.64\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleoid&#39;) -->\n",
       "<g id=\"node15\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleoid&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"496.39\" cy=\"-288.58\" rx=\"28.68\" ry=\"28.68\"/>\n",
       "<text text-anchor=\"start\" x=\"475.73\" y=\"-284.98\" font-family=\"Times,serif\" font-size=\"12.00\">nucleoid</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;nucleoid&#39;) -->\n",
       "<g id=\"edge14\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;,)&#45;&gt;(&#39;cell&#39;, &#39;nucleoid&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M330.87,-374.47C362.63,-358.18 432.56,-322.32 470.57,-302.83\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane&lt;br/&gt;transport&#39;) -->\n",
       "<g id=\"node6\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane&lt;br/&gt;transport&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"57.39\" cy=\"-161\" rx=\"57.39\" ry=\"57.39\"/>\n",
       "<text text-anchor=\"start\" x=\"20.41\" y=\"-163.4\" font-family=\"Times,serif\" font-size=\"12.00\">transmembrane</text>\n",
       "<text text-anchor=\"start\" x=\"36.07\" y=\"-151.4\" font-family=\"Times,serif\" font-size=\"12.00\">transport</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;membrane&#39;)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane&lt;br/&gt;transport&#39;) -->\n",
       "<g id=\"edge4\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;membrane&#39;)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane&lt;br/&gt;transport&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M144.87,-261.87C130.54,-245.5 111.62,-223.91 95.05,-204.99\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane&lt;br/&gt;transport&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"node7\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane&lt;br/&gt;transport&#39;, &#39;wires&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"57.39\" cy=\"-33.8\" rx=\"19.62\" ry=\"19.62\"/>\n",
       "<text text-anchor=\"start\" x=\"44.4\" y=\"-30.2\" font-family=\"Times,serif\" font-size=\"12.00\">wires</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane&lt;br/&gt;transport&#39;)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane&lt;br/&gt;transport&#39;, &#39;wires&#39;) -->\n",
       "<g id=\"edge3\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane&lt;br/&gt;transport&#39;)&#45;&gt;(&#39;cell&#39;, &#39;membrane&#39;, &#39;transmembrane&lt;br/&gt;transport&#39;, &#39;wires&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M57.39,-102.9C57.39,-85.48 57.39,-67.5 57.39,-54.39\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;metabolites&#39;) -->\n",
       "<g id=\"node9\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;metabolites&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"169.39\" cy=\"-161\" rx=\"36.95\" ry=\"36.95\"/>\n",
       "<text text-anchor=\"start\" x=\"141.74\" y=\"-157.4\" font-family=\"Times,serif\" font-size=\"12.00\">metabolites</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;metabolites&#39;) -->\n",
       "<g id=\"edge7\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;metabolites&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M285.53,-270.42C264.9,-257.46 237.06,-238.51 215.39,-218.39 206.78,-210.4 198.36,-200.77 191.12,-191.75\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;ribosomal&lt;br/&gt;complexes&#39;) -->\n",
       "<g id=\"node10\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;ribosomal&lt;br/&gt;complexes&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"265.39\" cy=\"-161\" rx=\"41.38\" ry=\"41.38\"/>\n",
       "<text text-anchor=\"start\" x=\"241.4\" y=\"-163.4\" font-family=\"Times,serif\" font-size=\"12.00\">ribosomal</text>\n",
       "<text text-anchor=\"start\" x=\"239.74\" y=\"-151.4\" font-family=\"Times,serif\" font-size=\"12.00\">complexes</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;ribosomal&lt;br/&gt;complexes&#39;) -->\n",
       "<g id=\"edge8\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;ribosomal&lt;br/&gt;complexes&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M302.28,-256.54C295.65,-239.54 287.38,-218.34 280.35,-200.32\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript&lt;br/&gt;regulation&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"node11\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript&lt;br/&gt;regulation&lt;br/&gt;complex&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"364.39\" cy=\"-161\" rx=\"39.49\" ry=\"39.49\"/>\n",
       "<text text-anchor=\"start\" x=\"341.74\" y=\"-169.4\" font-family=\"Times,serif\" font-size=\"12.00\">transcript</text>\n",
       "<text text-anchor=\"start\" x=\"340.07\" y=\"-157.4\" font-family=\"Times,serif\" font-size=\"12.00\">regulation</text>\n",
       "<text text-anchor=\"start\" x=\"343.73\" y=\"-145.4\" font-family=\"Times,serif\" font-size=\"12.00\">complex</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript&lt;br/&gt;regulation&lt;br/&gt;complex&#39;) -->\n",
       "<g id=\"edge9\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript&lt;br/&gt;regulation&lt;br/&gt;complex&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M326.75,-256.54C333.75,-238.96 342.54,-216.88 349.87,-198.48\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;) -->\n",
       "<g id=\"node13\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"456.39\" cy=\"-161\" rx=\"34.2\" ry=\"34.2\"/>\n",
       "<text text-anchor=\"start\" x=\"431.07\" y=\"-157.4\" font-family=\"Times,serif\" font-size=\"12.00\">translation</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;) -->\n",
       "<g id=\"edge10\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M343.09,-270.51C363.57,-257.59 391.14,-238.66 412.39,-218.39 421.29,-209.91 429.89,-199.55 437.09,-190.04\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript&lt;br/&gt;regulation&lt;br/&gt;complex&#39;, &#39;transcripts&#39;) -->\n",
       "<g id=\"node12\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript&lt;br/&gt;regulation&lt;br/&gt;complex&#39;, &#39;transcripts&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"364.39\" cy=\"-33.8\" rx=\"33.8\" ry=\"33.8\"/>\n",
       "<text text-anchor=\"start\" x=\"339.4\" y=\"-30.2\" font-family=\"Times,serif\" font-size=\"12.00\">transcripts</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript&lt;br/&gt;regulation&lt;br/&gt;complex&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript&lt;br/&gt;regulation&lt;br/&gt;complex&#39;, &#39;transcripts&#39;) -->\n",
       "<g id=\"edge5\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript&lt;br/&gt;regulation&lt;br/&gt;complex&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;transcript&lt;br/&gt;regulation&lt;br/&gt;complex&#39;, &#39;transcripts&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M364.39,-120.73C364.39,-103.95 364.39,-84.56 364.39,-68.45\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;, &#39;outputs&#39;) -->\n",
       "<g id=\"node14\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;, &#39;outputs&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"456.39\" cy=\"-33.8\" rx=\"25.14\" ry=\"25.14\"/>\n",
       "<text text-anchor=\"start\" x=\"438.73\" y=\"-30.2\" font-family=\"Times,serif\" font-size=\"12.00\">outputs</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;, &#39;outputs&#39;) -->\n",
       "<g id=\"edge6\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;)&#45;&gt;(&#39;cell&#39;, &#39;cytoplasm&#39;, &#39;translation&#39;, &#39;outputs&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M456.39,-126.12C456.39,-105.12 456.39,-78.61 456.39,-59.57\"/>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleoid&#39;, &#39;chromosome&#39;) -->\n",
       "<g id=\"node16\" class=\"node\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleoid&#39;, &#39;chromosome&#39;)</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" stroke-width=\"2\" cx=\"549.39\" cy=\"-161\" rx=\"40.89\" ry=\"40.89\"/>\n",
       "<text text-anchor=\"start\" x=\"518.4\" y=\"-157.4\" font-family=\"Times,serif\" font-size=\"12.00\">chromosome</text>\n",
       "</g>\n",
       "<!-- (&#39;cell&#39;, &#39;nucleoid&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleoid&#39;, &#39;chromosome&#39;) -->\n",
       "<g id=\"edge11\" class=\"edge\">\n",
       "<title>(&#39;cell&#39;, &#39;nucleoid&#39;)&#45;&gt;(&#39;cell&#39;, &#39;nucleoid&#39;, &#39;chromosome&#39;)</title>\n",
       "<path fill=\"none\" stroke=\"black\" stroke-width=\"2\" d=\"M507.51,-261.26C515.03,-243.42 525.1,-219.57 533.47,-199.73\"/>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.graphs.Digraph at 0x10a474d00>"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell_struct_func = {\n",
    "    'cell': {\n",
    "        'membrane': {\n",
    "            'transporters': 'concentrations',\n",
    "            'lipids': 'concentrations',\n",
    "            'transmembrane transport': {\n",
    "                '_value': {\n",
    "                    '_process': 'transport URI',\n",
    "                    '_config': {'parameter': 1}\n",
    "                },\n",
    "                'outputs': {\n",
    "                    'transporters': 'transporters',\n",
    "                    'internal': ['..', 'cytoplasm', 'metabolites']},\n",
    "                '_outputs': {\n",
    "                    'transporters': 'concentrations',\n",
    "                    'internal': 'concentrations',\n",
    "                    'external': 'concentrations'\n",
    "                }\n",
    "            }\n",
    "        },\n",
    "        'cytoplasm': {\n",
    "            'metabolites': {\n",
    "                '_value': 1.1,\n",
    "                '_type': 'concentrations'\n",
    "            },\n",
    "            'ribosomal complexes': {\n",
    "                '_value': 2.2,\n",
    "                '_type': 'concentrations'\n",
    "            },\n",
    "            'transcript regulation complex': {\n",
    "                '_value': 0.01,\n",
    "                'transcripts': {\n",
    "                    '_value': 0.1,\n",
    "                    '_type': 'concentrations'\n",
    "                }\n",
    "            },\n",
    "            'translation': {\n",
    "                '_type': 'process',\n",
    "                'outputs': {\n",
    "                    'p1': ['ribosomal complexes'],\n",
    "                    'p2': ['transcript regulation complex', 'transcripts']}}},\n",
    "        'nucleoid': {\n",
    "            'chromosome': {\n",
    "                'genes': 'sequences'\n",
    "            }\n",
    "        }\n",
    "    }\n",
    "}\n",
    "core = TypeSystem()\n",
    "core.register('concentrations', 'float')\n",
    "cell_struct_func2 = replace_regex_recursive(cell_struct_func)\n",
    "plot_bigraph(cell_struct_func2, core=core, remove_process_place_edges=True, out_dir='out', filename='cell')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "188eb4a6-c526-4b38-a22d-97f7fb8b7244",
   "metadata": {},
   "source": [
    "### Whole-cell *E. coli* model\n",
    "\n",
    "We here plot the processes and wiring of a whole-cell model of *E. coli* from [Macklin et al. Simultaneous cross-evaluation of heterogeneous E. coli datasets via mechanistic simulation. Science. 2020.](https://pubmed.ncbi.nlm.nih.gov/32703847/)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "16960d1b-6e35-421e-bb7d-3efe5052e4fc",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "ecoli = { \n",
    "    'chromosome-structure': { \n",
    "        'wires': {\n",
    "            'fragmentBases': ('bulk',),\n",
    "            'molecules': ('bulk',),\n",
    "            'active_tfs': ('bulk',),\n",
    "            'subunits': ('bulk',),\n",
    "            'amino_acids': ('bulk',),\n",
    "            'active_replisomes': ('unique', 'active_replisome'),\n",
    "            'oriCs': ('unique', 'oriC'),\n",
    "            'chromosome_domains': ('unique', 'chromosome_domain'),\n",
    "            'active_RNAPs': ('unique', 'active_RNAP'),\n",
    "            'RNAs': ('unique', 'RNA'),\n",
    "            'active_ribosome': ('unique', 'active_ribosome'),\n",
    "            'full_chromosomes': ('unique', 'full_chromosome'),\n",
    "            'promoters': ('unique', 'promoter'),\n",
    "            'DnaA_boxes': ('unique', 'DnaA_box')\n",
    "        }\n",
    "    },\n",
    "    'metabolism': { \n",
    "        'wires': {\n",
    "            'metabolites': ('bulk',),\n",
    "            'catalysts': ('bulk',),\n",
    "            'kinetics_enzymes': ('bulk',),\n",
    "            'kinetics_substrates': ('bulk',),\n",
    "            'amino_acids': ('bulk',),\n",
    "            'environment': ('environment',),\n",
    "            'amino_acids_total': ('bulk',)\n",
    "        }\n",
    "    },\n",
    "    'tf-binding': { \n",
    "        'wires': {\n",
    "            'promoters': ('unique', 'promoter'),\n",
    "            'active_tfs': ('bulk',),\n",
    "            'active_tfs_total': ('bulk',),\n",
    "            'inactive_tfs_total': ('bulk',),\n",
    "        }\n",
    "    },\n",
    "    'transcript-initiation': { \n",
    "        'wires': {\n",
    "            'environment': ('environment',),\n",
    "            'full_chromosomes': ('unique', 'full_chromosome'),\n",
    "            'RNAs': ('unique', 'RNA'),\n",
    "            'active_RNAPs': ('unique', 'active_RNAP'),\n",
    "            'promoters': ('unique', 'promoter'),\n",
    "            'molecules': ('bulk',),\n",
    "        }\n",
    "    },\n",
    "    'transcript-elongation': { \n",
    "        'wires': {\n",
    "            'environment': ('environment',),\n",
    "            'RNAs': ('unique', 'RNA'),\n",
    "            'active_RNAPs': ('unique', 'active_RNAP'),\n",
    "            'molecules': ('bulk',),\n",
    "            'bulk_RNAs': ('bulk',),\n",
    "            'ntps': ('bulk',),\n",
    "        }\n",
    "    },\n",
    "    'rna-degradation': { \n",
    "        'wires': {\n",
    "            'charged_trna': ('bulk',),\n",
    "            'bulk_RNAs': ('bulk',),\n",
    "            'nmps': ('bulk',),\n",
    "            'fragmentMetabolites': ('bulk',),\n",
    "            'fragmentBases': ('bulk',),\n",
    "            'endoRnases': ('bulk',),\n",
    "            'exoRnases': ('bulk',),\n",
    "            'subunits': ('bulk',),\n",
    "            'molecules': ('bulk',),\n",
    "            'RNAs': ('unique', 'RNA'),\n",
    "            'active_ribosome': ('unique', 'active_ribosome'),\n",
    "        }\n",
    "    },\n",
    "    'polypeptide-initiation': { \n",
    "        'wires': {\n",
    "            'environment': ('environment',),\n",
    "            'active_ribosome': ('unique', 'active_ribosome'),\n",
    "            'RNA': ('unique', 'RNA'),\n",
    "            'subunits': ('bulk',)\n",
    "        }\n",
    "    },\n",
    "    'polypeptide-elongation': { \n",
    "        'wires': {\n",
    "            'environment': ('environment',),\n",
    "            'active_ribosome': ('unique', 'active_ribosome'),\n",
    "            'molecules': ('bulk',),\n",
    "            'monomers': ('bulk',),\n",
    "            'amino_acids': ('bulk',),\n",
    "            'ppgpp_reaction_metabolites': ('bulk',),\n",
    "            'uncharged_trna': ('bulk',),\n",
    "            'charged_trna': ('bulk',),\n",
    "            'charging_molecules': ('bulk',),\n",
    "            'synthetases': ('bulk',),\n",
    "            'subunits': ('bulk',),\n",
    "            'molecules_total': ('bulk',),\n",
    "            'amino_acids_total': ('bulk',),\n",
    "            'charged_trna_total': ('bulk',),\n",
    "            'uncharged_trna_total': ('bulk',)\n",
    "        }\n",
    "    },\n",
    "    'complexation': {\n",
    "        'wires': {\n",
    "            'molecules': ('bulk',), \n",
    "        }\n",
    "    },\n",
    "    'two-component-system': { \n",
    "        'wires': {\n",
    "            'molecules': ('bulk',)\n",
    "        }\n",
    "    },\n",
    "    'equilibrium': {\n",
    "        'wires': {\n",
    "            'molecules': ('bulk',)\n",
    "        }\n",
    "    },\n",
    "    'protein-degradation': { \n",
    "        'wires': {\n",
    "            'metabolites': ('bulk',),\n",
    "            'proteins': ('bulk',)\n",
    "        }\n",
    "    },\n",
    "    'chromosome-replication': { \n",
    "        'wires': {\n",
    "            'replisome_trimers': ('bulk',),\n",
    "            'replisome_monomers': ('bulk',),\n",
    "            'dntps': ('bulk',),\n",
    "            'ppi': ('bulk',),\n",
    "            'active_replisomes': ('unique', 'active_replisome'),\n",
    "            'oriCs': ('unique', 'oriC'),\n",
    "            'chromosome_domains': ('unique', 'chromosome_domain'),\n",
    "            'full_chromosomes': ('unique', 'full_chromosome'),\n",
    "            'environment': ('environment',)\n",
    "        }\n",
    "    },\n",
    "    'unique': {\n",
    "        'chromosome_domain': {},\n",
    "        'full_chromosome': {},\n",
    "        'oriC': {},\n",
    "        'active_replisome': {},\n",
    "        'RNA': {},\n",
    "        'active_ribosome': {},\n",
    "        'DnaA_box': {},\n",
    "        'promoter': {},\n",
    "    },\n",
    "    'bulk': {},\n",
    "    'environment': {},\n",
    "}\n",
    "node_groups = [[ #line up the top nodes for visual effect\n",
    "        ('unique',), ('bulk',), ('environment',), \n",
    "    ],\n",
    "]\n",
    "# plot settings\n",
    "plot_settings_ecoli = {\n",
    "    'remove_process_place_edges': True,\n",
    "    'rankdir': 'RL',\n",
    "    'dpi': '100',\n",
    "    'out_dir': 'out',\n",
    "}\n",
    "fig = plot_bigraph(ecoli, **plot_settings_ecoli, node_groups=node_groups, filename='ecoli')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "54e1102d-521c-49cc-8ed2-be26c2b84c3d",
   "metadata": {
    "tags": []
   },
   "source": [
    "[See E. coli wiring diagram](https://github.com/vivarium-collective/bigraph-viz/blob/main/doc/_static/ecoli.png?raw=true)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
